// starknetkit/dist/connector-d58f0dd8.js
var C = Object.defineProperty;
var O = (s, u, o) => u in s ? C(s, u, { enumerable: !0, configurable: !0, writable: !0, value: o }) : s[u] = o;
var d = (s, u, o) => (O(s, typeof u != "symbol" ? u + "" : u, o), o);
import "starknet";
class A extends Error {
  constructor() {
    super(...arguments);
    d(this, "name", "ConnectorNotConnectedError");
    d(this, "message", "Connector not connected");
  }
}
class R extends Error {
  constructor() {
    super(...arguments);
    d(this, "name", "ConnectorNotFoundError");
    d(this, "message", "Connector not found");
  }
}
class T extends Error {
  constructor() {
    super(...arguments);
    d(this, "name", "UserRejectedRequestError");
    d(this, "message", "User rejected request");
  }
}
class k extends Error {
  constructor() {
    super(...arguments);
    d(this, "name", "UserNotConnectedError");
    d(this, "message", "User not connected");
  }
}
var q = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function j(s) {
  return s && s.__esModule && Object.prototype.hasOwnProperty.call(s, "default") ? s.default : s;
}
function F(s) {
  if (s.__esModule)
    return s;
  var u = s.default;
  if (typeof u == "function") {
    var o = function v() {
      return this instanceof v ? Reflect.construct(u, arguments, this.constructor) : u.apply(this, arguments);
    };
    o.prototype = u.prototype;
  } else
    o = {};
  return Object.defineProperty(o, "__esModule", { value: !0 }), Object.keys(s).forEach(function(v) {
    var g = Object.getOwnPropertyDescriptor(s, v);
    Object.defineProperty(o, v, g.get ? g : {
      enumerable: !0,
      get: function() {
        return s[v];
      }
    });
  }), o;
}
var w = { exports: {} };
(function(s) {
  var u = Object.prototype.hasOwnProperty, o = "~";
  function v() {
  }
  Object.create && (v.prototype = /* @__PURE__ */ Object.create(null), new v().__proto__ || (o = !1));
  function g(c, t, n) {
    this.fn = c, this.context = t, this.once = n || !1;
  }
  function b(c, t, n, r, p) {
    if (typeof n != "function")
      throw new TypeError("The listener must be a function");
    var l = new g(n, r || c, p), i = o ? o + t : t;
    return c._events[i] ? c._events[i].fn ? c._events[i] = [c._events[i], l] : c._events[i].push(l) : (c._events[i] = l, c._eventsCount++), c;
  }
  function E(c, t) {
    --c._eventsCount === 0 ? c._events = new v() : delete c._events[t];
  }
  function a() {
    this._events = new v(), this._eventsCount = 0;
  }
  a.prototype.eventNames = function() {
    var t = [], n, r;
    if (this._eventsCount === 0)
      return t;
    for (r in n = this._events)
      u.call(n, r) && t.push(o ? r.slice(1) : r);
    return Object.getOwnPropertySymbols ? t.concat(Object.getOwnPropertySymbols(n)) : t;
  }, a.prototype.listeners = function(t) {
    var n = o ? o + t : t, r = this._events[n];
    if (!r)
      return [];
    if (r.fn)
      return [r.fn];
    for (var p = 0, l = r.length, i = new Array(l); p < l; p++)
      i[p] = r[p].fn;
    return i;
  }, a.prototype.listenerCount = function(t) {
    var n = o ? o + t : t, r = this._events[n];
    return r ? r.fn ? 1 : r.length : 0;
  }, a.prototype.emit = function(t, n, r, p, l, i) {
    var h = o ? o + t : t;
    if (!this._events[h])
      return !1;
    var e = this._events[h], y = arguments.length, _, f;
    if (e.fn) {
      switch (e.once && this.removeListener(t, e.fn, void 0, !0), y) {
        case 1:
          return e.fn.call(e.context), !0;
        case 2:
          return e.fn.call(e.context, n), !0;
        case 3:
          return e.fn.call(e.context, n, r), !0;
        case 4:
          return e.fn.call(e.context, n, r, p), !0;
        case 5:
          return e.fn.call(e.context, n, r, p, l), !0;
        case 6:
          return e.fn.call(e.context, n, r, p, l, i), !0;
      }
      for (f = 1, _ = new Array(y - 1); f < y; f++)
        _[f - 1] = arguments[f];
      e.fn.apply(e.context, _);
    } else {
      var x = e.length, m;
      for (f = 0; f < x; f++)
        switch (e[f].once && this.removeListener(t, e[f].fn, void 0, !0), y) {
          case 1:
            e[f].fn.call(e[f].context);
            break;
          case 2:
            e[f].fn.call(e[f].context, n);
            break;
          case 3:
            e[f].fn.call(e[f].context, n, r);
            break;
          case 4:
            e[f].fn.call(e[f].context, n, r, p);
            break;
          default:
            if (!_)
              for (m = 1, _ = new Array(y - 1); m < y; m++)
                _[m - 1] = arguments[m];
            e[f].fn.apply(e[f].context, _);
        }
    }
    return !0;
  }, a.prototype.on = function(t, n, r) {
    return b(this, t, n, r, !1);
  }, a.prototype.once = function(t, n, r) {
    return b(this, t, n, r, !0);
  }, a.prototype.removeListener = function(t, n, r, p) {
    var l = o ? o + t : t;
    if (!this._events[l])
      return this;
    if (!n)
      return E(this, l), this;
    var i = this._events[l];
    if (i.fn)
      i.fn === n && (!p || i.once) && (!r || i.context === r) && E(this, l);
    else {
      for (var h = 0, e = [], y = i.length; h < y; h++)
        (i[h].fn !== n || p && !i[h].once || r && i[h].context !== r) && e.push(i[h]);
      e.length ? this._events[l] = e.length === 1 ? e[0] : e : E(this, l);
    }
    return this;
  }, a.prototype.removeAllListeners = function(t) {
    var n;
    return t ? (n = o ? o + t : t, this._events[n] && E(this, n)) : (this._events = new v(), this._eventsCount = 0), this;
  }, a.prototype.off = a.prototype.removeListener, a.prototype.addListener = a.prototype.on, a.prefixed = o, a.EventEmitter = a, s.exports = a;
})(w);
var N = w.exports;
const L = /* @__PURE__ */ j(N);
class M extends L {
  /** Handle starknet-react type compatibility */
  eventNames() {
    return ["change", "connect", "disconnect"];
  }
}
export {
  M as C,
  T as U,
  A as a,
  R as b,
  q as c,
  j as d,
  k as e,
  F as g
};
