// starknetkit/dist/Modal-4ccf5985.js
var _r = Object.defineProperty;
var yr = (l, e, o) => e in l ? _r(l, e, { enumerable: !0, configurable: !0, writable: !0, value: o }) : l[e] = o;
var Bt = (l, e, o) => (yr(l, typeof e != "symbol" ? e + "" : e, o), o);
import "./connector-d58f0dd8.js";
import "starknet";
import "./trpc-870d4938.js";
import { a as xr, e as Fr, i as Lr, b as Gt, g as Br } from "./index-eca87b80.js";
import { ArgentX as Mr } from "./argentX.js";
import { Braavos as Sr } from "./braavos.js";
var Q = /* @__PURE__ */ ((l) => (l.walletList = "walletList", l.connecting = "connecting", l.success = "success", l.loginFailure = "loginFailure", l.requestFailure = "requestFailure", l.qrCode = "qrCode", l.download = "download", l.approval = "approval", l))(Q || {});
const $r = `@import"https://fonts.googleapis.com/css2?family=Barlow:wght@500;600&display=swap";.visible{visibility:visible}.static{position:static}.fixed{position:fixed}.absolute{position:absolute}.relative{position:relative}.sticky{position:sticky}.inset-0{inset:0}.left-0{left:0}.right-0{right:0}.top-0{top:0}.z-\\[9998\\]{z-index:9998}.z-\\[9999\\]{z-index:9999}.mx-6{margin-left:1.5rem;margin-right:1.5rem}.mb-4{margin-bottom:1rem}.ml-\\[-32px\\]{margin-left:-32px}.block{display:block}.inline-block{display:inline-block}.inline{display:inline}.flex{display:flex}.inline-flex{display:inline-flex}.table{display:table}.hidden{display:none}.h-12{height:3rem}.h-8{height:2rem}.h-\\[245px\\]{height:245px}.h-\\[32px\\]{height:32px}.h-\\[60px\\]{height:60px}.h-full{height:100%}.min-h-12{min-height:3rem}.min-h-8{min-height:2rem}.min-h-\\[26px\\]{min-height:26px}.min-h-\\[570px\\]{min-height:570px}.w-12{width:3rem}.w-8{width:2rem}.w-full{width:100%}.min-w-\\[140px\\]{min-width:140px}.max-w-\\[240px\\]{max-width:240px}.max-w-\\[32px\\]{max-width:32px}.max-w-\\[380px\\]{max-width:380px}.flex-shrink{flex-shrink:1}.flex-grow{flex-grow:1}.transform{transform:translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skew(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))}.cursor-pointer{cursor:pointer}.resize{resize:both}.flex-row{flex-direction:row}.flex-row-reverse{flex-direction:row-reverse}.flex-col{flex-direction:column}.items-end{align-items:flex-end}.items-center{align-items:center}.justify-center{justify-content:center}.justify-between{justify-content:space-between}.gap-1{gap:.25rem}.gap-2{gap:.5rem}.gap-3{gap:.75rem}.gap-4{gap:1rem}.gap-8{gap:2rem}.overflow-hidden{overflow:hidden}.text-ellipsis{text-overflow:ellipsis}.whitespace-nowrap{white-space:nowrap}.rounded{border-radius:.25rem}.rounded-3xl{border-radius:1.5rem}.rounded-\\[12px\\]{border-radius:12px}.rounded-full{border-radius:9999px}.rounded-xl{border-radius:.75rem}.border{border-width:1px}.border-stroke-default{border-color:var(--color-stroke-default)}.border-stroke-focused{border-color:var(--color-stroke-focused)}.bg-black\\/25{background-color:#00000040}.bg-button-danger{background-color:var(--color-button-danger)}.bg-button-primary{background-color:var(--color-button-primary)}.bg-button-secondary{background-color:var(--color-button-secondary)}.bg-button-tertiary{background-color:var(--color-button-tertiary)}.bg-surface-default{background-color:var(--color-surface-default)}.bg-surface-elevated{background-color:var(--color-surface-elevated)}.bg-surface-elevated-web{background-color:var(--color-surface-elevated-web)}.stroke-brand{stroke:var(--color-text-brand)}.stroke-danger{stroke:var(--color-text-danger)}.stroke-subtle{stroke:var(--color-text-subtle)}.stroke-success{stroke:var(--color-text-success)}.stroke-warning{stroke:var(--color-text-warning)}.p-2{padding:.5rem}.p-5{padding:1.25rem}.p-6{padding:1.5rem}.px-3{padding-left:.75rem;padding-right:.75rem}.px-4{padding-left:1rem;padding-right:1rem}.px-8{padding-left:2rem;padding-right:2rem}.py-3{padding-top:.75rem;padding-bottom:.75rem}.text-left{text-align:left}.text-center{text-align:center}.text-\\[13px\\]{font-size:13px}.text-\\[15px\\]{font-size:15px}.text-\\[24px\\]{font-size:24px}.text-b1{font-size:1.125rem;line-height:1.125rem;font-weight:600}.text-b2{font-size:.9375rem;line-height:1rem;font-weight:600}.text-b3{font-size:.8125rem;line-height:.875rem;font-weight:600}.text-h4{font-size:1.375rem;line-height:1.5625rem;font-weight:700}.text-h5,.text-heading5{font-size:1.125rem;line-height:1.375rem;font-weight:600}.text-l1{font-size:.875rem;line-height:1.125rem;font-weight:400}.text-l2{font-size:.75rem;line-height:1rem;font-weight:400}.text-p3{font-size:1rem;line-height:1.3125rem;font-weight:400}.font-\\[400\\]{font-weight:400}.font-bold{font-weight:700}.font-semibold{font-weight:600}.capitalize{text-transform:capitalize}.text-adaptive{color:var(--color-text-adaptive)}.text-brand{color:var(--color-text-brand)}.text-danger{color:var(--color-text-danger)}.text-danger-accessible{color:var(--color-text-danger-accessible)}.text-info{color:var(--color-text-info)}.text-info-accessible{color:var(--color-text-info-accessible)}.text-link{color:var(--color-text-link)}.text-link-pressed{color:var(--color-text-link-pressed)}.text-primary{color:var(--color-text-primary)}.text-secondary{color:var(--color-text-secondary)}.text-secondary-web{color:var(--color-text-secondary-web)}.text-stable{color:var(--color-text-stable)}.text-subtle{color:var(--color-text-subtle)}.text-success{color:var(--color-text-success)}.text-success-accessible{color:var(--color-text-success-accessible)}.text-warning{color:var(--color-text-warning)}.text-warning-accessible{color:var(--color-text-warning-accessible)}.underline{text-decoration-line:underline}.line-through{text-decoration-line:line-through}.accent-brand{accent-color:var(--color-text-brand)}.shadow{--tw-shadow: 0 1px 3px 0 rgb(0 0 0 / .1), 0 1px 2px -1px rgb(0 0 0 / .1);--tw-shadow-colored: 0 1px 3px 0 var(--tw-shadow-color), 0 1px 2px -1px var(--tw-shadow-color);box-shadow:var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow)}.shadow-list-item{--tw-shadow: 0px 4px 12px 0px rgba(0, 0, 0, .06);--tw-shadow-colored: 0px 4px 12px 0px var(--tw-shadow-color);box-shadow:var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow)}.shadow-modal{--tw-shadow: 0px 4px 20px rgba(0, 0, 0, .5);--tw-shadow-colored: 0px 4px 20px var(--tw-shadow-color);box-shadow:var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow)}.outline{outline-style:solid}.blur{--tw-blur: blur(8px);filter:var(--tw-blur) var(--tw-brightness) var(--tw-contrast) var(--tw-grayscale) var(--tw-hue-rotate) var(--tw-invert) var(--tw-saturate) var(--tw-sepia) var(--tw-drop-shadow)}.drop-shadow{--tw-drop-shadow: drop-shadow(0 1px 2px rgb(0 0 0 / .1)) drop-shadow(0 1px 1px rgb(0 0 0 / .06));filter:var(--tw-blur) var(--tw-brightness) var(--tw-contrast) var(--tw-grayscale) var(--tw-hue-rotate) var(--tw-invert) var(--tw-saturate) var(--tw-sepia) var(--tw-drop-shadow)}.filter{filter:var(--tw-blur) var(--tw-brightness) var(--tw-contrast) var(--tw-grayscale) var(--tw-hue-rotate) var(--tw-invert) var(--tw-saturate) var(--tw-sepia) var(--tw-drop-shadow)}.backdrop-blur-sm{--tw-backdrop-blur: blur(4px);-webkit-backdrop-filter:var(--tw-backdrop-blur) var(--tw-backdrop-brightness) var(--tw-backdrop-contrast) var(--tw-backdrop-grayscale) var(--tw-backdrop-hue-rotate) var(--tw-backdrop-invert) var(--tw-backdrop-opacity) var(--tw-backdrop-saturate) var(--tw-backdrop-sepia);backdrop-filter:var(--tw-backdrop-blur) var(--tw-backdrop-brightness) var(--tw-backdrop-contrast) var(--tw-backdrop-grayscale) var(--tw-backdrop-hue-rotate) var(--tw-backdrop-invert) var(--tw-backdrop-opacity) var(--tw-backdrop-saturate) var(--tw-backdrop-sepia)}.transition{transition-property:color,background-color,border-color,text-decoration-color,fill,stroke,opacity,box-shadow,transform,filter,-webkit-backdrop-filter;transition-property:color,background-color,border-color,text-decoration-color,fill,stroke,opacity,box-shadow,transform,filter,backdrop-filter;transition-property:color,background-color,border-color,text-decoration-color,fill,stroke,opacity,box-shadow,transform,filter,backdrop-filter,-webkit-backdrop-filter;transition-timing-function:cubic-bezier(.4,0,.2,1);transition-duration:.15s}.transition-colors{transition-property:color,background-color,border-color,text-decoration-color,fill,stroke;transition-timing-function:cubic-bezier(.4,0,.2,1);transition-duration:.15s}.duration-100{transition-duration:.1s}.ease-in{transition-timing-function:cubic-bezier(.4,0,1,1)}.ease-in-out{transition-timing-function:cubic-bezier(.4,0,.2,1)}.ease-out{transition-timing-function:cubic-bezier(0,0,.2,1)}.container{width:100%}@media (min-width: 640px){.container{max-width:640px}}@media (min-width: 768px){.container{max-width:768px}}@media (min-width: 1024px){.container{max-width:1024px}}@media (min-width: 1280px){.container{max-width:1280px}}@media (min-width: 1536px){.container{max-width:1536px}}*,:before,:after{box-sizing:border-box;border-width:0;border-style:solid;border-color:#e5e7eb}:before,:after{--tw-content: ""}html,:host{line-height:1.5;-webkit-text-size-adjust:100%;-moz-tab-size:4;-o-tab-size:4;tab-size:4;font-family:ui-sans-serif,system-ui,sans-serif,"Apple Color Emoji","Segoe UI Emoji",Segoe UI Symbol,"Noto Color Emoji";font-feature-settings:normal;font-variation-settings:normal;-webkit-tap-highlight-color:transparent}body{margin:0;line-height:inherit}hr{height:0;color:inherit;border-top-width:1px}abbr:where([title]){-webkit-text-decoration:underline dotted;text-decoration:underline dotted}h1,h2,h3,h4,h5,h6{font-size:inherit;font-weight:inherit}a{color:inherit;text-decoration:inherit}b,strong{font-weight:bolder}code,kbd,samp,pre{font-family:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace;font-feature-settings:normal;font-variation-settings:normal;font-size:1em}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}table{text-indent:0;border-color:inherit;border-collapse:collapse}button,input,optgroup,select,textarea{font-family:inherit;font-feature-settings:inherit;font-variation-settings:inherit;font-size:100%;font-weight:inherit;line-height:inherit;letter-spacing:inherit;color:inherit;margin:0;padding:0}button,select{text-transform:none}button,input:where([type=button]),input:where([type=reset]),input:where([type=submit]){-webkit-appearance:button;background-color:transparent;background-image:none}:-moz-focusring{outline:auto}:-moz-ui-invalid{box-shadow:none}progress{vertical-align:baseline}::-webkit-inner-spin-button,::-webkit-outer-spin-button{height:auto}[type=search]{-webkit-appearance:textfield;outline-offset:-2px}::-webkit-search-decoration{-webkit-appearance:none}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}summary{display:list-item}blockquote,dl,dd,h1,h2,h3,h4,h5,h6,hr,figure,p,pre{margin:0}fieldset{margin:0;padding:0}legend{padding:0}ol,ul,menu{list-style:none;margin:0;padding:0}dialog{padding:0}textarea{resize:vertical}input::-moz-placeholder,textarea::-moz-placeholder{opacity:1;color:#9ca3af}input::placeholder,textarea::placeholder{opacity:1;color:#9ca3af}button,[role=button]{cursor:pointer}:disabled{cursor:default}img,svg,video,canvas,audio,iframe,embed,object{display:block;vertical-align:middle}img,video{max-width:100%;height:auto}[hidden]{display:none}*,:before,:after{--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position: ;--tw-gradient-via-position: ;--tw-gradient-to-position: ;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / .5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: ;--tw-contain-size: ;--tw-contain-layout: ;--tw-contain-paint: ;--tw-contain-style: }::backdrop{--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position: ;--tw-gradient-via-position: ;--tw-gradient-to-position: ;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / .5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: ;--tw-contain-size: ;--tw-contain-layout: ;--tw-contain-paint: ;--tw-contain-style: }:root,:host{--color-black-0: 0, 0, 0, 0;--color-black-4: 0, 0, 0, .04;--color-black-6: 0, 0, 0, .06;--color-black-12: 0, 0, 0, .12;--color-black-30: 0, 0, 0, .3;--color-black-50: 0, 0, 0, .5;--color-black-100: 0 0 0;--color-deprecated-accent-accessible-green: 2 161 122;--color-deprecated-accent-accessible-sky-blue: 0 157 214;--color-deprecated-accent-accessible-yellow: 255 171 0;--color-deprecated-accent-blue: 0 120 164;--color-deprecated-accent-coral: 255 103 92;--color-deprecated-accent-green: 44 211 169;--color-deprecated-accent-hot-pink: 255 91 129;--color-deprecated-accent-light-orange: 255 168 92;--color-deprecated-accent-red-text-on-black: 219 59 78;--color-deprecated-accent-red-text-on-white: 193 32 38;--color-deprecated-accent-sky-blue: 41 197 255;--color-deprecated-accent-yellow: 255 191 61;--color-deprecated-black: 0 0 0;--color-deprecated-black-3: 0, 0, 0, .03;--color-deprecated-black-8: 0, 0, 0, .08;--color-deprecated-black-30: 0, 0, 0, .3;--color-deprecated-black-50: 0, 0, 0, .5;--color-deprecated-dark-green-shade: 55 129 118;--color-deprecated-darker-blue-shade: 33 83 112;--color-deprecated-darker-orange: 72 42 30;--color-deprecated-experimental-vault: 0 48 77;--color-deprecated-lighter-blue-shade: 232 242 246;--color-deprecated-lighter-sky-blue: 241 251 255;--color-deprecated-neutrals-100: 249 249 249;--color-deprecated-neutrals-200: 240 240 240;--color-deprecated-neutrals-300: 191 191 191;--color-deprecated-neutrals-400: 140 140 140;--color-deprecated-neutrals-500: 102 102 102;--color-deprecated-neutrals-600: 89 89 89;--color-deprecated-neutrals-700: 64 64 64;--color-deprecated-neutrals-800: 38 38 38;--color-deprecated-neutrals-900: 31 31 31;--color-deprecated-neutrals-1000: 15 15 15;--color-deprecated-primary-accessible-orange: 243 106 61;--color-deprecated-primary-orange: 255 135 91;--color-deprecated-white: 255 255 255;--color-deprecated-white-12: 255, 255, 255, .12;--color-deprecated-white-30: 255, 255, 255, .3;--color-deprecated-white-50: 255, 255, 255, .5;--color-neutrals-100: 249 249 249;--color-neutrals-200: 240 240 240;--color-neutrals-300: 191 191 191;--color-neutrals-400: 140 140 140;--color-neutrals-500: 102 102 102;--color-neutrals-600: 89 89 89;--color-neutrals-700: 64 64 64;--color-neutrals-800: 46 46 46;--color-neutrals-900: 36 36 36;--color-neutrals-1000: 23 23 23;--color-primary-blue-200: 203 241 255;--color-primary-blue-400: 41 197 255;--color-primary-blue-600: 0 157 214;--color-primary-blue-800: 0 120 164;--color-primary-blue-1000: 7 49 68;--color-primary-green-200: 206 255 243;--color-primary-green-400: 44 211 169;--color-primary-green-600: 2 161 122;--color-primary-green-800: 10 77 59;--color-primary-green-1000: 4 36 16;--color-primary-orange-6: 255, 135, 91, .06;--color-primary-orange-18: 255, 135, 91, .18;--color-primary-orange-200: 255 203 184;--color-primary-orange-300: 255 158 122;--color-primary-orange-400: 255 135 91;--color-primary-orange-600: 243 106 61;--color-primary-orange-800: 128 56 32;--color-primary-orange-1000: 55 23 9;--color-primary-red-200: 255 221 225;--color-primary-red-400: 233 95 114;--color-primary-red-600: 219 59 78;--color-primary-red-800: 99 9 21;--color-primary-red-1000: 75 1 7;--color-primary-red-dark: 193 32 38;--color-primary-yellow-200: 252 239 211;--color-primary-yellow-400: 255 191 61;--color-primary-yellow-600: 255 171 0;--color-primary-yellow-800: 233 129 1;--color-primary-yellow-1000: 63 45 12;--color-screen-border: 255, 255, 255, .2;--color-secondary-clementine: 255 168 92;--color-secondary-coral: 255 103 92;--color-secondary-hot-pink: 255 91 129;--color-secondary-sky-blue: 41 197 255;--color-white-0: 255, 255, 255, 0;--color-white-6: 255, 255, 255, .06;--color-white-12: 255, 255, 255, .12;--color-white-18: 255, 255, 255, .18;--color-white-30: 255, 255, 255, .3;--color-white-50: 255, 255, 255, .5;--color-white-100: 255 255 255;--color-accent-blue: rgba(var(--color-primary-blue-600));--color-accent-brand: rgba(var(--color-primary-orange-400));--color-accent-coral: rgba(var(--color-secondary-coral));--color-accent-green: rgba(var(--color-primary-green-400));--color-accent-hot-pink: rgba(var(--color-secondary-hot-pink));--color-accent-orange: rgba(var(--color-secondary-clementine));--color-accent-red: rgba(var(--color-primary-red-600));--color-accent-sky-blue: rgba(var(--color-secondary-sky-blue));--color-accent-yellow: rgba(var(--color-primary-yellow-400));--color-button-danger: rgba(var(--color-primary-red-600));--color-button-danger-hover: rgba(var(--color-primary-red-400));--color-button-info: rgba(var(--color-primary-blue-800));--color-button-info-hover: rgba(var(--color-primary-blue-600));--color-button-pressed-ghost-overlay: rgba(var(--color-black-12));--color-button-pressed-overlay: rgba(var(--color-black-12));--color-button-primary: rgba(var(--color-primary-orange-600));--color-button-primary-hover: rgba(var(--color-primary-orange-400));--color-button-secondary: rgba(var(--color-black-6));--color-button-secondary-hover: rgba(var(--color-black-12));--color-button-tertiary: rgba(var(--color-black-0));--color-button-tertiary-hover: rgba(var(--color-primary-orange-6));--color-button-tertiary-pressed: rgba(var(--color-primary-orange-18));--color-button-warning: rgba(var(--color-primary-yellow-800));--color-button-warning-hover: rgba(var(--color-primary-yellow-600));--color-icon-adaptive: rgba(var(--color-white-100));--color-icon-background: rgba(var(--color-black-6));--color-icon-background-brand: rgba(var(--color-primary-orange-400));--color-icon-brand: rgba(var(--color-primary-orange-600));--color-icon-danger: rgba(var(--color-primary-red-600));--color-icon-default: rgba(var(--color-neutrals-700));--color-icon-info: rgba(var(--color-primary-blue-600));--color-icon-secondary: rgba(var(--color-neutrals-400));--color-icon-stable: rgba(var(--color-white-100));--color-icon-subtle: rgba(var(--color-neutrals-300));--color-icon-success: rgba(var(--color-primary-green-600));--color-icon-warning: rgba(var(--color-primary-yellow-800));--color-ios-icon: rgb(80, 85, 92);--color-ios-keyboard-button: rgb(0, 122, 255);--color-ios-primary: rgb(0, 122, 255);--color-ios-surface-dark: rgb(172, 180, 190);--color-ios-surface-default: rgba(210, 213, 219, .9);--color-ios-surface-elevated: rgb(255, 255, 255);--color-ios-text: rgb(0, 0, 0);--color-ios-text-inverted: rgb(255, 255, 255);--color-shadow-default: rgba(var(--color-black-6));--color-shadow-medium: rgba(var(--color-black-12));--color-shadow-strong: rgba(var(--color-black-30));--color-stroke-brand: rgba(var(--color-primary-orange-600));--color-stroke-danger: rgba(var(--color-primary-red-600));--color-stroke-default: rgba(var(--color-neutrals-200));--color-stroke-default-web: rgba(var(--color-neutrals-200));--color-stroke-focused: rgba(var(--color-black-30));--color-stroke-subtle: rgba(var(--color-black-6));--color-stroke-success: rgba(var(--color-primary-green-600));--color-stroke-warning: rgba(var(--color-primary-yellow-800));--color-surface-brand: rgba(var(--color-primary-orange-400));--color-surface-danger-default: rgba(var(--color-primary-red-200));--color-surface-danger-vibrant: rgba(var(--color-primary-red-600));--color-surface-default: rgba(var(--color-neutrals-100));--color-surface-default-web: rgba(var(--color-neutrals-100));--color-surface-elevated: rgba(var(--color-white-100));--color-surface-elevated-web: rgba(var(--color-neutrals-200));--color-surface-info-default: rgba(var(--color-primary-blue-200));--color-surface-info-vibrant: rgba(var(--color-primary-blue-400));--color-surface-input: rgba(var(--color-black-4));--color-surface-inverted: rgba(var(--color-black-100));--color-surface-match-mode: rgba(var(--color-white-100));--color-surface-pressed-overlay: rgba(var(--color-black-12));--color-surface-scrimedge: rgb(219, 219, 219);--color-surface-stable-dark: rgba(var(--color-black-100));--color-surface-stable-light: rgba(var(--color-white-100));--color-surface-success-default: rgba(var(--color-primary-green-200));--color-surface-success-vibrant: rgba(var(--color-primary-green-600));--color-surface-sunken: rgba(var(--color-neutrals-200));--color-surface-text: rgba(var(--color-primary-orange-200));--color-surface-transparent: rgba(var(--color-black-6));--color-surface-warning-default: rgba(var(--color-primary-yellow-200));--color-surface-warning-vibrant: rgba(var(--color-primary-yellow-800));--color-text-adaptive: rgba(var(--color-white-100));--color-text-brand: rgba(var(--color-primary-orange-600));--color-text-danger: rgba(var(--color-primary-red-600));--color-text-danger-accessible: rgba(var(--color-primary-red-1000));--color-text-info: rgba(var(--color-primary-blue-600));--color-text-info-accessible: rgba(var(--color-primary-blue-1000));--color-text-link: rgba(var(--color-primary-blue-600));--color-text-link-pressed: rgba(var(--color-primary-blue-800));--color-text-primary: rgba(var(--color-neutrals-700));--color-text-secondary: rgba(var(--color-neutrals-400));--color-text-secondary-web: rgba(var(--color-neutrals-500));--color-text-stable: rgba(var(--color-white-100));--color-text-subtle: rgba(var(--color-neutrals-300));--color-text-success: rgba(var(--color-primary-green-600));--color-text-success-accessible: rgba(var(--color-primary-green-1000));--color-text-warning: rgba(var(--color-primary-yellow-800));--color-text-warning-accessible: rgba(var(--color-primary-yellow-1000))}.dark:root{--color-accent-blue: rgba(var(--color-primary-blue-600));--color-accent-brand: rgba(var(--color-primary-orange-600));--color-accent-coral: rgba(var(--color-secondary-coral));--color-accent-green: rgba(var(--color-primary-green-600));--color-accent-hot-pink: rgba(var(--color-secondary-hot-pink));--color-accent-orange: rgba(var(--color-secondary-clementine));--color-accent-red: rgba(var(--color-primary-red-dark));--color-accent-sky-blue: rgba(var(--color-primary-blue-400));--color-accent-yellow: rgba(var(--color-primary-yellow-600));--color-button-danger: rgba(var(--color-primary-red-600));--color-button-danger-hover: rgba(var(--color-primary-red-400));--color-button-info: rgba(var(--color-primary-blue-800));--color-button-info-hover: rgba(var(--color-primary-blue-600));--color-button-pressed-ghost-overlay: rgba(var(--color-white-12));--color-button-pressed-overlay: rgba(var(--color-black-30));--color-button-primary: rgba(var(--color-primary-orange-600));--color-button-primary-hover: rgba(var(--color-primary-orange-400));--color-button-secondary: rgba(var(--color-white-12));--color-button-secondary-hover: rgba(var(--color-white-18));--color-button-tertiary: rgba(var(--color-white-0));--color-button-tertiary-hover: rgba(var(--color-primary-orange-6));--color-button-tertiary-pressed: rgba(var(--color-primary-orange-18));--color-button-warning: rgba(var(--color-primary-yellow-800));--color-button-warning-hover: rgba(var(--color-primary-yellow-600));--color-icon-adaptive: rgba(var(--color-neutrals-1000));--color-icon-background: rgba(var(--color-white-12));--color-icon-background-brand: rgba(var(--color-primary-orange-600));--color-icon-brand: rgba(var(--color-primary-orange-400));--color-icon-danger: rgba(var(--color-primary-red-400));--color-icon-default: rgba(var(--color-white-100));--color-icon-info: rgba(var(--color-primary-blue-400));--color-icon-secondary: rgba(var(--color-neutrals-400));--color-icon-stable: rgba(var(--color-white-100));--color-icon-subtle: rgba(var(--color-neutrals-500));--color-icon-success: rgba(var(--color-primary-green-400));--color-icon-warning: rgba(var(--color-primary-yellow-600));--color-ios-icon: rgb(231, 231, 231);--color-ios-keyboard-button: rgb(36, 36, 36);--color-ios-primary: rgb(0, 122, 255);--color-ios-surface-dark: rgb(36, 36, 36);--color-ios-surface-default: rgba(32, 32, 32, .92);--color-ios-surface-elevated: rgb(67, 67, 67);--color-ios-text: rgb(255, 255, 255);--color-ios-text-inverted: rgb(255, 255, 255);--color-shadow-default: rgba(var(--color-black-30));--color-shadow-medium: rgba(var(--color-black-50));--color-shadow-strong: rgba(var(--color-black-100));--color-stroke-brand: rgba(var(--color-primary-orange-600));--color-stroke-danger: rgba(var(--color-primary-red-400));--color-stroke-default: rgba(var(--color-neutrals-1000));--color-stroke-default-web: rgba(var(--color-neutrals-700));--color-stroke-focused: rgba(var(--color-white-30));--color-stroke-subtle: rgba(var(--color-white-12));--color-stroke-success: rgba(var(--color-primary-green-400));--color-stroke-warning: rgba(var(--color-primary-yellow-600));--color-surface-brand: rgba(var(--color-primary-orange-600));--color-surface-danger-default: rgba(var(--color-primary-red-1000));--color-surface-danger-vibrant: rgba(var(--color-primary-red-600));--color-surface-default: rgba(var(--color-neutrals-1000));--color-surface-default-web: rgba(var(--color-black-100));--color-surface-elevated: rgba(var(--color-neutrals-900));--color-surface-elevated-web: rgba(var(--color-neutrals-900));--color-surface-info-default: rgba(var(--color-primary-blue-1000));--color-surface-info-vibrant: rgba(var(--color-primary-blue-400));--color-surface-input: rgba(var(--color-white-12));--color-surface-inverted: rgba(var(--color-white-100));--color-surface-match-mode: rgba(var(--color-black-100));--color-surface-pressed-overlay: rgba(var(--color-white-18));--color-surface-scrimedge: rgb(34, 34, 36);--color-surface-stable-dark: rgba(var(--color-black-100));--color-surface-stable-light: rgba(var(--color-white-100));--color-surface-success-default: rgba(var(--color-primary-green-800));--color-surface-success-vibrant: rgba(var(--color-primary-green-400));--color-surface-sunken: rgba(var(--color-black-100));--color-surface-text: rgba(var(--color-primary-orange-800));--color-surface-transparent: rgba(var(--color-white-12));--color-surface-warning-default: rgba(var(--color-primary-yellow-1000));--color-surface-warning-vibrant: rgba(var(--color-primary-yellow-600));--color-text-adaptive: rgba(var(--color-neutrals-1000));--color-text-brand: rgba(var(--color-primary-orange-400));--color-text-danger: rgba(var(--color-primary-red-400));--color-text-danger-accessible: rgba(var(--color-primary-red-200));--color-text-info: rgba(var(--color-primary-blue-400));--color-text-info-accessible: rgba(var(--color-primary-blue-200));--color-text-link: rgba(var(--color-primary-blue-400));--color-text-link-pressed: rgba(var(--color-primary-blue-600));--color-text-primary: rgba(var(--color-white-100));--color-text-secondary: rgba(var(--color-neutrals-400));--color-text-secondary-web: rgba(var(--color-neutrals-300));--color-text-stable: rgba(var(--color-white-100));--color-text-subtle: rgba(var(--color-neutrals-500));--color-text-success: rgba(var(--color-primary-green-400));--color-text-success-accessible: rgba(var(--color-primary-green-200));--color-text-warning: rgba(var(--color-primary-yellow-600));--color-text-warning-accessible: rgba(var(--color-primary-yellow-200))}.modal-font{font-family:Barlow,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Fira Sans,Droid Sans,Helvetica Neue,sans-serif;-webkit-font-smoothing:antialiased;text-rendering:optimizeLegibility;-webkit-text-size-adjust:100%;-moz-text-size-adjust:100%;text-size-adjust:100%;font-feature-settings:"kern"}.l2{color:#8c8c8c;font-size:12px;font-weight:500;line-height:14px;letter-spacing:0;text-align:left}.p{margin:0}.hover\\:bg-button-danger-hover:hover{background-color:var(--color-button-danger-hover)}.hover\\:bg-button-primary-hover:hover{background-color:var(--color-button-primary-hover)}.hover\\:bg-button-secondary-hover:hover{background-color:var(--color-button-secondary-hover)}.hover\\:bg-button-tertiary-hover:hover{background-color:var(--color-button-tertiary-hover)}.hover\\:bg-surface-elevated:hover{background-color:var(--color-surface-elevated)}.focus\\:outline-none:focus{outline:2px solid transparent;outline-offset:2px}.focus\\:ring-2:focus{--tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow, 0 0 #0000)}.focus\\:ring-neutral-200:focus{--tw-ring-opacity: 1;--tw-ring-color: rgb(229 229 229 / var(--tw-ring-opacity))}.focus-visible\\:outline-none:focus-visible{outline:2px solid transparent;outline-offset:2px}.focus-visible\\:ring-2:focus-visible{--tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);--tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);box-shadow:var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow, 0 0 #0000)}.focus-visible\\:ring-offset-2:focus-visible{--tw-ring-offset-width: 2px}.disabled\\:pointer-events-none:disabled{pointer-events:none}.disabled\\:opacity-50:disabled{opacity:.5}.dark\\:shadow-none:is(.dark *){--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;box-shadow:var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow)}.dark\\:focus\\:ring-neutral-700:focus:is(.dark *){--tw-ring-opacity: 1;--tw-ring-color: rgb(64 64 64 / var(--tw-ring-opacity))}
`, bl = () => {
  const l = "starknetkit-modal-container", e = document.getElementById(l);
  if (e) {
    if (e.shadowRoot)
      return e.shadowRoot;
    e.remove();
  }
  const o = document.createElement("div");
  o.id = l, document.body.appendChild(o);
  const r = o.attachShadow({ mode: "open" }), a = document.createElement("style");
  return a.textContent = $r.replace(
    /\.dark:root/g,
    ":host-context(.dark), .dark"
  ), r.appendChild(a), r;
};
function B() {
}
function xt(l, e) {
  for (const o in e)
    l[o] = e[o];
  return (
    /** @type {T & S} */
    l
  );
}
function cr(l) {
  return l();
}
function It() {
  return /* @__PURE__ */ Object.create(null);
}
function ke(l) {
  l.forEach(cr);
}
function me(l) {
  return typeof l == "function";
}
function U(l, e) {
  return l != l ? e == e : l !== e || l && typeof l == "object" || typeof l == "function";
}
let _t;
function Er(l, e) {
  return l === e ? !0 : (_t || (_t = document.createElement("a")), _t.href = e, l === _t.href);
}
function Nr(l) {
  return Object.keys(l).length === 0;
}
function be(l, e, o, r) {
  if (l) {
    const a = fr(l, e, o, r);
    return l[0](a);
  }
}
function fr(l, e, o, r) {
  return l[1] && r ? xt(o.ctx.slice(), l[1](r(e))) : o.ctx;
}
function we(l, e, o, r) {
  if (l[2] && r) {
    const a = l[2](r(o));
    if (e.dirty === void 0)
      return a;
    if (typeof a == "object") {
      const n = [], s = Math.max(e.dirty.length, a.length);
      for (let f = 0; f < s; f += 1)
        n[f] = e.dirty[f] | a[f];
      return n;
    }
    return e.dirty | a;
  }
  return e.dirty;
}
function Ce(l, e, o, r, a, n) {
  if (a) {
    const s = fr(e, o, r, n);
    l.p(s, a);
  }
}
function ve(l) {
  if (l.ctx.length > 32) {
    const e = [], o = l.ctx.length / 32;
    for (let r = 0; r < o; r++)
      e[r] = -1;
    return e;
  }
  return -1;
}
function i(l, e) {
  l.appendChild(e);
}
function Zr(l, e, o) {
  const r = Ar(l);
  if (!r.getElementById(e)) {
    const a = x("style");
    a.id = e, a.textContent = o, zr(r, a);
  }
}
function Ar(l) {
  if (!l)
    return document;
  const e = l.getRootNode ? l.getRootNode() : l.ownerDocument;
  return e && /** @type {ShadowRoot} */
  e.host ? (
    /** @type {ShadowRoot} */
    e
  ) : l.ownerDocument;
}
function zr(l, e) {
  return i(
    /** @type {Document} */
    l.head || l,
    e
  ), e.sheet;
}
function $(l, e, o) {
  l.insertBefore(e, o || null);
}
function M(l) {
  l.parentNode && l.parentNode.removeChild(l);
}
function Or(l, e) {
  for (let o = 0; o < l.length; o += 1)
    l[o] && l[o].d(e);
}
function x(l) {
  return document.createElement(l);
}
function c(l) {
  return document.createElementNS("http://www.w3.org/2000/svg", l);
}
function X(l) {
  return document.createTextNode(l);
}
function K() {
  return X(" ");
}
function ye() {
  return X("");
}
function $e(l, e, o, r) {
  return l.addEventListener(e, o, r), () => l.removeEventListener(e, o, r);
}
function t(l, e, o) {
  o == null ? l.removeAttribute(e) : l.getAttribute(e) !== o && l.setAttribute(e, o);
}
const Ur = ["width", "height"];
function Gr(l, e) {
  const o = Object.getOwnPropertyDescriptors(l.__proto__);
  for (const r in e)
    e[r] == null ? l.removeAttribute(r) : r === "style" ? l.style.cssText = e[r] : r === "__value" ? l.value = l[r] = e[r] : o[r] && o[r].set && Ur.indexOf(r) === -1 ? l[r] = e[r] : t(l, r, e[r]);
}
function Ir(l, e) {
  Object.keys(e).forEach((o) => {
    Wr(l, o, e[o]);
  });
}
function Wr(l, e, o) {
  const r = e.toLowerCase();
  r in l ? l[r] = typeof l[r] == "boolean" && o === "" ? !0 : o : e in l ? l[e] = typeof l[e] == "boolean" && o === "" ? !0 : o : t(l, e, o);
}
function Ee(l) {
  return /-/.test(l) ? Ir : Gr;
}
function jr(l) {
  return Array.from(l.childNodes);
}
function ge(l, e) {
  e = "" + e, l.data !== e && (l.data = /** @type {string} */
  e);
}
function xe(l, e, o, r) {
  o == null ? l.style.removeProperty(e) : l.style.setProperty(e, o, r ? "important" : "");
}
function Wt(l, e) {
  return new l(e);
}
let je;
function We(l) {
  je = l;
}
function Dr() {
  if (!je)
    throw new Error("Function called outside component initialization");
  return je;
}
function Vr(l) {
  Dr().$$.on_mount.push(l);
}
const Me = [], Nt = [];
let Se = [];
const jt = [], Rr = /* @__PURE__ */ Promise.resolve();
let Zt = !1;
function Tr() {
  Zt || (Zt = !0, Rr.then(ur));
}
function At(l) {
  Se.push(l);
}
const Mt = /* @__PURE__ */ new Set();
let Be = 0;
function ur() {
  if (Be !== 0)
    return;
  const l = je;
  do {
    try {
      for (; Be < Me.length; ) {
        const e = Me[Be];
        Be++, We(e), Hr(e.$$);
      }
    } catch (e) {
      throw Me.length = 0, Be = 0, e;
    }
    for (We(null), Me.length = 0, Be = 0; Nt.length; )
      Nt.pop()();
    for (let e = 0; e < Se.length; e += 1) {
      const o = Se[e];
      Mt.has(o) || (Mt.add(o), o());
    }
    Se.length = 0;
  } while (Me.length);
  for (; jt.length; )
    jt.pop()();
  Zt = !1, Mt.clear(), We(l);
}
function Hr(l) {
  if (l.fragment !== null) {
    l.update(), ke(l.before_update);
    const e = l.dirty;
    l.dirty = [-1], l.fragment && l.fragment.p(l.ctx, e), l.after_update.forEach(At);
  }
}
function Kr(l) {
  const e = [], o = [];
  Se.forEach((r) => l.indexOf(r) === -1 ? e.push(r) : o.push(r)), o.forEach((r) => r()), Se = e;
}
const yt = /* @__PURE__ */ new Set();
let _e;
function ce() {
  _e = {
    r: 0,
    c: [],
    p: _e
    // parent group
  };
}
function fe() {
  _e.r || ke(_e.c), _e = _e.p;
}
function w(l, e) {
  l && l.i && (yt.delete(l), l.i(e));
}
function C(l, e, o, r) {
  if (l && l.o) {
    if (yt.has(l))
      return;
    yt.add(l), _e.c.push(() => {
      yt.delete(l), r && (o && l.d(1), r());
    }), l.o(e);
  } else
    r && r();
}
function Dt(l) {
  return l?.length !== void 0 ? l : Array.from(l);
}
function zt(l, e) {
  const o = {}, r = {}, a = { $$scope: 1 };
  let n = l.length;
  for (; n--; ) {
    const s = l[n], f = e[n];
    if (f) {
      for (const d in s)
        d in f || (r[d] = 1);
      for (const d in f)
        a[d] || (o[d] = f[d], a[d] = 1);
      l[n] = f;
    } else
      for (const d in s)
        a[d] = 1;
  }
  for (const s in r)
    s in o || (o[s] = void 0);
  return o;
}
function A(l) {
  l && l.c();
}
function N(l, e, o) {
  const { fragment: r, after_update: a } = l.$$;
  r && r.m(e, o), At(() => {
    const n = l.$$.on_mount.map(cr).filter(me);
    l.$$.on_destroy ? l.$$.on_destroy.push(...n) : ke(n), l.$$.on_mount = [];
  }), a.forEach(At);
}
function Z(l, e) {
  const o = l.$$;
  o.fragment !== null && (Kr(o.after_update), ke(o.on_destroy), o.fragment && o.fragment.d(e), o.on_destroy = o.fragment = null, o.ctx = []);
}
function Pr(l, e) {
  l.$$.dirty[0] === -1 && (Me.push(l), Tr(), l.$$.dirty.fill(0)), l.$$.dirty[e / 31 | 0] |= 1 << e % 31;
}
function W(l, e, o, r, a, n, s = null, f = [-1]) {
  const d = je;
  We(l);
  const p = l.$$ = {
    fragment: null,
    ctx: [],
    // state
    props: n,
    update: B,
    not_equal: a,
    bound: It(),
    // lifecycle
    on_mount: [],
    on_destroy: [],
    on_disconnect: [],
    before_update: [],
    after_update: [],
    context: new Map(e.context || (d ? d.$$.context : [])),
    // everything else
    callbacks: It(),
    dirty: f,
    skip_bound: !1,
    root: e.target || d.$$.root
  };
  s && s(p.root);
  let h = !1;
  if (p.ctx = o ? o(l, e.props || {}, (u, g, ...m) => {
    const b = m.length ? m[0] : g;
    return p.ctx && a(p.ctx[u], p.ctx[u] = b) && (!p.skip_bound && p.bound[u] && p.bound[u](b), h && Pr(l, u)), g;
  }) : [], p.update(), h = !0, ke(p.before_update), p.fragment = r ? r(p.ctx) : !1, e.target) {
    if (e.hydrate) {
      const u = jr(e.target);
      p.fragment && p.fragment.l(u), u.forEach(M);
    } else
      p.fragment && p.fragment.c();
    e.intro && w(l.$$.fragment), N(l, e.target, e.anchor), ur();
  }
  We(d);
}
class j {
  constructor() {
    /**
     * ### PRIVATE API
     *
     * Do not use, may change at any time
     *
     * @type {any}
     */
    Bt(this, "$$");
    /**
     * ### PRIVATE API
     *
     * Do not use, may change at any time
     *
     * @type {any}
     */
    Bt(this, "$$set");
  }
  /** @returns {void} */
  $destroy() {
    Z(this, 1), this.$destroy = B;
  }
  /**
   * @template {Extract<keyof Events, string>} K
   * @param {K} type
   * @param {((e: Events[K]) => void) | null | undefined} callback
   * @returns {() => void}
   */
  $on(e, o) {
    if (!me(o))
      return B;
    const r = this.$$.callbacks[e] || (this.$$.callbacks[e] = []);
    return r.push(o), () => {
      const a = r.indexOf(o);
      a !== -1 && r.splice(a, 1);
    };
  }
  /**
   * @param {Partial<Props>} props
   * @returns {void}
   */
  $set(e) {
    this.$$set && !Nr(e) && (this.$$.skip_bound = !0, this.$$set(e), this.$$.skip_bound = !1);
  }
}
const qr = "4";
typeof window < "u" && (window.__svelte || (window.__svelte = { v: /* @__PURE__ */ new Set() })).v.add(qr);
function St(l) {
  let e, o, r, a, n;
  const s = (
    /*#slots*/
    l[7].default
  ), f = be(
    s,
    l,
    /*$$scope*/
    l[6],
    null
  );
  let d = [
    { href: (
      /*href*/
      l[0]
    ) },
    {
      class: o = `
    p-2 cursor-pointer text-primary rounded-full bg-surface-elevated-web
		focus:outline-none focus:ring-2 focus:ring-neutral-200 dark:focus:ring-neutral-700
		transition-colors ${/*className*/
      l[2]}
  `
    },
    { rel: "noreferrer noopener" },
    { target: "_blank" },
    { role: "button" },
    { tabindex: "0" },
    { "aria-label": (
      /*ariaLabel*/
      l[1]
    ) }
  ], p = {};
  for (let h = 0; h < d.length; h += 1)
    p = xt(p, d[h]);
  return {
    c() {
      e = x(
        /*as*/
        l[3]
      ), f && f.c(), Ee(
        /*as*/
        l[3]
      )(e, p);
    },
    m(h, u) {
      $(h, e, u), f && f.m(e, null), r = !0, a || (n = [
        $e(e, "click", function() {
          me(
            /*handleClick*/
            l[4]
          ) && l[4].apply(this, arguments);
        }),
        $e(e, "keyup", function() {
          me(
            /*handleKeyup*/
            l[5]
          ) && l[5].apply(this, arguments);
        })
      ], a = !0);
    },
    p(h, u) {
      l = h, f && f.p && (!r || u & /*$$scope*/
      64) && Ce(
        f,
        s,
        l,
        /*$$scope*/
        l[6],
        r ? we(
          s,
          /*$$scope*/
          l[6],
          u,
          null
        ) : ve(
          /*$$scope*/
          l[6]
        ),
        null
      ), Ee(
        /*as*/
        l[3]
      )(e, p = zt(d, [
        (!r || u & /*href*/
        1) && { href: (
          /*href*/
          l[0]
        ) },
        (!r || u & /*className*/
        4 && o !== (o = `
    p-2 cursor-pointer text-primary rounded-full bg-surface-elevated-web
		focus:outline-none focus:ring-2 focus:ring-neutral-200 dark:focus:ring-neutral-700
		transition-colors ${/*className*/
        l[2]}
  `)) && { class: o },
        { rel: "noreferrer noopener" },
        { target: "_blank" },
        { role: "button" },
        { tabindex: "0" },
        (!r || u & /*ariaLabel*/
        2) && { "aria-label": (
          /*ariaLabel*/
          l[1]
        ) }
      ]));
    },
    i(h) {
      r || (w(f, h), r = !0);
    },
    o(h) {
      C(f, h), r = !1;
    },
    d(h) {
      h && M(e), f && f.d(h), a = !1, ke(n);
    }
  };
}
function Xr(l) {
  let e = (
    /*as*/
    l[3]
  ), o, r, a = (
    /*as*/
    l[3] && St(l)
  );
  return {
    c() {
      a && a.c(), o = ye();
    },
    m(n, s) {
      a && a.m(n, s), $(n, o, s), r = !0;
    },
    p(n, [s]) {
      /*as*/
      n[3] ? e ? U(
        e,
        /*as*/
        n[3]
      ) ? (a.d(1), a = St(n), e = /*as*/
      n[3], a.c(), a.m(o.parentNode, o)) : a.p(n, s) : (a = St(n), e = /*as*/
      n[3], a.c(), a.m(o.parentNode, o)) : e && (a.d(1), a = null, e = /*as*/
      n[3]);
    },
    i(n) {
      r || (w(a, n), r = !0);
    },
    o(n) {
      C(a, n), r = !1;
    },
    d(n) {
      n && M(o), a && a.d(n);
    }
  };
}
function Qr(l, e, o) {
  let { $$slots: r = {}, $$scope: a } = e, { href: n = "" } = e, { ariaLabel: s } = e, { className: f } = e, { as: d = "button" } = e, { handleClick: p = (u) => console.log("click", u) } = e, { handleKeyup: h = (u) => console.log("keyup", u) } = e;
  return l.$$set = (u) => {
    "href" in u && o(0, n = u.href), "ariaLabel" in u && o(1, s = u.ariaLabel), "className" in u && o(2, f = u.className), "as" in u && o(3, d = u.as), "handleClick" in u && o(4, p = u.handleClick), "handleKeyup" in u && o(5, h = u.handleKeyup), "$$scope" in u && o(6, a = u.$$scope);
  }, [n, s, f, d, p, h, a, r];
}
class dr extends j {
  constructor(e) {
    super(), W(this, e, Qr, Xr, U, {
      href: 0,
      ariaLabel: 1,
      className: 2,
      as: 3,
      handleClick: 4,
      handleKeyup: 5
    });
  }
}
function Yr(l) {
  let e, o;
  return {
    c() {
      e = c("svg"), o = c("path"), t(o, "d", "M9.77275 3.02275C9.99242 2.80308 9.99242 2.44692 9.77275 2.22725C9.55308 2.00758 9.19692 2.00758 8.97725 2.22725L6 5.20451L3.02275 2.22725C2.80308 2.00758 2.44692 2.00758 2.22725 2.22725C2.00758 2.44692 2.00758 2.80308 2.22725 3.02275L5.20451 6L2.22725 8.97725C2.00758 9.19692 2.00758 9.55308 2.22725 9.77275C2.44692 9.99242 2.80308 9.99242 3.02275 9.77275L6 6.79549L8.97725 9.77275C9.19692 9.99242 9.55308 9.99242 9.77275 9.77275C9.99242 9.55308 9.99242 9.19692 9.77275 8.97725L6.79549 6L9.77275 3.02275Z"), t(o, "fill", "currentColor"), t(e, "width", "12"), t(e, "height", "12"), t(e, "viewBox", "0 0 12 12"), t(e, "fill", "none"), t(e, "xmlns", "http://www.w3.org/2000/svg");
    },
    m(r, a) {
      $(r, e, a), i(e, o);
    },
    p: B,
    i: B,
    o: B,
    d(r) {
      r && M(e);
    }
  };
}
class Jr extends j {
  constructor(e) {
    super(), W(this, e, null, Yr, U, {});
  }
}
function e1(l) {
  let e, o;
  return {
    c() {
      e = c("svg"), o = c("path"), t(o, "fill-rule", "evenodd"), t(o, "clip-rule", "evenodd"), t(o, "d", "M8.55383 1.24376C8.83406 1.54808 8.81134 2.01905 8.50309 2.2957L4.37567 6L8.50309 9.7043C8.81134 9.98095 8.83406 10.4519 8.55383 10.7562C8.2736 11.0606 7.79654 11.083 7.48828 10.8063L2.74691 6.55102C2.58965 6.40989 2.5 6.20981 2.5 6C2.5 5.79019 2.58965 5.59012 2.74691 5.44898L7.48828 1.19367C7.79654 0.917013 8.2736 0.93944 8.55383 1.24376Z"), t(o, "fill", "currentColor"), t(e, "xmlns", "http://www.w3.org/2000/svg"), t(e, "width", "12"), t(e, "height", "12"), t(e, "viewBox", "0 0 12 12"), t(e, "fill", "none");
    },
    m(r, a) {
      $(r, e, a), i(e, o);
    },
    p: B,
    i: B,
    o: B,
    d(r) {
      r && M(e);
    }
  };
}
class t1 extends j {
  constructor(e) {
    super(), W(this, e, null, e1, U, {});
  }
}
function Vt(l) {
  let e, o;
  return e = new dr({
    props: {
      handleClick: (
        /*handleBack*/
        l[4]
      ),
      handleKeyup: (
        /*func*/
        l[6]
      ),
      ariaLabel: "Go back",
      className: "absolute top-0 left-0",
      $$slots: { default: [r1] },
      $$scope: { ctx: l }
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, a) {
      const n = {};
      a & /*handleBack*/
      16 && (n.handleClick = /*handleBack*/
      r[4]), a & /*handleBack*/
      16 && (n.handleKeyup = /*func*/
      r[6]), a & /*$$scope*/
      256 && (n.$$scope = { dirty: a, ctx: r }), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function r1(l) {
  let e, o;
  return e = new t1({}), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function Rt(l) {
  let e, o;
  return {
    c() {
      e = x("h1"), o = X(
        /*title*/
        l[0]
      ), t(e, "class", "text-[24px] text-primary font-semibold max-w-[240px] overflow-hidden whitespace-nowrap text-ellipsis");
    },
    m(r, a) {
      $(r, e, a), i(e, o);
    },
    p(r, a) {
      a & /*title*/
      1 && ge(
        o,
        /*title*/
        r[0]
      );
    },
    d(r) {
      r && M(e);
    }
  };
}
function Tt(l) {
  let e, o;
  return e = new dr({
    props: {
      handleClick: (
        /*handleClose*/
        l[5]
      ),
      handleKeyup: (
        /*func_1*/
        l[7]
      ),
      ariaLabel: "Close",
      className: "absolute top-0 right-0",
      $$slots: { default: [o1] },
      $$scope: { ctx: l }
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, a) {
      const n = {};
      a & /*handleClose*/
      32 && (n.handleClick = /*handleClose*/
      r[5]), a & /*handleClose*/
      32 && (n.handleKeyup = /*func_1*/
      r[7]), a & /*$$scope*/
      256 && (n.$$scope = { dirty: a, ctx: r }), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function o1(l) {
  let e, o;
  return e = new Jr({}), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function l1(l) {
  let e, o, r, a, n, s, f, d, p = (
    /*showBackButton*/
    l[2] && Vt(l)
  ), h = !!/*title*/
  l[0] && Rt(l), u = (
    /*showCloseButton*/
    l[3] && Tt(l)
  );
  return {
    c() {
      e = x("header"), p && p.c(), o = K(), r = x("hgroup"), a = x("h2"), n = X(
        /*subtitle*/
        l[1]
      ), s = K(), h && h.c(), f = K(), u && u.c(), t(a, "class", "text-p3 text-secondary font-semibold"), t(e, "class", "flex items-center justify-center flex-col relative");
    },
    m(g, m) {
      $(g, e, m), p && p.m(e, null), i(e, o), i(e, r), i(r, a), i(a, n), i(r, s), h && h.m(r, null), i(e, f), u && u.m(e, null), d = !0;
    },
    p(g, [m]) {
      /*showBackButton*/
      g[2] ? p ? (p.p(g, m), m & /*showBackButton*/
      4 && w(p, 1)) : (p = Vt(g), p.c(), w(p, 1), p.m(e, o)) : p && (ce(), C(p, 1, 1, () => {
        p = null;
      }), fe()), (!d || m & /*subtitle*/
      2) && ge(
        n,
        /*subtitle*/
        g[1]
      ), /*title*/
      g[0] ? h ? h.p(g, m) : (h = Rt(g), h.c(), h.m(r, null)) : h && (h.d(1), h = null), /*showCloseButton*/
      g[3] ? u ? (u.p(g, m), m & /*showCloseButton*/
      8 && w(u, 1)) : (u = Tt(g), u.c(), w(u, 1), u.m(e, null)) : u && (ce(), C(u, 1, 1, () => {
        u = null;
      }), fe());
    },
    i(g) {
      d || (w(p), w(u), d = !0);
    },
    o(g) {
      C(p), C(u), d = !1;
    },
    d(g) {
      g && M(e), p && p.d(), h && h.d(), u && u.d();
    }
  };
}
function a1(l, e, o) {
  let { title: r } = e, { subtitle: a = "Connect to" } = e, { showBackButton: n = !0 } = e, { showCloseButton: s = !0 } = e, { handleBack: f = () => {
  } } = e, { handleClose: d = () => {
  } } = e;
  const p = (u) => {
    u.key === "Enter" && f();
  }, h = (u) => {
    u.key === "Enter" && d();
  };
  return l.$$set = (u) => {
    "title" in u && o(0, r = u.title), "subtitle" in u && o(1, a = u.subtitle), "showBackButton" in u && o(2, n = u.showBackButton), "showCloseButton" in u && o(3, s = u.showCloseButton), "handleBack" in u && o(4, f = u.handleBack), "handleClose" in u && o(5, d = u.handleClose);
  }, [
    r,
    a,
    n,
    s,
    f,
    d,
    p,
    h
  ];
}
class n1 extends j {
  constructor(e) {
    super(), W(this, e, a1, l1, U, {
      title: 0,
      subtitle: 1,
      showBackButton: 2,
      showCloseButton: 3,
      handleBack: 4,
      handleClose: 5
    });
  }
}
function $t(l) {
  let e, o, r, a, n;
  const s = (
    /*#slots*/
    l[7].default
  ), f = be(
    s,
    l,
    /*$$scope*/
    l[6],
    null
  );
  let d = [
    { href: (
      /*href*/
      l[0]
    ) },
    { rel: "noreferrer noopener" },
    { target: "_blank" },
    { role: "button" },
    { "aria-label": (
      /*ariaLabel*/
      l[1]
    ) },
    { tabindex: "0" },
    {
      class: o = `
		flex justify-center items-center w-full px-3 h-[60px] rounded-[12px] cursor-pointer shadow-list-item
		dark:shadow-none bg-surface-elevated focus:outline-none focus:ring-2 text-primary
		focus:ring-neutral-200 dark:focus:ring-neutral-700 transition-colors ${/*className*/
      l[2]}
	`
    }
  ], p = {};
  for (let h = 0; h < d.length; h += 1)
    p = xt(p, d[h]);
  return {
    c() {
      e = x(
        /*as*/
        l[3]
      ), f && f.c(), Ee(
        /*as*/
        l[3]
      )(e, p);
    },
    m(h, u) {
      $(h, e, u), f && f.m(e, null), r = !0, a || (n = [
        $e(e, "click", function() {
          me(
            /*handleClick*/
            l[4]
          ) && l[4].apply(this, arguments);
        }),
        $e(e, "keyup", function() {
          me(
            /*handleKeyup*/
            l[5]
          ) && l[5].apply(this, arguments);
        })
      ], a = !0);
    },
    p(h, u) {
      l = h, f && f.p && (!r || u & /*$$scope*/
      64) && Ce(
        f,
        s,
        l,
        /*$$scope*/
        l[6],
        r ? we(
          s,
          /*$$scope*/
          l[6],
          u,
          null
        ) : ve(
          /*$$scope*/
          l[6]
        ),
        null
      ), Ee(
        /*as*/
        l[3]
      )(e, p = zt(d, [
        (!r || u & /*href*/
        1) && { href: (
          /*href*/
          l[0]
        ) },
        { rel: "noreferrer noopener" },
        { target: "_blank" },
        { role: "button" },
        (!r || u & /*ariaLabel*/
        2) && { "aria-label": (
          /*ariaLabel*/
          l[1]
        ) },
        { tabindex: "0" },
        (!r || u & /*className*/
        4 && o !== (o = `
		flex justify-center items-center w-full px-3 h-[60px] rounded-[12px] cursor-pointer shadow-list-item
		dark:shadow-none bg-surface-elevated focus:outline-none focus:ring-2 text-primary
		focus:ring-neutral-200 dark:focus:ring-neutral-700 transition-colors ${/*className*/
        l[2]}
	`)) && { class: o }
      ]));
    },
    i(h) {
      r || (w(f, h), r = !0);
    },
    o(h) {
      C(f, h), r = !1;
    },
    d(h) {
      h && M(e), f && f.d(h), a = !1, ke(n);
    }
  };
}
function i1(l) {
  let e = (
    /*as*/
    l[3]
  ), o, r, a = (
    /*as*/
    l[3] && $t(l)
  );
  return {
    c() {
      a && a.c(), o = ye();
    },
    m(n, s) {
      a && a.m(n, s), $(n, o, s), r = !0;
    },
    p(n, [s]) {
      /*as*/
      n[3] ? e ? U(
        e,
        /*as*/
        n[3]
      ) ? (a.d(1), a = $t(n), e = /*as*/
      n[3], a.c(), a.m(o.parentNode, o)) : a.p(n, s) : (a = $t(n), e = /*as*/
      n[3], a.c(), a.m(o.parentNode, o)) : e && (a.d(1), a = null, e = /*as*/
      n[3]);
    },
    i(n) {
      r || (w(a, n), r = !0);
    },
    o(n) {
      C(a, n), r = !1;
    },
    d(n) {
      n && M(o), a && a.d(n);
    }
  };
}
function s1(l, e, o) {
  let { $$slots: r = {}, $$scope: a } = e, { href: n = "" } = e, { ariaLabel: s = "" } = e, { className: f = "" } = e, { as: d = "button" } = e, { handleClick: p = (u) => console.log("click", u) } = e, { handleKeyup: h = (u) => console.log("keyup", u) } = e;
  return l.$$set = (u) => {
    "href" in u && o(0, n = u.href), "ariaLabel" in u && o(1, s = u.ariaLabel), "className" in u && o(2, f = u.className), "as" in u && o(3, d = u.as), "handleClick" in u && o(4, p = u.handleClick), "handleKeyup" in u && o(5, h = u.handleKeyup), "$$scope" in u && o(6, a = u.$$scope);
  }, [n, s, f, d, p, h, a, r];
}
class Ot extends j {
  constructor(e) {
    super(), W(this, e, s1, i1, U, {
      href: 0,
      ariaLabel: 1,
      className: 2,
      as: 3,
      handleClick: 4,
      handleKeyup: 5
    });
  }
}
function c1(l) {
  let e, o;
  return {
    c() {
      e = x("img"), Er(e.src, o = /*renderIcon*/
      l[0]) || t(e, "src", o), t(e, "class", "w-8 h-8 rounded"), t(e, "alt", "");
    },
    m(r, a) {
      $(r, e, a);
    },
    p: B,
    d(r) {
      r && M(e);
    }
  };
}
function f1(l) {
  let e;
  return {
    c() {
      e = x("div"), t(e, "class", "flex flex-shrink items-center max-w-[32px]"), xe(e, "position", "relative");
    },
    m(o, r) {
      $(o, e, r), e.innerHTML = /*renderIcon*/
      l[0];
    },
    p: B,
    d(o) {
      o && M(e);
    }
  };
}
function u1(l) {
  let e;
  function o(n, s) {
    return (
      /*renderIcon*/
      n[0].trimStart()?.startsWith("<svg") ? f1 : c1
    );
  }
  let a = o(l)(l);
  return {
    c() {
      a.c(), e = ye();
    },
    m(n, s) {
      a.m(n, s), $(n, e, s);
    },
    p(n, [s]) {
      a.p(n, s);
    },
    i: B,
    o: B,
    d(n) {
      n && M(e), a.d(n);
    }
  };
}
function d1(l, e, o) {
  let { theme: r } = e, { icon: a } = e;
  const n = typeof a == "string" ? a : r === "dark" ? a.dark : a.light;
  return l.$$set = (s) => {
    "theme" in s && o(1, r = s.theme), "icon" in s && o(2, a = s.icon);
  }, [n, r, a];
}
class pr extends j {
  constructor(e) {
    super(), W(this, e, d1, u1, U, { theme: 1, icon: 2 });
  }
}
function Ht(l) {
  let e, o = (
    /*wallet*/
    l[0].subtitle + ""
  ), r;
  return {
    c() {
      e = x("p"), r = X(o), t(e, "class", "text-l2 text-subtle"), xe(e, "text-align", "center");
    },
    m(a, n) {
      $(a, e, n), i(e, r);
    },
    p(a, n) {
      n & /*wallet*/
      1 && o !== (o = /*wallet*/
      a[0].subtitle + "") && ge(r, o);
    },
    d(a) {
      a && M(e);
    }
  };
}
function p1(l) {
  let e, o, r, a, n = (
    /*wallet*/
    (l[0].title ?? /*wallet*/
    l[0].name) + ""
  ), s, f, d, p, h, u = (
    /*wallet*/
    l[0].subtitle && Ht(l)
  );
  return p = new pr({
    props: {
      icon: (
        /*wallet*/
        l[0].icon
      ),
      theme: (
        /*theme*/
        l[1]
      )
    }
  }), {
    c() {
      e = x("div"), o = x("div"), r = x("div"), a = x("p"), s = X(n), f = K(), u && u.c(), d = K(), A(p.$$.fragment), t(a, "class", "font-semibold text-[15px] text-primary"), t(r, "class", "ml-[-32px]"), t(o, "class", "flex flex-grow flex-col justify-center items-center"), t(e, "class", "w-full flex flex-row-reverse justify-between");
    },
    m(g, m) {
      $(g, e, m), i(e, o), i(o, r), i(r, a), i(a, s), i(r, f), u && u.m(r, null), i(e, d), N(p, e, null), h = !0;
    },
    p(g, m) {
      (!h || m & /*wallet*/
      1) && n !== (n = /*wallet*/
      (g[0].title ?? /*wallet*/
      g[0].name) + "") && ge(s, n), /*wallet*/
      g[0].subtitle ? u ? u.p(g, m) : (u = Ht(g), u.c(), u.m(r, null)) : u && (u.d(1), u = null);
      const b = {};
      m & /*wallet*/
      1 && (b.icon = /*wallet*/
      g[0].icon), m & /*theme*/
      2 && (b.theme = /*theme*/
      g[1]), p.$set(b);
    },
    i(g) {
      h || (w(p.$$.fragment, g), h = !0);
    },
    o(g) {
      C(p.$$.fragment, g), h = !1;
    },
    d(g) {
      g && M(e), u && u.d(), Z(p);
    }
  };
}
function h1(l) {
  let e, o;
  return e = new Ot({
    props: {
      handleClick: (
        /*func*/
        l[3]
      ),
      handleKeyup: (
        /*func_1*/
        l[4]
      ),
      $$slots: { default: [p1] },
      $$scope: { ctx: l }
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, [a]) {
      const n = {};
      a & /*callback, wallet*/
      5 && (n.handleClick = /*func*/
      r[3]), a & /*callback, wallet*/
      5 && (n.handleKeyup = /*func_1*/
      r[4]), a & /*$$scope, wallet, theme*/
      35 && (n.$$scope = { dirty: a, ctx: r }), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function g1(l, e, o) {
  let { wallet: r } = e, { theme: a } = e, { callback: n = async () => {
  } } = e;
  const s = () => n(r), f = (d) => {
    d.key === "Enter" && n(r);
  };
  return l.$$set = (d) => {
    "wallet" in d && o(0, r = d.wallet), "theme" in d && o(1, a = d.theme), "callback" in d && o(2, n = d.callback);
  }, [r, a, n, s, f];
}
class m1 extends j {
  constructor(e) {
    super(), W(this, e, g1, h1, U, { wallet: 0, theme: 1, callback: 2 });
  }
}
function Kt(l, e, o) {
  const r = l.slice();
  return r[3] = e[o], r;
}
function Pt(l) {
  let e, o, r;
  return o = new m1({
    props: {
      wallet: (
        /*wallet*/
        l[3]
      ),
      theme: (
        /*theme*/
        l[0]
      ),
      callback: (
        /*callback*/
        l[1]
      )
    }
  }), {
    c() {
      e = x("li"), A(o.$$.fragment);
    },
    m(a, n) {
      $(a, e, n), N(o, e, null), r = !0;
    },
    p(a, n) {
      const s = {};
      n & /*walletList*/
      4 && (s.wallet = /*wallet*/
      a[3]), n & /*theme*/
      1 && (s.theme = /*theme*/
      a[0]), n & /*callback*/
      2 && (s.callback = /*callback*/
      a[1]), o.$set(s);
    },
    i(a) {
      r || (w(o.$$.fragment, a), r = !0);
    },
    o(a) {
      C(o.$$.fragment, a), r = !1;
    },
    d(a) {
      a && M(e), Z(o);
    }
  };
}
function b1(l) {
  let e, o, r, a = Dt(
    /*walletList*/
    l[2]
  ), n = [];
  for (let f = 0; f < a.length; f += 1)
    n[f] = Pt(Kt(l, a, f));
  const s = (f) => C(n[f], 1, 1, () => {
    n[f] = null;
  });
  return {
    c() {
      e = x("section"), o = x("ul");
      for (let f = 0; f < n.length; f += 1)
        n[f].c();
      t(o, "class", "flex flex-col gap-3");
    },
    m(f, d) {
      $(f, e, d), i(e, o);
      for (let p = 0; p < n.length; p += 1)
        n[p] && n[p].m(o, null);
      r = !0;
    },
    p(f, [d]) {
      if (d & /*walletList, theme, callback*/
      7) {
        a = Dt(
          /*walletList*/
          f[2]
        );
        let p;
        for (p = 0; p < a.length; p += 1) {
          const h = Kt(f, a, p);
          n[p] ? (n[p].p(h, d), w(n[p], 1)) : (n[p] = Pt(h), n[p].c(), w(n[p], 1), n[p].m(o, null));
        }
        for (ce(), p = a.length; p < n.length; p += 1)
          s(p);
        fe();
      }
    },
    i(f) {
      if (!r) {
        for (let d = 0; d < a.length; d += 1)
          w(n[d]);
        r = !0;
      }
    },
    o(f) {
      n = n.filter(Boolean);
      for (let d = 0; d < n.length; d += 1)
        C(n[d]);
      r = !1;
    },
    d(f) {
      f && M(e), Or(n, f);
    }
  };
}
function w1(l, e, o) {
  let { theme: r } = e, { callback: a = async () => {
  } } = e, { walletList: n } = e;
  return l.$$set = (s) => {
    "theme" in s && o(0, r = s.theme), "callback" in s && o(1, a = s.callback), "walletList" in s && o(2, n = s.walletList);
  }, [r, a, n];
}
class C1 extends j {
  constructor(e) {
    super(), W(this, e, w1, b1, U, { theme: 0, callback: 1, walletList: 2 });
  }
}
function v1(l) {
  let e, o;
  return {
    c() {
      e = x("hr"), t(e, "class", o = `border-stroke-default w-full ${/*className*/
      l[0]}`);
    },
    m(r, a) {
      $(r, e, a);
    },
    p(r, [a]) {
      a & /*className*/
      1 && o !== (o = `border-stroke-default w-full ${/*className*/
      r[0]}`) && t(e, "class", o);
    },
    i: B,
    o: B,
    d(r) {
      r && M(e);
    }
  };
}
function k1(l, e, o) {
  let { className: r = "" } = e;
  return l.$$set = (a) => {
    "className" in a && o(0, r = a.className);
  }, [r];
}
class Ut extends j {
  constructor(e) {
    super(), W(this, e, k1, v1, U, { className: 0 });
  }
}
function _1(l) {
  let e;
  return {
    c() {
      e = x("div"), e.innerHTML = '<span class="text-primary">Connect to Ready mobile instead</span> <svg xmlns="http://www.w3.org/2000/svg" width="27" height="48" viewBox="0 0 27 48" fill="none"><path d="M10.9682 45.7174L3.14321 13.0668C2.75958 11.4692 3.43225 9.79803 4.82488 8.91515L15.5876 2.07812C16.2077 1.68398 16.9539 1.72077 17.5162 2.06761C16.3548 0.81687 14.8098 0.795849 14.1319 1.33714L2.86994 8.48948C1.26184 9.50899 0.484071 11.4377 0.925509 13.2875L8.40368 44.5034C8.78205 46.0905 12.014 47.6303 13.3751 46.7264V46.7159C12.4449 47.2782 11.2257 46.779 10.9682 45.7069V45.7174Z" fill="#8C8C8C" stroke="#0F0F0F" stroke-width="0.262761" stroke-miterlimit="10"></path><path d="M12.8282 43.2841L14.0632 46.164L13.3748 46.7315C12.0137 47.6354 8.78696 46.0957 8.40333 44.5086L7.67285 41.4658L12.8282 43.2841Z" fill="#0F0F0F" stroke="#0F0F0F" stroke-width="0.315313" stroke-miterlimit="10"></path><path d="M26.0828 35.1019L18.3313 3.18697C18.037 1.97827 16.6391 1.41596 15.5881 2.07812L4.82542 8.91515C3.43805 9.79802 2.76012 11.4639 3.14375 13.0668L10.9635 45.7174C11.2263 46.8105 12.4928 47.3097 13.4335 46.6896L24.3958 39.4111C25.8095 38.4705 26.4822 36.752 26.0828 35.1019Z" fill="#8C8C8C" stroke="#0F0F0F" stroke-width="0.262761" stroke-miterlimit="10"></path><path d="M25.3354 34.6552L18.1252 4.10667C17.852 2.95052 16.5644 2.40398 15.6027 3.03986L14.2627 3.92274L13.7739 5.66747C13.6846 5.98803 13.4849 6.26131 13.2064 6.44524L8.79724 9.39867C8.4031 9.66143 7.90386 9.70872 7.47293 9.51954L6.44291 9.0781L5.72294 9.55632C4.44593 10.3972 3.83106 11.9895 4.18842 13.524L11.4669 44.782C11.7139 45.8278 12.8805 46.3113 13.7424 45.7174L23.8061 38.7858C25.1042 37.8924 25.719 36.2423 25.3459 34.6657L25.3354 34.6552Z" fill="#FF875B" stroke="#0F0F0F" stroke-width="0.315313" stroke-miterlimit="10"></path><path d="M9.23809 7.71168C9.26962 7.80627 9.37998 7.82729 9.48508 7.75372L12.6802 5.60434C12.7854 5.53602 12.8432 5.39938 12.8116 5.30479C12.7801 5.2102 12.6697 5.18917 12.5646 5.26275L9.36947 7.41213C9.26436 7.48045 9.20655 7.61708 9.23809 7.71168Z" fill="#0F0F0F"></path><path d="M8.51836 8.563C8.68127 8.55249 8.80214 8.38432 8.78637 8.19513C8.77061 8.00069 8.62872 7.85354 8.47106 7.86406C8.30815 7.87457 8.18728 8.04273 8.20304 8.23192C8.21881 8.42636 8.3607 8.57351 8.51836 8.563Z" fill="#0F0F0F"></path><path d="M8.93919 33.951L10.6629 8.14795L8.79204 9.39869C8.4452 9.62992 8.01427 9.69298 7.62013 9.57737L6.46924 23.3723L8.93393 33.9563L8.93919 33.951Z" fill="#FFCBB8"></path><path d="M5.69775 20.0089L5.97103 21.1755L7.01681 9.31975L6.68048 9.17261L5.69775 20.0036V20.0089Z" fill="#FFCBB8"></path><path d="M25.3354 34.6552L18.1252 4.10667C17.852 2.95052 16.5644 2.40398 15.6027 3.03986L14.2627 3.92274L13.7739 5.66747C13.6846 5.98803 13.4849 6.26131 13.2064 6.44524L8.79724 9.39867C8.4031 9.66143 7.90385 9.70872 7.47293 9.51954L6.44291 9.0781L5.72294 9.55632C4.44593 10.3972 3.83106 11.9895 4.18842 13.524L11.4669 44.782C11.7139 45.8278 12.8805 46.3113 13.7424 45.7174L23.8061 38.7858C25.1042 37.8924 25.719 36.2423 25.3459 34.6657L25.3354 34.6552Z" stroke="#0F0F0F" stroke-width="0.262761" stroke-miterlimit="10"></path><path d="M7.28408 39.5635L9.71724 40.2257L3.94702 16.4038L1.56641 15.9729L7.28408 39.5688V39.5635Z" fill="#0F0F0F" stroke="#0F0F0F" stroke-width="0.315313" stroke-miterlimit="10"></path><path d="M2.99527 13.1299C2.6747 11.7741 3.01629 10.3447 4.1304 9.25683L2.51705 8.94678C1.32411 9.93476 0.824868 11.4167 1.00355 12.8514L2.99527 13.1299Z" fill="#F8F8F8"></path><path d="M14.7094 20.855L13.5585 21.4804C13.5217 21.5014 13.5007 21.5592 13.5165 21.6065C13.9422 23.0149 13.7898 24.6492 13.0698 26.1207C13.0488 26.168 13.0593 26.2258 13.0961 26.2573L14.2207 27.0929C14.2575 27.1192 14.31 27.1087 14.3363 27.0614C14.783 26.147 15.0563 25.1748 15.1509 24.2026C15.7867 24.7911 16.5487 25.2273 17.3948 25.4796C17.4369 25.4901 17.4789 25.4585 17.4894 25.406L17.7259 23.8347C17.7312 23.7821 17.7049 23.7296 17.6629 23.7191C16.3385 23.3092 15.3033 22.2949 14.8145 20.9075C14.7988 20.8602 14.7515 20.834 14.7147 20.855H14.7094Z" fill="white"></path></svg>', t(e, "class", "flex flex-row-reverse items-center gap-2");
    },
    m(o, r) {
      $(o, e, r);
    },
    p: B,
    d(o) {
      o && M(e);
    }
  };
}
function y1(l) {
  let e, o, r, a, n;
  return o = new Ut({}), a = new Ot({
    props: {
      handleClick: (
        /*handleClick*/
        l[0]
      ),
      handleKeyup: (
        /*func*/
        l[1]
      ),
      $$slots: { default: [_1] },
      $$scope: { ctx: l }
    }
  }), {
    c() {
      e = x("footer"), A(o.$$.fragment), r = K(), A(a.$$.fragment), t(e, "class", "flex flex-col gap-4 w-full");
    },
    m(s, f) {
      $(s, e, f), N(o, e, null), i(e, r), N(a, e, null), n = !0;
    },
    p(s, [f]) {
      const d = {};
      f & /*handleClick*/
      1 && (d.handleClick = /*handleClick*/
      s[0]), f & /*handleClick*/
      1 && (d.handleKeyup = /*func*/
      s[1]), f & /*$$scope*/
      4 && (d.$$scope = { dirty: f, ctx: s }), a.$set(d);
    },
    i(s) {
      n || (w(o.$$.fragment, s), w(a.$$.fragment, s), n = !0);
    },
    o(s) {
      C(o.$$.fragment, s), C(a.$$.fragment, s), n = !1;
    },
    d(s) {
      s && M(e), Z(o), Z(a);
    }
  };
}
function x1(l, e, o) {
  let { handleClick: r = () => {
  } } = e;
  const a = (n) => {
    n.key === "Enter" && r();
  };
  return l.$$set = (n) => {
    "handleClick" in n && o(0, r = n.handleClick);
  }, [r, a];
}
class hr extends j {
  constructor(e) {
    super(), W(this, e, x1, y1, U, { handleClick: 0 });
  }
}
function qt(l) {
  let e, o;
  return e = new hr({
    props: { handleClick: (
      /*handleFallback*/
      l[2]
    ) }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, a) {
      const n = {};
      a & /*handleFallback*/
      4 && (n.handleClick = /*handleFallback*/
      r[2]), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function F1(l) {
  let e, o, r, a, n, s, f, d, p, h;
  const u = (
    /*#slots*/
    l[4].default
  ), g = be(
    u,
    l,
    /*$$scope*/
    l[3],
    null
  );
  let m = (
    /*showFallback*/
    l[1] && qt(l)
  );
  return {
    c() {
      e = x("section"), o = x("div"), r = x("div"), g && g.c(), a = K(), n = x("h3"), s = X("Connecting to "), f = X(
        /*walletName*/
        l[0]
      ), d = X("..."), p = K(), m && m.c(), t(r, "class", "bg-button-secondary rounded-full p-5"), t(n, "class", "text-primary text-h4 font-bold"), t(o, "class", "flex flex-col h-full justify-center items-center gap-4 w-full flex-grow"), t(e, "class", "flex flex-col justify-center items-center flex-grow");
    },
    m(b, v) {
      $(b, e, v), i(e, o), i(o, r), g && g.m(r, null), i(o, a), i(o, n), i(n, s), i(n, f), i(n, d), i(e, p), m && m.m(e, null), h = !0;
    },
    p(b, [v]) {
      g && g.p && (!h || v & /*$$scope*/
      8) && Ce(
        g,
        u,
        b,
        /*$$scope*/
        b[3],
        h ? we(
          u,
          /*$$scope*/
          b[3],
          v,
          null
        ) : ve(
          /*$$scope*/
          b[3]
        ),
        null
      ), (!h || v & /*walletName*/
      1) && ge(
        f,
        /*walletName*/
        b[0]
      ), /*showFallback*/
      b[1] ? m ? (m.p(b, v), v & /*showFallback*/
      2 && w(m, 1)) : (m = qt(b), m.c(), w(m, 1), m.m(e, null)) : m && (ce(), C(m, 1, 1, () => {
        m = null;
      }), fe());
    },
    i(b) {
      h || (w(g, b), w(m), h = !0);
    },
    o(b) {
      C(g, b), C(m), h = !1;
    },
    d(b) {
      b && M(e), g && g.d(b), m && m.d();
    }
  };
}
function L1(l, e, o) {
  let { $$slots: r = {}, $$scope: a } = e, { walletName: n = "" } = e, { showFallback: s = !1 } = e, { handleFallback: f = async () => {
  } } = e;
  return l.$$set = (d) => {
    "walletName" in d && o(0, n = d.walletName), "showFallback" in d && o(1, s = d.showFallback), "handleFallback" in d && o(2, f = d.handleFallback), "$$scope" in d && o(3, a = d.$$scope);
  }, [n, s, f, a, r];
}
class B1 extends j {
  constructor(e) {
    super(), W(this, e, L1, F1, U, {
      walletName: 0,
      showFallback: 1,
      handleFallback: 2
    });
  }
}
function M1(l) {
  let e;
  return {
    c() {
      e = x("section"), e.innerHTML = '<div class="flex flex-col h-full justify-center items-center gap-8 w-full flex-grow"><div><svg width="140" height="141" viewBox="0 0 140 141" fill="none" xmlns="http://www.w3.org/2000/svg"><mask id="mask0_82_210" style="mask-type:alpha" maskUnits="userSpaceOnUse" x="0" y="0" width="140" height="141"><path d="M69.9639 140.094C108.42 140.094 139.595 108.92 139.595 70.4639C139.595 32.008 108.42 0.833313 69.9639 0.833313C31.508 0.833313 0.333344 32.008 0.333344 70.4639C0.333344 108.92 31.508 140.094 69.9639 140.094Z" fill="#BFBFBF"></path></mask><g mask="url(#mask0_82_210)"><path d="M69.9639 140.094C108.42 140.094 139.595 108.92 139.595 70.4639C139.595 32.008 108.42 0.833313 69.9639 0.833313C31.508 0.833313 0.333344 32.008 0.333344 70.4639C0.333344 108.92 31.508 140.094 69.9639 140.094Z" fill="#BFBFBF" fill-opacity="0.2"></path><circle opacity="0.5" cx="69.6835" cy="70.1833" r="28.5" stroke="#FF7A52" stroke-width="0.633333"></circle><circle opacity="0.4" cx="69.6835" cy="70.1833" r="36.1" stroke="#FF7A52" stroke-width="0.633333"></circle><circle opacity="0.2" cx="69.6834" cy="70.1833" r="50.0333" stroke="#FF7A52" stroke-width="0.633333"></circle><path d="M71.63 64.1667H68.3704C68.2614 64.1667 68.1742 64.2607 68.1718 64.3775C68.1059 67.6613 66.5044 70.7781 63.7479 72.9857C63.6603 73.0557 63.6404 73.1874 63.7044 73.2823L65.6115 76.1111C65.6764 76.2074 65.8028 76.2291 65.8918 76.1584C67.6154 74.7871 69.0018 73.1327 70.0002 71.2991C70.9985 73.1327 72.385 74.7871 74.1086 76.1584C74.1975 76.2291 74.3239 76.2074 74.3889 76.1111L76.296 73.2823C76.3599 73.1874 76.34 73.0557 76.2525 72.9857C73.4959 70.7781 71.8944 67.6613 71.8286 64.3775C71.8262 64.2607 71.7389 64.1667 71.63 64.1667Z" fill="#FF875B"></path><path d="M42.2656 41.1541L97.3366 41.1541C102.578 41.1541 106.827 45.4032 106.827 50.6448L106.827 170.156C106.827 175.398 102.578 179.647 97.3366 179.647L42.2656 179.647C37.0241 179.647 32.775 175.398 32.775 170.156L32.775 50.6447C32.775 45.4032 37.0241 41.1541 42.2656 41.1541Z" fill="#FF875B" stroke="black" stroke-width="3.51504"></path><path d="M44.6111 139.224L38.2837 139.224L31.9568 139.224L32.6599 47.1296L37.5617 41.0012L94.5247 40.803L44.6111 139.224Z" fill="white" fill-opacity="0.3"></path><path d="M42.266 41.1541L97.337 41.1541C102.578 41.1541 106.828 45.4032 106.828 50.6448L106.828 170.156C106.828 175.398 102.578 179.647 97.3369 179.647L42.2659 179.647C37.0244 179.647 32.7753 175.398 32.7753 170.156L32.7753 50.6447C32.7753 45.4032 37.0244 41.1541 42.266 41.1541Z" stroke="black" stroke-width="3.51504"></path><path d="M53.6909 39.3966L87.1044 39.3966L87.1044 40.9326C87.1044 44.0387 84.5865 46.5567 81.4804 46.5567L59.3149 46.5567C56.2088 46.5567 53.6909 44.0387 53.6909 40.9326L53.6909 39.3966Z" fill="black"></path><path d="M73.0967 82.5333H66.9034C66.6965 82.5333 66.5307 82.7164 66.5262 82.9439C66.401 89.3387 63.3581 95.4081 58.1207 99.7072C57.9544 99.8436 57.9165 100.1 58.0381 100.285L61.6616 105.793C61.7849 105.981 62.0251 106.023 62.1942 105.886C65.469 103.215 68.1031 99.9935 70 96.4228C71.897 99.9935 74.5312 103.215 77.8061 105.886C77.975 106.023 78.2152 105.981 78.3386 105.793L81.9622 100.285C82.0836 100.1 82.0457 99.8436 81.8796 99.7072C76.642 95.4081 73.5991 89.3387 73.4741 82.9439C73.4696 82.7164 73.3036 82.5333 73.0967 82.5333Z" fill="white"></path></g></svg></div> <hgroup class="flex flex-col gap-1"><h3 class="text-primary text-h4 font-bold">Approve the transaction</h3> <h4 class="text-primary text-p3">Open Ready on your phone to <br/> approve the transaction</h4></hgroup></div>', t(e, "class", "flex flex-col justify-center items-center flex-grow");
    },
    m(o, r) {
      $(o, e, r);
    },
    p: B,
    i: B,
    o: B,
    d(o) {
      o && M(e);
    }
  };
}
function S1(l) {
  return [];
}
class $1 extends j {
  constructor(e) {
    super(), W(this, e, S1, M1, U, {});
  }
}
var Xt = globalThis && globalThis.__spreadArray || function(l, e, o) {
  if (o || arguments.length === 2)
    for (var r = 0, a = e.length, n; r < a; r++)
      (n || !(r in e)) && (n || (n = Array.prototype.slice.call(e, 0, r)), n[r] = e[r]);
  return l.concat(n || Array.prototype.slice.call(e));
}, E1 = (
  /** @class */
  function() {
    function l(e, o, r) {
      this.name = e, this.version = o, this.os = r, this.type = "browser";
    }
    return l;
  }()
), N1 = (
  /** @class */
  function() {
    function l(e) {
      this.version = e, this.type = "node", this.name = "node", this.os = process.platform;
    }
    return l;
  }()
), Z1 = (
  /** @class */
  function() {
    function l(e, o, r, a) {
      this.name = e, this.version = o, this.os = r, this.bot = a, this.type = "bot-device";
    }
    return l;
  }()
), A1 = (
  /** @class */
  function() {
    function l() {
      this.type = "bot", this.bot = !0, this.name = "bot", this.version = null, this.os = null;
    }
    return l;
  }()
), z1 = (
  /** @class */
  function() {
    function l() {
      this.type = "react-native", this.name = "react-native", this.version = null, this.os = null;
    }
    return l;
  }()
), O1 = /alexa|bot|crawl(er|ing)|facebookexternalhit|feedburner|google web preview|nagios|postrank|pingdom|slurp|spider|yahoo!|yandex/, U1 = /(nuhk|curl|Googlebot|Yammybot|Openbot|Slurp|MSNBot|Ask\ Jeeves\/Teoma|ia_archiver)/, Qt = 3, G1 = [
  ["aol", /AOLShield\/([0-9\._]+)/],
  ["edge", /Edge\/([0-9\._]+)/],
  ["edge-ios", /EdgiOS\/([0-9\._]+)/],
  ["yandexbrowser", /YaBrowser\/([0-9\._]+)/],
  ["kakaotalk", /KAKAOTALK\s([0-9\.]+)/],
  ["samsung", /SamsungBrowser\/([0-9\.]+)/],
  ["silk", /\bSilk\/([0-9._-]+)\b/],
  ["miui", /MiuiBrowser\/([0-9\.]+)$/],
  ["beaker", /BeakerBrowser\/([0-9\.]+)/],
  ["edge-chromium", /EdgA?\/([0-9\.]+)/],
  [
    "chromium-webview",
    /(?!Chrom.*OPR)wv\).*Chrom(?:e|ium)\/([0-9\.]+)(:?\s|$)/
  ],
  ["chrome", /(?!Chrom.*OPR)Chrom(?:e|ium)\/([0-9\.]+)(:?\s|$)/],
  ["phantomjs", /PhantomJS\/([0-9\.]+)(:?\s|$)/],
  ["crios", /CriOS\/([0-9\.]+)(:?\s|$)/],
  ["firefox", /Firefox\/([0-9\.]+)(?:\s|$)/],
  ["fxios", /FxiOS\/([0-9\.]+)/],
  ["opera-mini", /Opera Mini.*Version\/([0-9\.]+)/],
  ["opera", /Opera\/([0-9\.]+)(?:\s|$)/],
  ["opera", /OPR\/([0-9\.]+)(:?\s|$)/],
  ["pie", /^Microsoft Pocket Internet Explorer\/(\d+\.\d+)$/],
  ["pie", /^Mozilla\/\d\.\d+\s\(compatible;\s(?:MSP?IE|MSInternet Explorer) (\d+\.\d+);.*Windows CE.*\)$/],
  ["netfront", /^Mozilla\/\d\.\d+.*NetFront\/(\d.\d)/],
  ["ie", /Trident\/7\.0.*rv\:([0-9\.]+).*\).*Gecko$/],
  ["ie", /MSIE\s([0-9\.]+);.*Trident\/[4-7].0/],
  ["ie", /MSIE\s(7\.0)/],
  ["bb10", /BB10;\sTouch.*Version\/([0-9\.]+)/],
  ["android", /Android\s([0-9\.]+)/],
  ["ios", /Version\/([0-9\._]+).*Mobile.*Safari.*/],
  ["safari", /Version\/([0-9\._]+).*Safari/],
  ["facebook", /FB[AS]V\/([0-9\.]+)/],
  ["instagram", /Instagram\s([0-9\.]+)/],
  ["ios-webview", /AppleWebKit\/([0-9\.]+).*Mobile/],
  ["ios-webview", /AppleWebKit\/([0-9\.]+).*Gecko\)$/],
  ["curl", /^curl\/([0-9\.]+)$/],
  ["searchbot", O1]
], Yt = [
  ["iOS", /iP(hone|od|ad)/],
  ["Android OS", /Android/],
  ["BlackBerry OS", /BlackBerry|BB10/],
  ["Windows Mobile", /IEMobile/],
  ["Amazon OS", /Kindle/],
  ["Windows 3.11", /Win16/],
  ["Windows 95", /(Windows 95)|(Win95)|(Windows_95)/],
  ["Windows 98", /(Windows 98)|(Win98)/],
  ["Windows 2000", /(Windows NT 5.0)|(Windows 2000)/],
  ["Windows XP", /(Windows NT 5.1)|(Windows XP)/],
  ["Windows Server 2003", /(Windows NT 5.2)/],
  ["Windows Vista", /(Windows NT 6.0)/],
  ["Windows 7", /(Windows NT 6.1)/],
  ["Windows 8", /(Windows NT 6.2)/],
  ["Windows 8.1", /(Windows NT 6.3)/],
  ["Windows 10", /(Windows NT 10.0)/],
  ["Windows ME", /Windows ME/],
  ["Windows CE", /Windows CE|WinCE|Microsoft Pocket Internet Explorer/],
  ["Open BSD", /OpenBSD/],
  ["Sun OS", /SunOS/],
  ["Chrome OS", /CrOS/],
  ["Linux", /(Linux)|(X11)/],
  ["Mac OS", /(Mac_PowerPC)|(Macintosh)/],
  ["QNX", /QNX/],
  ["BeOS", /BeOS/],
  ["OS/2", /OS\/2/]
];
function I1(l) {
  return l ? Jt(l) : typeof document > "u" && typeof navigator < "u" && navigator.product === "ReactNative" ? new z1() : typeof navigator < "u" ? Jt(navigator.userAgent) : D1();
}
function W1(l) {
  return l !== "" && G1.reduce(function(e, o) {
    var r = o[0], a = o[1];
    if (e)
      return e;
    var n = a.exec(l);
    return !!n && [r, n];
  }, !1);
}
function Jt(l) {
  var e = W1(l);
  if (!e)
    return null;
  var o = e[0], r = e[1];
  if (o === "searchbot")
    return new A1();
  var a = r[1] && r[1].split(".").join("_").split("_").slice(0, 3);
  a ? a.length < Qt && (a = Xt(Xt([], a, !0), V1(Qt - a.length), !0)) : a = [];
  var n = a.join("."), s = j1(l), f = U1.exec(l);
  return f && f[1] ? new Z1(o, n, s, f[1]) : new E1(o, n, s);
}
function j1(l) {
  for (var e = 0, o = Yt.length; e < o; e++) {
    var r = Yt[e], a = r[0], n = r[1], s = n.exec(l);
    if (s)
      return a;
  }
  return null;
}
function D1() {
  var l = typeof process < "u" && process.version;
  return l ? new N1(process.version.slice(1)) : null;
}
function V1(l) {
  for (var e = [], o = 0; o < l; o++)
    e.push("0");
  return e;
}
const R1 = () => {
  const e = I1()?.os?.toLowerCase();
  return e?.includes("android") ? "android" : e?.toLowerCase().includes("ios") || e?.toLowerCase().includes("mac") && navigator.maxTouchPoints > 1 ? "ios" : "desktop";
};
function T1(l) {
  let e, o, r, a, n, s, f, d, p, h, u, g, m, b, v, F, O, z, L, k, y, S, G, _, E, R, D;
  return {
    c() {
      e = c("svg"), o = c("path"), r = c("path"), a = c("path"), n = c("path"), s = c("defs"), f = c("linearGradient"), d = c("stop"), p = c("stop"), h = c("stop"), u = c("stop"), g = c("stop"), m = c("stop"), b = c("linearGradient"), v = c("stop"), F = c("stop"), O = c("stop"), z = c("stop"), L = c("linearGradient"), k = c("stop"), y = c("stop"), S = c("linearGradient"), G = c("stop"), _ = c("stop"), E = c("stop"), R = c("stop"), D = c("stop"), t(o, "d", "M0.467777 0.822764C0.216151 1.08602 0.0705566 1.49585 0.0705566 2.02664V20.9585C0.0705566 21.4892 0.216151 21.8991 0.467777 22.1623L0.529877 22.2201L11.1456 11.6156V11.3652L0.529877 0.760742L0.467777 0.822764Z"), t(o, "fill", "url(#paint0_linear_6328_20701)"), t(r, "d", "M14.6806 15.1527L11.146 11.6161V11.3657L14.6849 7.8291L14.7642 7.87512L18.9551 10.2582C20.1512 10.9345 20.1512 12.0474 18.9551 12.728L14.7642 15.1067L14.6806 15.1527Z"), t(r, "fill", "url(#paint1_linear_6328_20701)"), t(a, "d", "M14.7646 15.1065L11.1464 11.4907L0.468628 22.1626C0.865912 22.58 1.51368 22.6302 2.25039 22.2129L14.7646 15.1065Z"), t(a, "fill", "url(#paint2_linear_6328_20701)"), t(n, "d", "M14.7646 7.87514L2.25039 0.768742C1.51368 0.355689 0.865912 0.405984 0.468628 0.823338L11.1464 11.4909L14.7646 7.87514Z"), t(n, "fill", "url(#paint3_linear_6328_20701)"), t(d, "stop-color", "#00A0FF"), t(p, "offset", "0.0066"), t(p, "stop-color", "#00A1FF"), t(h, "offset", "0.2601"), t(h, "stop-color", "#00BEFF"), t(u, "offset", "0.5122"), t(u, "stop-color", "#00D2FF"), t(g, "offset", "0.7604"), t(g, "stop-color", "#00DFFF"), t(m, "offset", "1"), t(m, "stop-color", "#00E3FF"), t(f, "id", "paint0_linear_6328_20701"), t(f, "x1", "10.2031"), t(f, "y1", "21.1557"), t(f, "x2", "-4.16375"), t(f, "y2", "6.77945"), t(f, "gradientUnits", "userSpaceOnUse"), t(v, "stop-color", "#FFE000"), t(F, "offset", "0.4087"), t(F, "stop-color", "#FFBD00"), t(O, "offset", "0.7754"), t(O, "stop-color", "#FFA500"), t(z, "offset", "1"), t(z, "stop-color", "#FF9C00"), t(b, "id", "paint1_linear_6328_20701"), t(b, "x1", "20.5126"), t(b, "y1", "11.4897"), t(b, "x2", "-0.214967"), t(b, "y2", "11.4897"), t(b, "gradientUnits", "userSpaceOnUse"), t(k, "stop-color", "#FF3A44"), t(y, "offset", "1"), t(y, "stop-color", "#C31162"), t(L, "id", "paint2_linear_6328_20701"), t(L, "x1", "12.7972"), t(L, "y1", "9.52511"), t(L, "x2", "-6.68541"), t(L, "y2", "-9.97016"), t(L, "gradientUnits", "userSpaceOnUse"), t(G, "stop-color", "#32A071"), t(_, "offset", "0.0685"), t(_, "stop-color", "#2DA771"), t(E, "offset", "0.4762"), t(E, "stop-color", "#15CF74"), t(R, "offset", "0.8009"), t(R, "stop-color", "#06E775"), t(D, "offset", "1"), t(D, "stop-color", "#00F076"), t(S, "id", "paint3_linear_6328_20701"), t(S, "x1", "-2.2192"), t(S, "y1", "28.4614"), t(S, "x2", "6.48066"), t(S, "y2", "19.756"), t(S, "gradientUnits", "userSpaceOnUse"), t(e, "xmlns", "http://www.w3.org/2000/svg"), t(e, "width", "20"), t(e, "height", "23"), t(e, "viewBox", "0 0 20 23"), t(e, "fill", "none");
    },
    m(T, q) {
      $(T, e, q), i(e, o), i(e, r), i(e, a), i(e, n), i(e, s), i(s, f), i(f, d), i(f, p), i(f, h), i(f, u), i(f, g), i(f, m), i(s, b), i(b, v), i(b, F), i(b, O), i(b, z), i(s, L), i(L, k), i(L, y), i(s, S), i(S, G), i(S, _), i(S, E), i(S, R), i(S, D);
    },
    p: B,
    i: B,
    o: B,
    d(T) {
      T && M(e);
    }
  };
}
class gr extends j {
  constructor(e) {
    super(), W(this, e, null, T1, U, {});
  }
}
function H1(l) {
  let e, o, r, a, n, s, f, d, p;
  return {
    c() {
      e = c("svg"), o = c("mask"), r = c("path"), a = c("g"), n = c("path"), s = c("path"), f = c("path"), d = c("path"), p = c("path"), t(r, "d", "M0.147583 10.7547C0.147583 4.81908 4.97505 0.00732422 10.93 0.00732422C16.885 0.00732422 21.7125 4.81908 21.7125 10.7547C21.7125 16.6903 16.885 21.502 10.93 21.502C4.97505 21.502 0.147583 16.6903 0.147583 10.7547Z"), t(r, "fill", "#F1F1F1"), t(o, "id", "mask0_6328_20691"), xe(o, "mask-type", "alpha"), t(o, "maskUnits", "userSpaceOnUse"), t(o, "x", "0"), t(o, "y", "0"), t(
        o,
        "width",
        /*size*/
        l[0]
      ), t(
        o,
        "height",
        /*size*/
        l[0]
      ), t(n, "d", "M1.85962 0.00787449V13.1973H6.68474L10.9296 5.86942H21.712V0.00787449H1.85962ZM1.85962 0.00787449V13.1973H6.68474L10.9296 5.86942H21.712V0.00787449H1.85962Z"), t(n, "fill", "#EA4335"), t(s, "d", "M0.147583 21.502H10.4107L15.1749 16.7533V13.1981L6.68507 13.1967L0.147583 2.02082V21.502ZM0.147583 21.502H10.4107L15.1749 16.7533V13.1981L6.68507 13.1967L0.147583 2.02082V21.502Z"), t(s, "fill", "#34A853"), t(f, "d", "M10.9289 5.86867L15.1738 13.1965L10.411 21.502H21.7113V5.86867H10.9289ZM10.9289 5.86867L15.1738 13.1965L10.411 21.502H21.7113V5.86867H10.9289Z"), t(f, "fill", "#FBBC05"), t(d, "d", "M10.9297 5.86791C13.6369 5.86791 15.8314 8.05549 15.8314 10.7537C15.8314 13.4521 13.6369 15.6396 10.9297 15.6396C8.22245 15.6396 6.02795 13.4521 6.02795 10.7537C6.02795 8.05549 8.22245 5.86791 10.9297 5.86791Z"), t(d, "fill", "#F1F1F1"), t(p, "d", "M10.9286 6.84588C13.0943 6.84588 14.8499 8.59581 14.8499 10.7545C14.8499 12.9132 13.0943 14.6631 10.9286 14.6631C8.76284 14.6631 7.0072 12.9132 7.0072 10.7545C7.0072 8.59581 8.76284 6.84588 10.9286 6.84588Z"), t(p, "fill", "#4285F4"), t(a, "mask", "url(#mask0_6328_20691)"), t(e, "xmlns", "http://www.w3.org/2000/svg"), t(
        e,
        "width",
        /*size*/
        l[0]
      ), t(
        e,
        "height",
        /*size*/
        l[0]
      ), t(e, "viewBox", "0 0 22 22"), t(e, "fill", "none");
    },
    m(h, u) {
      $(h, e, u), i(e, o), i(o, r), i(e, a), i(a, n), i(a, s), i(a, f), i(a, d), i(a, p);
    },
    p(h, [u]) {
      u & /*size*/
      1 && t(
        o,
        "width",
        /*size*/
        h[0]
      ), u & /*size*/
      1 && t(
        o,
        "height",
        /*size*/
        h[0]
      ), u & /*size*/
      1 && t(
        e,
        "width",
        /*size*/
        h[0]
      ), u & /*size*/
      1 && t(
        e,
        "height",
        /*size*/
        h[0]
      );
    },
    i: B,
    o: B,
    d(h) {
      h && M(e);
    }
  };
}
function K1(l, e, o) {
  let { size: r = "22" } = e;
  return l.$$set = (a) => {
    "size" in a && o(0, r = a.size);
  }, [r];
}
class mr extends j {
  constructor(e) {
    super(), W(this, e, K1, H1, U, { size: 0 });
  }
}
function P1(l) {
  let e, o, r;
  return {
    c() {
      e = c("svg"), o = c("path"), r = c("path"), t(o, "d", "M17.71 13.7971C17.6815 10.6387 20.296 9.10219 20.4156 9.03069C18.9349 6.87185 16.6399 6.57692 15.8334 6.5534C13.9059 6.35052 12.0362 7.70675 11.0543 7.70675C10.0528 7.70675 8.54067 6.573 6.91107 6.60632C4.814 6.63862 2.85216 7.8528 1.77619 9.7382C-0.444353 13.5825 1.21174 19.2319 3.33916 22.3392C4.40337 23.8611 5.64695 25.5603 7.27459 25.5005C8.86703 25.4349 9.46182 24.4853 11.3835 24.4853C13.2875 24.4853 13.8461 25.5005 15.5061 25.4623C17.2151 25.4349 18.2911 23.9336 19.318 22.398C20.5479 20.6537 21.0418 18.9359 21.0614 18.8477C21.0212 18.834 17.7423 17.5826 17.71 13.7971Z"), t(o, "fill", "currentColor"), t(r, "d", "M14.5762 4.5089C15.4327 3.43787 16.0187 1.98066 15.856 0.501953C14.6164 0.55683 13.0662 1.3594 12.1734 2.40697C11.3836 3.33007 10.678 4.8431 10.8603 6.26597C12.2528 6.36984 13.6825 5.56332 14.5762 4.5089Z"), t(r, "fill", "currentColor"), t(e, "xmlns", "http://www.w3.org/2000/svg"), t(e, "width", "22"), t(e, "height", "26"), t(e, "viewBox", "0 0 22 26"), t(e, "fill", "none");
    },
    m(a, n) {
      $(a, e, n), i(e, o), i(e, r);
    },
    p: B,
    i: B,
    o: B,
    d(a) {
      a && M(e);
    }
  };
}
class br extends j {
  constructor(e) {
    super(), W(this, e, null, P1, U, {});
  }
}
function q1(l) {
  let e, o, r, a, n, s, f, d, p, h;
  return n = new mr({}), f = new br({}), p = new gr({}), {
    c() {
      e = x("div"), o = x("p"), o.textContent = "Get it", r = K(), a = x("div"), A(n.$$.fragment), s = K(), A(f.$$.fragment), d = K(), A(p.$$.fragment), t(o, "class", "font-semibold text-p3 text-primary"), t(a, "class", "flex gap-3 items-end"), xe(a, "position", "relative"), t(e, "class", "flex flex-row-reverse justify-center items-center gap-2");
    },
    m(u, g) {
      $(u, e, g), i(e, o), i(e, r), i(e, a), N(n, a, null), i(a, s), N(f, a, null), i(a, d), N(p, a, null), h = !0;
    },
    p: B,
    i(u) {
      h || (w(n.$$.fragment, u), w(f.$$.fragment, u), w(p.$$.fragment, u), h = !0);
    },
    o(u) {
      C(n.$$.fragment, u), C(f.$$.fragment, u), C(p.$$.fragment, u), h = !1;
    },
    d(u) {
      u && M(e), Z(n), Z(f), Z(p);
    }
  };
}
function X1(l) {
  let e, o, r, a, n, s, f, d, p, h;
  return o = new Ut({}), p = new Ot({
    props: {
      handleClick: (
        /*handleClick*/
        l[1]
      ),
      handleKeyup: (
        /*func*/
        l[2]
      ),
      $$slots: { default: [q1] },
      $$scope: { ctx: l }
    }
  }), {
    c() {
      e = x("footer"), A(o.$$.fragment), r = K(), a = x("p"), n = X("Don't have "), s = X(
        /*walletName*/
        l[0]
      ), f = X("?"), d = K(), A(p.$$.fragment), t(a, "class", "text-h5 text-primary"), t(e, "class", "flex flex-col gap-4");
    },
    m(u, g) {
      $(u, e, g), N(o, e, null), i(e, r), i(e, a), i(a, n), i(a, s), i(a, f), i(e, d), N(p, e, null), h = !0;
    },
    p(u, [g]) {
      (!h || g & /*walletName*/
      1) && ge(
        s,
        /*walletName*/
        u[0]
      );
      const m = {};
      g & /*handleClick*/
      2 && (m.handleClick = /*handleClick*/
      u[1]), g & /*handleClick*/
      2 && (m.handleKeyup = /*func*/
      u[2]), g & /*$$scope*/
      8 && (m.$$scope = { dirty: g, ctx: u }), p.$set(m);
    },
    i(u) {
      h || (w(o.$$.fragment, u), w(p.$$.fragment, u), h = !0);
    },
    o(u) {
      C(o.$$.fragment, u), C(p.$$.fragment, u), h = !1;
    },
    d(u) {
      u && M(e), Z(o), Z(p);
    }
  };
}
function Q1(l, e, o) {
  let { walletName: r = "" } = e, { handleClick: a = () => {
  } } = e;
  const n = (s) => {
    s.key === "Enter" && a();
  };
  return l.$$set = (s) => {
    "walletName" in s && o(0, r = s.walletName), "handleClick" in s && o(1, a = s.handleClick);
  }, [r, a, n];
}
class Y1 extends j {
  constructor(e) {
    super(), W(this, e, Q1, X1, U, { walletName: 0, handleClick: 1 });
  }
}
function J1(l) {
  Zr(l, "svelte-roekpq", "@keyframes svelte-roekpq-dash{0%{stroke-dashoffset:187}50%{stroke-dashoffset:46.75;transform:rotate(135deg)}100%{stroke-dashoffset:187;transform:rotate(450deg)}}@keyframes svelte-roekpq-rotator{0%{transform:rotate(0deg)}100%{transform:rotate(270deg)}}.spinner.svelte-roekpq.svelte-roekpq{color:#ff875b;animation:svelte-roekpq-rotator 1.4s linear infinite}.spinner.svelte-roekpq circle.svelte-roekpq{stroke:currentColor;stroke-dasharray:187;stroke-dashoffset:0;transform-origin:center;animation:svelte-roekpq-dash 1.4s ease-in-out infinite}");
}
function eo(l) {
  let e;
  return {
    c() {
      e = x("h3"), e.innerHTML = "Connect Ready mobile by<br/>scanning QR code:", t(e, "class", "mb-4 text-h5 text-primary font-semibold");
    },
    m(o, r) {
      $(o, e, r);
    },
    d(o) {
      o && M(e);
    }
  };
}
function to(l) {
  let e;
  return {
    c() {
      e = x("h3"), e.textContent = "Connecting to Ready mobile...", t(e, "class", "mb-4 text-h5 text-primary font-semibold");
    },
    m(o, r) {
      $(o, e, r);
    },
    d(o) {
      o && M(e);
    }
  };
}
function ro(l) {
  let e, o, r, a, n, s, f;
  function d(u, g) {
    return (
      /*isMobile*/
      u[1] ? to : eo
    );
  }
  let h = d(l)(l);
  return s = new Y1({
    props: {
      walletName: "Ready",
      handleClick: (
        /*handleInstallClick*/
        l[0]
      )
    }
  }), {
    c() {
      e = x("section"), o = x("div"), h.c(), r = K(), a = x("div"), a.innerHTML = '<svg class="spinner svelte-roekpq" viewBox="0 0 66 66" width="100px" height="100px" xmlns="http://www.w3.org/2000/svg"><circle fill="none" stroke-width="6" stroke-linecap="round" cx="33" cy="33" r="30" class="svelte-roekpq"></circle></svg>', n = K(), A(s.$$.fragment), t(a, "class", "w-full flex justify-center items-center qr-code-slot h-[245px]"), t(e, "class", "flex flex-col gap-4");
    },
    m(u, g) {
      $(u, e, g), i(e, o), h.m(o, null), i(o, r), i(o, a), i(e, n), N(s, e, null), f = !0;
    },
    p(u, [g]) {
      const m = {};
      g & /*handleInstallClick*/
      1 && (m.handleClick = /*handleInstallClick*/
      u[0]), s.$set(m);
    },
    i(u) {
      f || (w(s.$$.fragment, u), f = !0);
    },
    o(u) {
      C(s.$$.fragment, u), f = !1;
    },
    d(u) {
      u && M(e), h.d(), Z(s);
    }
  };
}
function oo(l, e, o) {
  const r = R1(), a = ["android", "ios"].includes(r);
  let { handleInstallClick: n = () => {
  } } = e;
  return l.$$set = (s) => {
    "handleInstallClick" in s && o(0, n = s.handleInstallClick);
  }, [n, a, r];
}
class lo extends j {
  constructor(e) {
    super(), W(this, e, oo, ro, U, { device: 2, handleInstallClick: 0 }, J1);
  }
  get device() {
    return this.$$.ctx[2];
  }
}
function ao(l) {
  let e, o;
  return {
    c() {
      e = c("svg"), o = c("path"), t(o, "d", "M37.5407 29.0784L23.8767 5.34867C23.4809 4.67181 22.9148 4.11038 22.2347 3.72021C21.5546 3.33005 20.7842 3.12476 20.0001 3.12476C19.216 3.12476 18.4456 3.33005 17.7655 3.72021C17.0854 4.11038 16.5193 4.67181 16.1235 5.34867L2.45947 29.0784C2.07544 29.7352 1.87305 30.4823 1.87305 31.2432C1.87305 32.0041 2.07544 32.7512 2.45947 33.408C2.85086 34.0883 3.41618 34.6521 4.09741 35.0418C4.77865 35.4314 5.55128 35.6328 6.33604 35.6252H33.6642C34.4484 35.6322 35.2204 35.4306 35.901 35.041C36.5817 34.6514 37.1465 34.0878 37.5376 33.408C37.9221 32.7515 38.125 32.0045 38.1256 31.2436C38.1261 30.4828 37.9243 29.7355 37.5407 29.0784ZM34.2907 31.5315C34.2272 31.6398 34.1356 31.7289 34.0255 31.7894C33.9155 31.8499 33.7912 31.8795 33.6657 31.8752H6.33604C6.21055 31.8795 6.08624 31.8499 5.97622 31.7894C5.86621 31.7289 5.77458 31.6398 5.71104 31.5315C5.65713 31.4438 5.6286 31.343 5.6286 31.2401C5.6286 31.1372 5.65713 31.0363 5.71104 30.9487L19.3751 7.21898C19.4428 7.11483 19.5354 7.02925 19.6445 6.96999C19.7537 6.91074 19.8759 6.87971 20.0001 6.87971C20.1243 6.87971 20.2465 6.91074 20.3557 6.96999C20.4648 7.02925 20.5574 7.11483 20.6251 7.21898L34.2876 30.9487C34.342 31.036 34.3711 31.1367 34.3716 31.2396C34.3722 31.3425 34.3442 31.4436 34.2907 31.5315ZM18.1251 21.2502V16.2502C18.1251 15.753 18.3226 15.276 18.6743 14.9244C19.0259 14.5728 19.5028 14.3752 20.0001 14.3752C20.4974 14.3752 20.9743 14.5728 21.3259 14.9244C21.6776 15.276 21.8751 15.753 21.8751 16.2502V21.2502C21.8751 21.7475 21.6776 22.2244 21.3259 22.5761C20.9743 22.9277 20.4974 23.1252 20.0001 23.1252C19.5028 23.1252 19.0259 22.9277 18.6743 22.5761C18.3226 22.2244 18.1251 21.7475 18.1251 21.2502ZM22.5001 27.5002C22.5001 27.9947 22.3535 28.478 22.0788 28.8892C21.8041 29.3003 21.4136 29.6207 20.9568 29.8099C20.5 29.9991 19.9973 30.0487 19.5124 29.9522C19.0274 29.8557 18.582 29.6176 18.2323 29.268C17.8827 28.9184 17.6446 28.4729 17.5481 27.988C17.4517 27.503 17.5012 27.0003 17.6904 26.5435C17.8796 26.0867 18.2001 25.6963 18.6112 25.4216C19.0223 25.1469 19.5056 25.0002 20.0001 25.0002C20.6631 25.0002 21.299 25.2636 21.7679 25.7325C22.2367 26.2013 22.5001 26.8372 22.5001 27.5002Z"), t(o, "fill", "#DB3B4E"), t(e, "xmlns", "http://www.w3.org/2000/svg"), t(e, "width", "40"), t(e, "height", "40"), t(e, "viewBox", "0 0 40 40"), t(e, "fill", "none");
    },
    m(r, a) {
      $(r, e, a), i(e, o);
    },
    p: B,
    i: B,
    o: B,
    d(r) {
      r && M(e);
    }
  };
}
class wr extends j {
  constructor(e) {
    super(), W(this, e, null, ao, U, {});
  }
}
function Et(l) {
  let e, o, r, a;
  const n = (
    /*#slots*/
    l[6].default
  ), s = be(
    n,
    l,
    /*$$scope*/
    l[5],
    null
  );
  let f = [
    { class: (
      /*className*/
      l[0]
    ) },
    { href: (
      /*href*/
      l[1]
    ) },
    { rel: "noreferrer noopener" },
    { target: "_blank" },
    { role: "button" },
    { tabindex: "0" }
  ], d = {};
  for (let p = 0; p < f.length; p += 1)
    d = xt(d, f[p]);
  return {
    c() {
      e = x(
        /*as*/
        l[2]
      ), s && s.c(), Ee(
        /*as*/
        l[2]
      )(e, d);
    },
    m(p, h) {
      $(p, e, h), s && s.m(e, null), o = !0, r || (a = [
        $e(e, "click", function() {
          me(
            /*handleClick*/
            l[3]
          ) && l[3].apply(this, arguments);
        }),
        $e(e, "keyup", function() {
          me(
            /*handleKeyup*/
            l[4]
          ) && l[4].apply(this, arguments);
        })
      ], r = !0);
    },
    p(p, h) {
      l = p, s && s.p && (!o || h & /*$$scope*/
      32) && Ce(
        s,
        n,
        l,
        /*$$scope*/
        l[5],
        o ? we(
          n,
          /*$$scope*/
          l[5],
          h,
          null
        ) : ve(
          /*$$scope*/
          l[5]
        ),
        null
      ), Ee(
        /*as*/
        l[2]
      )(e, d = zt(f, [
        (!o || h & /*className*/
        1) && { class: (
          /*className*/
          l[0]
        ) },
        (!o || h & /*href*/
        2) && { href: (
          /*href*/
          l[1]
        ) },
        { rel: "noreferrer noopener" },
        { target: "_blank" },
        { role: "button" },
        { tabindex: "0" }
      ]));
    },
    i(p) {
      o || (w(s, p), o = !0);
    },
    o(p) {
      C(s, p), o = !1;
    },
    d(p) {
      p && M(e), s && s.d(p), r = !1, ke(a);
    }
  };
}
function no(l) {
  let e = (
    /*as*/
    l[2]
  ), o, r, a = (
    /*as*/
    l[2] && Et(l)
  );
  return {
    c() {
      a && a.c(), o = ye();
    },
    m(n, s) {
      a && a.m(n, s), $(n, o, s), r = !0;
    },
    p(n, [s]) {
      /*as*/
      n[2] ? e ? U(
        e,
        /*as*/
        n[2]
      ) ? (a.d(1), a = Et(n), e = /*as*/
      n[2], a.c(), a.m(o.parentNode, o)) : a.p(n, s) : (a = Et(n), e = /*as*/
      n[2], a.c(), a.m(o.parentNode, o)) : e && (a.d(1), a = null, e = /*as*/
      n[2]);
    },
    i(n) {
      r || (w(a, n), r = !0);
    },
    o(n) {
      C(a, n), r = !1;
    },
    d(n) {
      n && M(o), a && a.d(n);
    }
  };
}
function io(l, e, o) {
  let { $$slots: r = {}, $$scope: a } = e, { className: n } = e, { href: s = void 0 } = e, { as: f = "a" } = e, { handleClick: d = void 0 } = e, { handleKeyup: p = void 0 } = e;
  return l.$$set = (h) => {
    "className" in h && o(0, n = h.className), "href" in h && o(1, s = h.href), "as" in h && o(2, f = h.as), "handleClick" in h && o(3, d = h.handleClick), "handleKeyup" in h && o(4, p = h.handleKeyup), "$$scope" in h && o(5, a = h.$$scope);
  }, [n, s, f, d, p, a, r];
}
class Cr extends j {
  constructor(e) {
    super(), W(this, e, io, no, U, {
      className: 0,
      href: 1,
      as: 2,
      handleClick: 3,
      handleKeyup: 4
    });
  }
}
function so(l) {
  let e;
  const o = (
    /*#slots*/
    l[7].default
  ), r = be(
    o,
    l,
    /*$$scope*/
    l[8],
    null
  );
  return {
    c() {
      r && r.c();
    },
    m(a, n) {
      r && r.m(a, n), e = !0;
    },
    p(a, n) {
      r && r.p && (!e || n & /*$$scope*/
      256) && Ce(
        r,
        o,
        a,
        /*$$scope*/
        a[8],
        e ? we(
          o,
          /*$$scope*/
          a[8],
          n,
          null
        ) : ve(
          /*$$scope*/
          a[8]
        ),
        null
      );
    },
    i(a) {
      e || (w(r, a), e = !0);
    },
    o(a) {
      C(r, a), e = !1;
    },
    d(a) {
      r && r.d(a);
    }
  };
}
function co(l) {
  let e, o;
  return e = new Cr({
    props: {
      handleClick: (
        /*handleClick*/
        l[3]
      ),
      handleKeyup: (
        /*handleKeyup*/
        l[4]
      ),
      as: (
        /*as*/
        l[2]
      ),
      href: (
        /*href*/
        l[1]
      ),
      className: `
        ${/*className*/
      l[0]} ${/*height*/
      l[5]} px-4
        bg-button-secondary text-primary hover:bg-button-secondary-hover flex
        min-w-[140px] text-b3 rounded-3xl justify-center items-center
    `,
      $$slots: { default: [so] },
      $$scope: { ctx: l }
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, [a]) {
      const n = {};
      a & /*handleClick*/
      8 && (n.handleClick = /*handleClick*/
      r[3]), a & /*handleKeyup*/
      16 && (n.handleKeyup = /*handleKeyup*/
      r[4]), a & /*as*/
      4 && (n.as = /*as*/
      r[2]), a & /*href*/
      2 && (n.href = /*href*/
      r[1]), a & /*className*/
      1 && (n.className = `
        ${/*className*/
      r[0]} ${/*height*/
      r[5]} px-4
        bg-button-secondary text-primary hover:bg-button-secondary-hover flex
        min-w-[140px] text-b3 rounded-3xl justify-center items-center
    `), a & /*$$scope*/
      256 && (n.$$scope = { dirty: a, ctx: r }), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function fo(l, e, o) {
  let { $$slots: r = {}, $$scope: a } = e, { className: n = "" } = e, { href: s = void 0 } = e, { as: f = "button" } = e, { isLarge: d = !1 } = e;
  const p = d ? "min-h-12" : "min-h-8";
  let { handleClick: h = (g) => console.log("click", g) } = e, { handleKeyup: u = (g) => console.log("keyup", g) } = e;
  return l.$$set = (g) => {
    "className" in g && o(0, n = g.className), "href" in g && o(1, s = g.href), "as" in g && o(2, f = g.as), "isLarge" in g && o(6, d = g.isLarge), "handleClick" in g && o(3, h = g.handleClick), "handleKeyup" in g && o(4, u = g.handleKeyup), "$$scope" in g && o(8, a = g.$$scope);
  }, [n, s, f, h, u, p, d, r, a];
}
class vr extends j {
  constructor(e) {
    super(), W(this, e, fo, co, U, {
      className: 0,
      href: 1,
      as: 2,
      isLarge: 6,
      handleClick: 3,
      handleKeyup: 4
    });
  }
}
function uo(l) {
  let e, o;
  return {
    c() {
      e = c("svg"), o = c("path"), t(o, "d", "M15.25 3.50011V6.50011C15.25 6.69902 15.171 6.88979 15.0303 7.03044C14.8897 7.17109 14.6989 7.25011 14.5 7.25011H11.5C11.3011 7.25011 11.1103 7.17109 10.9697 7.03044C10.829 6.88979 10.75 6.69902 10.75 6.50011C10.75 6.3012 10.829 6.11043 10.9697 5.96978C11.1103 5.82913 11.3011 5.75011 11.5 5.75011H12.5687L11.3812 4.66386C11.3731 4.65636 11.365 4.64886 11.3575 4.64074C10.8069 4.09046 10.1295 3.68394 9.38483 3.45695C8.64022 3.22996 7.85122 3.18945 7.08728 3.33899C6.32333 3.48853 5.60784 3.82353 5.00377 4.31452C4.39969 4.80552 3.92555 5.43745 3.62305 6.15472C3.32055 6.87199 3.19898 7.65261 3.26902 8.4279C3.33906 9.20319 3.59857 9.94938 4.02471 10.6008C4.45086 11.2523 5.03058 11.789 5.71287 12.1638C6.39516 12.5386 7.1591 12.7399 7.93748 12.7501H7.99998C9.21243 12.753 10.3795 12.2893 11.2594 11.4551C11.404 11.3184 11.597 11.2448 11.7959 11.2504C11.9948 11.2561 12.1833 11.3405 12.32 11.4851C12.4566 11.6297 12.5303 11.8227 12.5246 12.0216C12.519 12.2205 12.4346 12.4091 12.29 12.5457C11.1318 13.6435 9.59575 14.2537 7.99998 14.2501H7.91436C6.89126 14.2357 5.88735 13.9703 4.99078 13.4773C4.09421 12.9842 3.3324 12.2786 2.77224 11.4224C2.21208 10.5661 1.87071 9.58543 1.7781 8.56644C1.68548 7.54744 1.84447 6.52129 2.24108 5.57809C2.63769 4.63489 3.25981 3.80349 4.05281 3.15688C4.8458 2.51028 5.78541 2.06826 6.78915 1.86963C7.79288 1.671 8.83003 1.72184 9.80952 2.01768C10.789 2.31352 11.6809 2.84531 12.4069 3.56636L13.75 4.79511V3.50011C13.75 3.3012 13.829 3.11043 13.9697 2.96978C14.1103 2.82913 14.3011 2.75011 14.5 2.75011C14.6989 2.75011 14.8897 2.82913 15.0303 2.96978C15.171 3.11043 15.25 3.3012 15.25 3.50011Z"), t(o, "fill", "currentColor"), t(e, "xmlns", "http://www.w3.org/2000/svg"), t(e, "width", "16"), t(e, "height", "16"), t(e, "viewBox", "0 0 16 16"), t(e, "fill", "none");
    },
    m(r, a) {
      $(r, e, a), i(e, o);
    },
    p: B,
    i: B,
    o: B,
    d(r) {
      r && M(e);
    }
  };
}
class po extends j {
  constructor(e) {
    super(), W(this, e, null, uo, U, {});
  }
}
function ho(l) {
  let e, o, r, a, n;
  return o = new po({}), {
    c() {
      e = x("div"), A(o.$$.fragment), r = X(" Retry connecting to "), a = X(
        /*walletName*/
        l[0]
      ), t(e, "class", "flex gap-2");
    },
    m(s, f) {
      $(s, e, f), N(o, e, null), i(e, r), i(e, a), n = !0;
    },
    p(s, f) {
      (!n || f & /*walletName*/
      1) && ge(
        a,
        /*walletName*/
        s[0]
      );
    },
    i(s) {
      n || (w(o.$$.fragment, s), n = !0);
    },
    o(s) {
      C(o.$$.fragment, s), n = !1;
    },
    d(s) {
      s && M(e), Z(o);
    }
  };
}
function er(l) {
  let e, o;
  return e = new hr({
    props: { handleClick: (
      /*handleFallback*/
      l[3]
    ) }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, a) {
      const n = {};
      a & /*handleFallback*/
      8 && (n.handleClick = /*handleFallback*/
      r[3]), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function go(l) {
  let e, o, r, a, n, s, f, d, p, h;
  a = new wr({}), d = new vr({
    props: {
      handleClick: (
        /*handleCallback*/
        l[2]
      ),
      handleKeyup: (
        /*func*/
        l[4]
      ),
      isLarge: !0,
      $$slots: { default: [ho] },
      $$scope: { ctx: l }
    }
  });
  let u = (
    /*showFallback*/
    l[1] && er(l)
  );
  return {
    c() {
      e = x("section"), o = x("div"), r = x("div"), A(a.$$.fragment), n = K(), s = x("div"), s.innerHTML = '<h3 class="text-primary text-h4 font-bold">Couldn&#39;t connect</h3> <p class="text-primary text-p3 font-[400]">Please try connecting again.</p>', f = K(), A(d.$$.fragment), p = K(), u && u.c(), t(r, "class", "bg-button-secondary rounded-full p-5"), t(o, "class", "flex flex-col h-full justify-center items-center gap-4 w-full flex-grow"), t(e, "class", "flex flex-col justify-center items-center flex-grow");
    },
    m(g, m) {
      $(g, e, m), i(e, o), i(o, r), N(a, r, null), i(o, n), i(o, s), i(o, f), N(d, o, null), i(e, p), u && u.m(e, null), h = !0;
    },
    p(g, [m]) {
      const b = {};
      m & /*handleCallback*/
      4 && (b.handleClick = /*handleCallback*/
      g[2]), m & /*handleCallback*/
      4 && (b.handleKeyup = /*func*/
      g[4]), m & /*$$scope, walletName*/
      33 && (b.$$scope = { dirty: m, ctx: g }), d.$set(b), /*showFallback*/
      g[1] ? u ? (u.p(g, m), m & /*showFallback*/
      2 && w(u, 1)) : (u = er(g), u.c(), w(u, 1), u.m(e, null)) : u && (ce(), C(u, 1, 1, () => {
        u = null;
      }), fe());
    },
    i(g) {
      h || (w(a.$$.fragment, g), w(d.$$.fragment, g), w(u), h = !0);
    },
    o(g) {
      C(a.$$.fragment, g), C(d.$$.fragment, g), C(u), h = !1;
    },
    d(g) {
      g && M(e), Z(a), Z(d), u && u.d();
    }
  };
}
function mo(l, e, o) {
  let { walletName: r = "" } = e, { showFallback: a = !1 } = e, { handleCallback: n = async () => {
  } } = e, { handleFallback: s = async () => {
  } } = e;
  const f = (d) => {
    d.key === "Enter" && n();
  };
  return l.$$set = (d) => {
    "walletName" in d && o(0, r = d.walletName), "showFallback" in d && o(1, a = d.showFallback), "handleCallback" in d && o(2, n = d.handleCallback), "handleFallback" in d && o(3, s = d.handleFallback);
  }, [r, a, n, s, f];
}
class bo extends j {
  constructor(e) {
    super(), W(this, e, mo, go, U, {
      walletName: 0,
      showFallback: 1,
      handleCallback: 2,
      handleFallback: 3
    });
  }
}
function wo(l) {
  let e, o, r;
  return {
    c() {
      e = c("svg"), o = c("path"), r = c("path"), t(o, "d", "M28.7868 15.1583C29.5186 13.9874 29.1627 12.445 27.9918 11.7133C26.821 10.9815 25.2786 11.3374 24.5468 12.5083L17.656 23.5335L13.1668 20.1666C12.0622 19.3382 10.4952 19.562 9.66681 20.6666C8.83839 21.7712 9.06224 23.3382 10.1668 24.1666L16.8335 29.1666C17.3934 29.5865 18.1037 29.7525 18.7917 29.6242C19.4798 29.4959 20.0825 29.0851 20.4535 28.4916L28.7868 15.1583Z"), t(o, "fill", "#02A17A"), t(r, "fill-rule", "evenodd"), t(r, "clip-rule", "evenodd"), t(r, "d", "M40 20.5C40 31.5457 31.0457 40.5 20 40.5C8.9543 40.5 0 31.5457 0 20.5C0 9.4543 8.9543 0.5 20 0.5C31.0457 0.5 40 9.4543 40 20.5ZM36.6667 20.5C36.6667 29.7047 29.2047 37.1667 20 37.1667C10.7953 37.1667 3.33333 29.7047 3.33333 20.5C3.33333 11.2953 10.7953 3.83333 20 3.83333C29.2047 3.83333 36.6667 11.2953 36.6667 20.5Z"), t(r, "fill", "#02A17A"), t(e, "xmlns", "http://www.w3.org/2000/svg"), t(e, "width", "40"), t(e, "height", "41"), t(e, "viewBox", "0 0 40 41"), t(e, "fill", "none");
    },
    m(a, n) {
      $(a, e, n), i(e, o), i(e, r);
    },
    p: B,
    i: B,
    o: B,
    d(a) {
      a && M(e);
    }
  };
}
class Co extends j {
  constructor(e) {
    super(), W(this, e, null, wo, U, {});
  }
}
function vo(l) {
  let e, o, r, a, n, s;
  return r = new Co({}), {
    c() {
      e = x("section"), o = x("div"), A(r.$$.fragment), a = K(), n = x("h3"), n.textContent = "Success!", t(o, "class", "bg-button-secondary rounded-full p-5"), t(n, "class", "text-primary text-h4 font-bold"), t(e, "class", "flex flex-col justify-center items-center gap-4 flex-grow");
    },
    m(f, d) {
      $(f, e, d), i(e, o), N(r, o, null), i(e, a), i(e, n), s = !0;
    },
    p: B,
    i(f) {
      s || (w(r.$$.fragment, f), s = !0);
    },
    o(f) {
      C(r.$$.fragment, f), s = !1;
    },
    d(f) {
      f && M(e), Z(r);
    }
  };
}
class ko extends j {
  constructor(e) {
    super(), W(this, e, null, vo, U, {});
  }
}
function _o(l) {
  let e, o, r, a, n, s, f, d, p, h, u, g, m, b, v, F, O, z, L, k, y, S, G, _, E, R, D, T, q, I, P, V, Y;
  return {
    c() {
      e = c("svg"), o = c("path"), r = c("path"), a = c("path"), n = c("path"), s = c("path"), f = c("path"), d = c("defs"), p = c("linearGradient"), h = c("stop"), u = c("stop"), g = c("radialGradient"), m = c("stop"), b = c("stop"), v = c("stop"), F = c("linearGradient"), O = c("stop"), z = c("stop"), L = c("stop"), k = c("stop"), y = c("radialGradient"), S = c("stop"), G = c("stop"), _ = c("stop"), E = c("radialGradient"), R = c("stop"), D = c("stop"), T = c("stop"), q = c("stop"), I = c("stop"), P = c("radialGradient"), V = c("stop"), Y = c("stop"), t(o, "d", "M18.9432 15.0009C18.6742 15.1415 18.3969 15.2653 18.1127 15.3717C17.208 15.7099 16.2497 15.8822 15.2839 15.8803C11.5552 15.8803 8.30719 13.3178 8.30719 10.0294C8.31196 9.58835 8.43446 9.15656 8.66205 8.77861C8.88963 8.40066 9.21405 8.09024 9.60182 7.87939C6.22931 8.0211 5.36255 11.5323 5.36255 13.5894C5.36255 19.4056 10.7278 19.9953 11.8838 19.9953C12.5071 19.9953 13.4471 19.8142 14.0113 19.6363L14.1145 19.6017C16.2856 18.8518 18.1364 17.3858 19.3624 15.4449C19.3999 15.3858 19.4167 15.3159 19.41 15.2462C19.4034 15.1765 19.3736 15.111 19.3256 15.0601C19.2775 15.0092 19.2138 14.9757 19.1445 14.965C19.0753 14.9543 19.0044 14.9669 18.9432 15.0009Z"), t(o, "fill", "url(#paint0_linear_6297_45099)"), t(r, "opacity", "0.35"), t(r, "d", "M18.9432 15.0009C18.6742 15.1415 18.3969 15.2653 18.1127 15.3717C17.208 15.7099 16.2497 15.8822 15.2839 15.8803C11.5552 15.8803 8.30719 13.3178 8.30719 10.0294C8.31196 9.58835 8.43446 9.15656 8.66205 8.77861C8.88963 8.40066 9.21405 8.09024 9.60182 7.87939C6.22931 8.0211 5.36255 11.5323 5.36255 13.5894C5.36255 19.4056 10.7278 19.9953 11.8838 19.9953C12.5071 19.9953 13.4471 19.8142 14.0113 19.6363L14.1145 19.6017C16.2856 18.8518 18.1364 17.3858 19.3624 15.4449C19.3999 15.3858 19.4167 15.3159 19.41 15.2462C19.4034 15.1765 19.3736 15.111 19.3256 15.0601C19.2775 15.0092 19.2138 14.9757 19.1445 14.965C19.0753 14.9543 19.0044 14.9669 18.9432 15.0009Z"), t(r, "fill", "url(#paint1_radial_6297_45099)"), t(a, "d", "M9.06695 19.0054C8.36414 18.5696 7.75505 17.9985 7.27511 17.3254C6.7281 16.5766 6.35167 15.7174 6.17224 14.8078C5.99281 13.8983 6.01472 12.9606 6.23643 12.0605C6.45815 11.1603 6.8743 10.3195 7.45569 9.59705C8.03708 8.87459 8.76964 8.28797 9.60198 7.87832C9.84783 7.7626 10.2678 7.55319 10.8265 7.56342C11.2201 7.56628 11.6077 7.66017 11.9589 7.83773C12.3101 8.01529 12.6154 8.2717 12.8508 8.58685C13.169 9.01133 13.3444 9.5256 13.3519 10.0559C13.3519 10.0393 15.2793 3.78931 7.04818 3.78931C3.589 3.78931 0.74443 7.06902 0.74443 9.94645C0.730785 11.4686 1.05676 12.9746 1.69866 14.3551C2.7477 16.5914 4.5822 18.3649 6.85378 19.3388C9.12536 20.3128 11.6758 20.4194 14.0209 19.6384C13.1998 19.897 12.3323 19.9747 11.4783 19.8659C10.6243 19.7572 9.80398 19.4647 9.07405 19.0086L9.06695 19.0054Z"), t(a, "fill", "url(#paint2_linear_6297_45099)"), t(n, "opacity", "0.41"), t(n, "d", "M9.06695 19.0054C8.36414 18.5696 7.75505 17.9985 7.27511 17.3254C6.7281 16.5766 6.35167 15.7174 6.17224 14.8078C5.99281 13.8983 6.01472 12.9606 6.23643 12.0605C6.45815 11.1603 6.8743 10.3195 7.45569 9.59705C8.03708 8.87459 8.76964 8.28797 9.60198 7.87832C9.84783 7.7626 10.2678 7.55319 10.8265 7.56342C11.2201 7.56628 11.6077 7.66017 11.9589 7.83773C12.3101 8.01529 12.6154 8.2717 12.8508 8.58685C13.169 9.01133 13.3444 9.5256 13.3519 10.0559C13.3519 10.0393 15.2793 3.78931 7.04818 3.78931C3.589 3.78931 0.74443 7.06902 0.74443 9.94645C0.730785 11.4686 1.05676 12.9746 1.69866 14.3551C2.7477 16.5914 4.5822 18.3649 6.85378 19.3388C9.12536 20.3128 11.6758 20.4194 14.0209 19.6384C13.1998 19.897 12.3323 19.9747 11.4783 19.8659C10.6243 19.7572 9.80398 19.4647 9.07405 19.0086L9.06695 19.0054Z"), t(n, "fill", "url(#paint3_radial_6297_45099)"), t(s, "d", "M12.7387 11.7191C12.6749 11.8018 12.4787 11.9159 12.4787 12.1647C12.4787 12.3702 12.6126 12.5678 12.8506 12.7339C13.9837 13.5211 16.1199 13.4172 16.1254 13.4172C16.965 13.4152 17.7887 13.1882 18.5106 12.7599C19.2388 12.335 19.8432 11.7273 20.2639 10.9971C20.6845 10.2668 20.9067 9.43939 20.9084 8.59685C20.9289 6.83261 20.278 5.6596 20.0148 5.14001C18.3451 1.87682 14.741 4.93329e-08 10.8216 4.93329e-08C8.17102 -0.000261808 5.62686 1.04193 3.7394 2.90118C1.85193 4.76042 0.772869 7.28727 0.735596 9.93519C0.773418 7.05855 3.63532 4.73535 7.03935 4.73535C7.31513 4.73535 8.88792 4.76212 10.3488 5.52812C11.6364 6.20359 12.3109 7.01919 12.7797 7.8277C13.2667 8.66771 13.3533 9.72893 13.3533 10.1517C13.3533 10.5744 13.1374 11.2011 12.7387 11.7191Z"), t(s, "fill", "url(#paint4_radial_6297_45099)"), t(f, "d", "M12.7387 11.7191C12.6749 11.8018 12.4787 11.9159 12.4787 12.1647C12.4787 12.3702 12.6126 12.5678 12.8506 12.7339C13.9837 13.5211 16.1199 13.4172 16.1254 13.4172C16.965 13.4152 17.7887 13.1882 18.5106 12.7599C19.2388 12.335 19.8432 11.7273 20.2639 10.9971C20.6845 10.2668 20.9067 9.43939 20.9084 8.59685C20.9289 6.83261 20.278 5.6596 20.0148 5.14001C18.3451 1.87682 14.741 4.93329e-08 10.8216 4.93329e-08C8.17102 -0.000261808 5.62686 1.04193 3.7394 2.90118C1.85193 4.76042 0.772869 7.28727 0.735596 9.93519C0.773418 7.05855 3.63532 4.73535 7.03935 4.73535C7.31513 4.73535 8.88792 4.76212 10.3488 5.52812C11.6364 6.20359 12.3109 7.01919 12.7797 7.8277C13.2667 8.66771 13.3533 9.72893 13.3533 10.1517C13.3533 10.5744 13.1374 11.2011 12.7387 11.7191Z"), t(f, "fill", "url(#paint5_radial_6297_45099)"), t(h, "stop-color", "#0C59A4"), t(u, "offset", "1"), t(u, "stop-color", "#114A8B"), t(p, "id", "paint0_linear_6297_45099"), t(p, "x1", "5.36255"), t(p, "y1", "13.9389"), t(p, "x2", "19.4152"), t(p, "y2", "13.9389"), t(p, "gradientUnits", "userSpaceOnUse"), t(m, "offset", "0.72"), t(m, "stop-opacity", "0"), t(b, "offset", "0.95"), t(b, "stop-opacity", "0.53"), t(v, "offset", "1"), t(g, "id", "paint1_radial_6297_45099"), t(g, "cx", "0"), t(g, "cy", "0"), t(g, "r", "1"), t(g, "gradientUnits", "userSpaceOnUse"), t(g, "gradientTransform", "translate(13.124 14.0496) scale(7.51564 7.13342)"), t(O, "stop-color", "#1B9DE2"), t(z, "offset", "0.16"), t(z, "stop-color", "#1595DF"), t(L, "offset", "0.67"), t(L, "stop-color", "#0680D7"), t(k, "offset", "1"), t(k, "stop-color", "#0078D4"), t(F, "id", "paint2_linear_6297_45099"), t(F, "x1", "12.7712"), t(F, "y1", "7.84762"), t(F, "x2", "4.00261"), t(F, "y2", "17.4073"), t(F, "gradientUnits", "userSpaceOnUse"), t(S, "offset", "0.76"), t(S, "stop-opacity", "0"), t(G, "offset", "0.95"), t(G, "stop-opacity", "0.5"), t(_, "offset", "1"), t(y, "id", "paint3_radial_6297_45099"), t(y, "cx", "0"), t(y, "cy", "0"), t(y, "r", "1"), t(y, "gradientUnits", "userSpaceOnUse"), t(y, "gradientTransform", "translate(6.29839 15.6665) rotate(-81.3767) scale(11.2916 9.13031)"), t(R, "stop-color", "#35C1F1"), t(D, "offset", "0.11"), t(D, "stop-color", "#34C1ED"), t(T, "offset", "0.23"), t(T, "stop-color", "#2FC2DF"), t(q, "offset", "0.31"), t(q, "stop-color", "#2BC3D2"), t(I, "offset", "0.67"), t(I, "stop-color", "#36C752"), t(E, "id", "paint4_radial_6297_45099"), t(E, "cx", "0"), t(E, "cy", "0"), t(E, "r", "1"), t(E, "gradientUnits", "userSpaceOnUse"), t(E, "gradientTransform", "translate(2.77348 3.72978) rotate(92.2927) scale(15.9492 33.9991)"), t(V, "stop-color", "#66EB6E"), t(Y, "offset", "1"), t(Y, "stop-color", "#66EB6E"), t(Y, "stop-opacity", "0"), t(P, "id", "paint5_radial_6297_45099"), t(P, "cx", "0"), t(P, "cy", "0"), t(P, "r", "1"), t(P, "gradientUnits", "userSpaceOnUse"), t(P, "gradientTransform", "translate(19.6525 6.09385) rotate(73.7259) scale(7.66371 6.23687)"), t(e, "xmlns", "http://www.w3.org/2000/svg"), t(
        e,
        "width",
        /*size*/
        l[0]
      ), t(
        e,
        "height",
        /*size*/
        l[0]
      ), t(e, "viewBox", "0 0 21 21"), t(e, "fill", "none");
    },
    m(H, J) {
      $(H, e, J), i(e, o), i(e, r), i(e, a), i(e, n), i(e, s), i(e, f), i(e, d), i(d, p), i(p, h), i(p, u), i(d, g), i(g, m), i(g, b), i(g, v), i(d, F), i(F, O), i(F, z), i(F, L), i(F, k), i(d, y), i(y, S), i(y, G), i(y, _), i(d, E), i(E, R), i(E, D), i(E, T), i(E, q), i(E, I), i(d, P), i(P, V), i(P, Y);
    },
    p(H, [J]) {
      J & /*size*/
      1 && t(
        e,
        "width",
        /*size*/
        H[0]
      ), J & /*size*/
      1 && t(
        e,
        "height",
        /*size*/
        H[0]
      );
    },
    i: B,
    o: B,
    d(H) {
      H && M(e);
    }
  };
}
function yo(l, e, o) {
  let { size: r = "21" } = e;
  return l.$$set = (a) => {
    "size" in a && o(0, r = a.size);
  }, [r];
}
class xo extends j {
  constructor(e) {
    super(), W(this, e, yo, _o, U, { size: 0 });
  }
}
function Fo(l) {
  let e, o, r, a, n, s, f, d, p, h, u, g, m, b, v, F, O, z, L, k, y, S, G, _, E, R, D, T, q, I, P, V, Y, H, J, oe, ee, le, ae, te, Fe, De, Ve, Re, ue, Te, He, Ke, Pe, de, Ne, Ze, Ae, ze, ne, Ft, qe, Xe, Qe, Ye, Je, ie, et, tt, rt, ot, lt, re, Lt, at, nt, it, st, ct, ft, ut, se, dt, pt, ht, gt, mt, pe, bt, wt, Ct, vt, he, Oe, Ue, Ge, Ie;
  return {
    c() {
      e = c("svg"), o = c("path"), r = c("path"), a = c("path"), n = c("path"), s = c("path"), f = c("path"), d = c("path"), p = c("path"), h = c("path"), u = c("path"), g = c("path"), m = c("path"), b = c("defs"), v = c("linearGradient"), F = c("stop"), O = c("stop"), z = c("stop"), L = c("stop"), k = c("stop"), y = c("stop"), S = c("stop"), G = c("stop"), _ = c("radialGradient"), E = c("stop"), R = c("stop"), D = c("stop"), T = c("stop"), q = c("stop"), I = c("stop"), P = c("stop"), V = c("stop"), Y = c("stop"), H = c("radialGradient"), J = c("stop"), oe = c("stop"), ee = c("stop"), le = c("stop"), ae = c("stop"), te = c("radialGradient"), Fe = c("stop"), De = c("stop"), Ve = c("stop"), Re = c("stop"), ue = c("radialGradient"), Te = c("stop"), He = c("stop"), Ke = c("stop"), Pe = c("stop"), de = c("radialGradient"), Ne = c("stop"), Ze = c("stop"), Ae = c("stop"), ze = c("stop"), ne = c("radialGradient"), Ft = c("stop"), qe = c("stop"), Xe = c("stop"), Qe = c("stop"), Ye = c("stop"), Je = c("stop"), ie = c("radialGradient"), et = c("stop"), tt = c("stop"), rt = c("stop"), ot = c("stop"), lt = c("stop"), re = c("radialGradient"), Lt = c("stop"), at = c("stop"), nt = c("stop"), it = c("stop"), st = c("stop"), ct = c("stop"), ft = c("stop"), ut = c("stop"), se = c("radialGradient"), dt = c("stop"), pt = c("stop"), ht = c("stop"), gt = c("stop"), mt = c("stop"), pe = c("radialGradient"), bt = c("stop"), wt = c("stop"), Ct = c("stop"), vt = c("stop"), he = c("linearGradient"), Oe = c("stop"), Ue = c("stop"), Ge = c("stop"), Ie = c("stop"), t(o, "d", "M19.0292 6.76185C18.6047 5.74052 17.7437 4.63778 17.0694 4.28919C17.5504 5.22056 17.8841 6.22082 18.0588 7.25436L18.0605 7.27075C16.956 4.51705 15.083 3.40675 13.5533 0.989041C13.4759 0.866794 13.3985 0.744296 13.3232 0.614992C13.2801 0.54114 13.2455 0.474597 13.2155 0.413348C13.1521 0.290485 13.1032 0.160639 13.0698 0.0264442C13.0699 0.0202518 13.0677 0.0142596 13.0636 0.0096107C13.0595 0.00496183 13.0538 0.00198167 13.0477 0.00123872C13.0416 -0.000412908 13.0353 -0.000412908 13.0293 0.00123872C13.028 0.00123872 13.026 0.00350722 13.0245 0.00401132C13.023 0.00451543 13.0197 0.00678392 13.0174 0.00779214L13.0212 0.00123872C10.5674 1.43795 9.73491 4.09763 9.65828 5.42798C8.67837 5.49533 7.7415 5.85643 6.96987 6.46417C6.88936 6.39598 6.80522 6.3322 6.71781 6.27311C6.49522 5.494 6.48573 4.66943 6.69034 3.8854C5.78923 4.32194 4.98854 4.94076 4.33892 5.70271H4.33439C3.94723 5.21196 3.97445 3.59377 3.99663 3.25577C3.8821 3.3018 3.77277 3.35984 3.67047 3.42893C3.3287 3.67286 3.00919 3.94657 2.71569 4.24685C2.38124 4.58594 2.07578 4.95245 1.8025 5.34253V5.34404V5.34228C1.17444 6.23267 0.728971 7.23866 0.491813 8.30216L0.478706 8.36668C0.460306 8.45263 0.394016 8.88314 0.382674 8.97665C0.382674 8.98396 0.381161 8.99077 0.380405 8.99808C0.294893 9.44254 0.241905 9.89264 0.221863 10.3448L0.221863 10.3952C0.224005 12.8412 1.14363 15.1974 2.799 16.9982C4.45436 18.799 6.72502 19.9132 9.16221 20.1208C11.5994 20.3284 14.0258 19.6141 15.9618 18.1193C17.8979 16.6244 19.2027 14.4576 19.6185 12.0472C19.6349 11.9212 19.6482 11.7964 19.6628 11.6691C19.864 10.0061 19.6462 8.31919 19.0292 6.76185ZM7.76989 14.4087C7.81551 14.4306 7.85836 14.4543 7.90524 14.4752L7.91205 14.4795C7.86466 14.4568 7.81728 14.4332 7.76989 14.4087ZM18.0613 7.27302V7.26369L18.063 7.27402L18.0613 7.27302Z"), t(o, "fill", "url(#paint0_linear_6297_45086)"), t(r, "d", "M19.029 6.76123C18.6045 5.7399 17.7435 4.63717 17.0693 4.28857C17.5502 5.21995 17.8839 6.2202 18.0586 7.25374V7.26307L18.0603 7.2734C18.8133 9.42808 18.7044 11.7904 17.7564 13.8667C16.6367 16.269 13.9266 18.7313 9.68457 18.6116C5.10095 18.4818 1.0643 15.0805 0.309647 10.6257C0.172277 9.92326 0.309647 9.5671 0.37871 8.9962C0.284499 9.43966 0.231927 9.89095 0.22168 10.3442L0.22168 10.3946C0.223821 12.8406 1.14345 15.1968 2.79882 16.9976C4.45418 18.7983 6.72484 19.9126 9.16203 20.1202C11.5992 20.3278 14.0256 19.6135 15.9616 18.1186C17.8977 16.6237 19.2025 14.457 19.6183 12.0466C19.6347 11.9205 19.648 11.7958 19.6627 11.6685C19.8639 10.0055 19.646 8.31857 19.029 6.76123Z"), t(r, "fill", "url(#paint1_radial_6297_45086)"), t(a, "d", "M19.029 6.76123C18.6045 5.7399 17.7435 4.63717 17.0693 4.28857C17.5502 5.21995 17.8839 6.2202 18.0586 7.25374V7.26307L18.0603 7.2734C18.8133 9.42808 18.7044 11.7904 17.7564 13.8667C16.6367 16.269 13.9266 18.7313 9.68457 18.6116C5.10095 18.4818 1.0643 15.0805 0.309647 10.6257C0.172277 9.92326 0.309647 9.5671 0.37871 8.9962C0.284499 9.43966 0.231927 9.89095 0.22168 10.3442L0.22168 10.3946C0.223821 12.8406 1.14345 15.1968 2.79882 16.9976C4.45418 18.7983 6.72484 19.9126 9.16203 20.1202C11.5992 20.3278 14.0256 19.6135 15.9616 18.1186C17.8977 16.6237 19.2025 14.457 19.6183 12.0466C19.6347 11.9205 19.648 11.7958 19.6627 11.6685C19.8639 10.0055 19.646 8.31857 19.029 6.76123Z"), t(a, "fill", "url(#paint2_radial_6297_45086)"), t(n, "d", "M14.2817 7.91068C14.3029 7.92555 14.3225 7.94042 14.3425 7.95529C14.0971 7.52017 13.7916 7.12179 13.4351 6.7719C10.3988 3.73514 12.6393 0.187473 13.0172 0.00725419L13.0209 0.00170898C10.5672 1.43842 9.73464 4.0981 9.65802 5.42844C9.77195 5.42063 9.88487 5.41105 10.0013 5.41105C10.872 5.41265 11.7267 5.64477 12.4785 6.08382C13.2303 6.52287 13.8525 7.15319 14.2817 7.91068Z"), t(n, "fill", "url(#paint3_radial_6297_45086)"), t(s, "d", "M10.0075 8.5172C9.9914 8.76018 9.1329 9.59826 8.8327 9.59826C6.05481 9.59826 5.60388 11.2787 5.60388 11.2787C5.72688 12.6937 6.71292 13.8592 7.90489 14.4758C7.95933 14.504 8.01453 14.5295 8.06973 14.5544C8.16535 14.5968 8.26096 14.6359 8.35657 14.6719C8.76565 14.8166 9.19411 14.8993 9.62768 14.9171C14.4966 15.1455 15.4398 9.09466 11.9262 7.33885C12.7551 7.23099 13.5945 7.43481 14.2816 7.91076C13.8524 7.15327 13.2302 6.52295 12.4784 6.0839C11.7266 5.64485 10.8719 5.41273 10.0012 5.41113C9.88528 5.41113 9.77186 5.42071 9.65793 5.42852C8.67802 5.49588 7.74115 5.85698 6.96951 6.46472C7.11848 6.59075 7.2866 6.75912 7.64074 7.10822C8.30364 7.76129 10.0037 8.4378 10.0075 8.5172Z"), t(s, "fill", "url(#paint4_radial_6297_45086)"), t(f, "d", "M10.0075 8.5172C9.9914 8.76018 9.1329 9.59826 8.8327 9.59826C6.05481 9.59826 5.60388 11.2787 5.60388 11.2787C5.72688 12.6937 6.71292 13.8592 7.90489 14.4758C7.95933 14.504 8.01453 14.5295 8.06973 14.5544C8.16535 14.5968 8.26096 14.6359 8.35657 14.6719C8.76565 14.8166 9.19411 14.8993 9.62768 14.9171C14.4966 15.1455 15.4398 9.09466 11.9262 7.33885C12.7551 7.23099 13.5945 7.43481 14.2816 7.91076C13.8524 7.15327 13.2302 6.52295 12.4784 6.0839C11.7266 5.64485 10.8719 5.41273 10.0012 5.41113C9.88528 5.41113 9.77186 5.42071 9.65793 5.42852C8.67802 5.49588 7.74115 5.85698 6.96951 6.46472C7.11848 6.59075 7.2866 6.75912 7.64074 7.10822C8.30364 7.76129 10.0037 8.4378 10.0075 8.5172Z"), t(f, "fill", "url(#paint5_radial_6297_45086)"), t(d, "d", "M6.51447 6.1401C6.59361 6.19051 6.65889 6.23437 6.71611 6.27394C6.49352 5.49483 6.48403 4.67026 6.68864 3.88623C5.78753 4.32277 4.98683 4.94159 4.33722 5.70354C4.38486 5.70228 5.80191 5.67683 6.51447 6.1401Z"), t(d, "fill", "url(#paint6_radial_6297_45086)"), t(p, "d", "M0.310452 10.6269C1.06435 15.0817 5.10176 18.4829 9.68537 18.6127C13.9275 18.7327 16.6375 16.2701 17.7572 13.8678C18.7052 11.7915 18.8141 9.4292 18.0611 7.27452V7.2652C18.0611 7.25789 18.0596 7.2536 18.0611 7.25587L18.0629 7.27225C18.4095 9.53495 17.2586 11.7271 15.4594 13.2094L15.4539 13.222C11.9483 16.077 8.59347 14.9445 7.91443 14.4823C7.86705 14.4596 7.81966 14.436 7.77228 14.4115C5.72837 13.4345 4.88398 11.5723 5.06496 9.97529C4.57978 9.98245 4.10294 9.84875 3.6922 9.59039C3.28147 9.33204 2.95448 8.96011 2.75084 8.51967C3.28737 8.19098 3.89913 8.00516 4.52783 7.97992C5.15653 7.95468 5.78122 8.09087 6.34237 8.3755C7.49926 8.90069 8.81572 8.95244 10.0103 8.51967C10.0065 8.44028 8.30638 7.76351 7.64348 7.11069C7.28934 6.76159 7.12122 6.59347 6.97226 6.46719C6.89175 6.399 6.80761 6.33522 6.7202 6.27614C6.66223 6.23656 6.59695 6.19371 6.51856 6.14229C5.806 5.67902 4.38895 5.70448 4.34207 5.70574H4.33753C3.95037 5.21499 3.97759 3.59679 3.99978 3.25879C3.88524 3.30482 3.77591 3.36286 3.67362 3.43195C3.33184 3.67588 3.01233 3.94959 2.71883 4.24987C2.38319 4.58801 2.07645 4.95367 1.80186 5.34303V5.34454V5.34278C1.1738 6.23317 0.728333 7.23916 0.491175 8.30265C0.486386 8.32257 0.139307 9.83994 0.310452 10.6269Z"), t(p, "fill", "url(#paint7_radial_6297_45086)"), t(h, "d", "M13.435 6.77143C13.7917 7.12171 14.0972 7.52052 14.3424 7.95608C14.3961 7.99666 14.4463 8.03699 14.4889 8.07606C16.7037 10.1177 15.5432 13.0037 15.4568 13.2092C17.2559 11.7268 18.4058 9.5347 18.0603 7.27201C16.9552 4.51705 15.0822 3.40675 13.5525 0.989041C13.4751 0.866794 13.3977 0.744296 13.3224 0.614992C13.2793 0.54114 13.2447 0.474597 13.2148 0.413348C13.1513 0.290485 13.1024 0.160639 13.0691 0.0264442C13.0691 0.0202518 13.0669 0.0142596 13.0628 0.0096107C13.0587 0.00496183 13.053 0.00198167 13.0469 0.00123872C13.0409 -0.000412908 13.0345 -0.000412908 13.0285 0.00123872C13.0272 0.00123872 13.0252 0.00350721 13.0237 0.00401132C13.0222 0.00451543 13.0189 0.00678392 13.0166 0.00779214C12.6388 0.187003 10.3983 3.73467 13.435 6.77143Z"), t(h, "fill", "url(#paint8_radial_6297_45086)"), t(u, "d", "M14.488 8.07354C14.4454 8.03447 14.3952 7.99414 14.3415 7.95356C14.3216 7.93869 14.302 7.92382 14.2808 7.90895C13.5936 7.433 12.7542 7.22919 11.9253 7.33704C15.439 9.09386 14.4963 15.1437 9.62685 14.9153C9.19327 14.8975 8.76481 14.8148 8.35573 14.6701C8.26012 14.6343 8.16451 14.5951 8.0689 14.5526C8.0137 14.5274 7.9585 14.5022 7.90405 14.474L7.91086 14.4782C8.58989 14.9418 11.9437 16.0743 15.4503 13.218L15.4559 13.2054C15.5433 13.0012 16.7038 10.1144 14.488 8.07354Z"), t(u, "fill", "url(#paint9_radial_6297_45086)"), t(g, "d", "M5.60413 11.2781C5.60413 11.2781 6.05506 9.59766 8.83295 9.59766C9.13315 9.59766 9.9924 8.75958 10.0078 8.51659C8.81323 8.94936 7.49677 8.89761 6.33988 8.37242C5.77873 8.08779 5.15404 7.9516 4.52534 7.97684C3.89664 8.00208 3.28488 8.1879 2.74835 8.51659C2.95199 8.95703 3.27898 9.32896 3.68971 9.58731C4.10044 9.84567 4.57729 9.97937 5.06247 9.97221C4.88149 11.5695 5.72587 13.4317 7.76978 14.4084C7.81541 14.4303 7.85825 14.454 7.90514 14.4749C6.71216 13.8586 5.72713 12.6931 5.60413 11.2781Z"), t(g, "fill", "url(#paint10_radial_6297_45086)"), t(m, "d", "M19.0292 6.76185C18.6047 5.74052 17.7437 4.63778 17.0694 4.28919C17.5504 5.22056 17.8841 6.22082 18.0588 7.25436L18.0605 7.27075C16.956 4.51705 15.083 3.40675 13.5533 0.989041C13.4759 0.866794 13.3985 0.744296 13.3232 0.614992C13.2801 0.54114 13.2455 0.474597 13.2155 0.413348C13.1521 0.290485 13.1032 0.160639 13.0698 0.0264442C13.0699 0.0202518 13.0677 0.0142596 13.0636 0.0096107C13.0595 0.00496183 13.0538 0.00198167 13.0477 0.00123872C13.0416 -0.000412908 13.0353 -0.000412908 13.0293 0.00123872C13.028 0.00123872 13.026 0.00350722 13.0245 0.00401132C13.023 0.00451543 13.0197 0.00678392 13.0174 0.00779214L13.0212 0.00123872C10.5674 1.43795 9.73491 4.09763 9.65828 5.42798C9.77221 5.42016 9.88513 5.41058 10.0016 5.41058C10.8722 5.41218 11.7269 5.6443 12.4788 6.08335C13.2306 6.5224 13.8527 7.15272 14.282 7.91021C13.5948 7.43426 12.7554 7.23044 11.9265 7.3383C15.4402 9.09512 14.4975 15.1449 9.62804 14.9166C9.19446 14.8988 8.766 14.8161 8.35693 14.6713C8.26131 14.6355 8.1657 14.5964 8.07009 14.5539C8.01489 14.5287 7.95969 14.5035 7.90524 14.4752L7.91205 14.4795C7.86466 14.4568 7.81728 14.4332 7.76989 14.4087C7.81551 14.4306 7.85836 14.4543 7.90524 14.4752C6.71227 13.8587 5.72724 12.6932 5.60424 11.2782C5.60424 11.2782 6.05516 9.59772 8.83306 9.59772C9.13325 9.59772 9.99251 8.75963 10.0079 8.51665C10.0041 8.43726 8.30399 7.76049 7.64109 7.10767C7.28695 6.75857 7.11883 6.59045 6.96987 6.46417C6.88936 6.39598 6.80522 6.3322 6.71781 6.27311C6.49522 5.494 6.48573 4.66943 6.69034 3.8854C5.78923 4.32194 4.98854 4.94076 4.33892 5.70271H4.33439C3.94723 5.21196 3.97445 3.59377 3.99663 3.25577C3.8821 3.3018 3.77277 3.35984 3.67047 3.42893C3.3287 3.67286 3.00919 3.94657 2.71569 4.24685C2.38124 4.58594 2.07578 4.95245 1.8025 5.34253V5.34404V5.34228C1.17444 6.23267 0.728971 7.23866 0.491813 8.30216L0.478706 8.36668C0.460306 8.45263 0.377885 8.88919 0.366038 8.98295C0.366038 8.97565 0.366038 8.99001 0.366038 8.98295C0.290651 9.43359 0.242505 9.88837 0.221863 10.3448L0.221863 10.3952C0.224005 12.8412 1.14363 15.1974 2.799 16.9982C4.45436 18.799 6.72502 19.9132 9.16221 20.1208C11.5994 20.3284 14.0258 19.6141 15.9618 18.1193C17.8979 16.6244 19.2027 14.4576 19.6185 12.0472C19.6349 11.9212 19.6482 11.7964 19.6628 11.6691C19.864 10.0061 19.6462 8.31919 19.0292 6.76185ZM18.06 7.26369L18.0618 7.27402L18.06 7.26369Z"), t(m, "fill", "url(#paint11_linear_6297_45086)"), t(F, "offset", "0.048"), t(F, "stop-color", "#FFF44F"), t(O, "offset", "0.111"), t(O, "stop-color", "#FFE847"), t(z, "offset", "0.225"), t(z, "stop-color", "#FFC830"), t(L, "offset", "0.368"), t(L, "stop-color", "#FF980E"), t(k, "offset", "0.401"), t(k, "stop-color", "#FF8B16"), t(y, "offset", "0.462"), t(y, "stop-color", "#FF672A"), t(S, "offset", "0.534"), t(S, "stop-color", "#FF3647"), t(G, "offset", "0.705"), t(G, "stop-color", "#E31587"), t(v, "id", "paint0_linear_6297_45086"), t(v, "x1", "17.7361"), t(v, "y1", "3.12268"), t(v, "x2", "1.51918"), t(v, "y2", "18.769"), t(v, "gradientUnits", "userSpaceOnUse"), t(E, "offset", "0.129"), t(E, "stop-color", "#FFBD4F"), t(R, "offset", "0.186"), t(R, "stop-color", "#FFAC31"), t(D, "offset", "0.247"), t(D, "stop-color", "#FF9D17"), t(T, "offset", "0.283"), t(T, "stop-color", "#FF980E"), t(q, "offset", "0.403"), t(q, "stop-color", "#FF563B"), t(I, "offset", "0.467"), t(I, "stop-color", "#FF3750"), t(P, "offset", "0.71"), t(P, "stop-color", "#F5156C"), t(V, "offset", "0.782"), t(V, "stop-color", "#EB0878"), t(Y, "offset", "0.86"), t(Y, "stop-color", "#E50080"), t(_, "id", "paint1_radial_6297_45086"), t(_, "cx", "0"), t(_, "cy", "0"), t(_, "r", "1"), t(_, "gradientUnits", "userSpaceOnUse"), t(_, "gradientTransform", "translate(16.9866 2.23629) scale(20.3653 20.3653)"), t(J, "offset", "0.3"), t(J, "stop-color", "#960E18"), t(oe, "offset", "0.351"), t(oe, "stop-color", "#B11927"), t(oe, "stop-opacity", "0.74"), t(ee, "offset", "0.435"), t(ee, "stop-color", "#DB293D"), t(ee, "stop-opacity", "0.343"), t(le, "offset", "0.497"), t(le, "stop-color", "#F5334B"), t(le, "stop-opacity", "0.094"), t(ae, "offset", "0.53"), t(ae, "stop-color", "#FF3750"), t(ae, "stop-opacity", "0"), t(H, "id", "paint2_radial_6297_45086"), t(H, "cx", "0"), t(H, "cy", "0"), t(H, "r", "1"), t(H, "gradientUnits", "userSpaceOnUse"), t(H, "gradientTransform", "translate(9.54495 10.5622) scale(20.3653 20.3653)"), t(Fe, "offset", "0.132"), t(Fe, "stop-color", "#FFF44F"), t(De, "offset", "0.252"), t(De, "stop-color", "#FFDC3E"), t(Ve, "offset", "0.506"), t(Ve, "stop-color", "#FF9D12"), t(Re, "offset", "0.526"), t(Re, "stop-color", "#FF980E"), t(te, "id", "paint3_radial_6297_45086"), t(te, "cx", "0"), t(te, "cy", "0"), t(te, "r", "1"), t(te, "gradientUnits", "userSpaceOnUse"), t(te, "gradientTransform", "translate(12.0001 -2.38427) scale(14.7538 14.7538)"), t(Te, "offset", "0.353"), t(Te, "stop-color", "#3A8EE6"), t(He, "offset", "0.472"), t(He, "stop-color", "#5C79F0"), t(Ke, "offset", "0.669"), t(Ke, "stop-color", "#9059FF"), t(Pe, "offset", "1"), t(Pe, "stop-color", "#C139E6"), t(ue, "id", "paint4_radial_6297_45086"), t(ue, "cx", "0"), t(ue, "cy", "0"), t(ue, "r", "1"), t(ue, "gradientUnits", "userSpaceOnUse"), t(ue, "gradientTransform", "translate(7.29217 15.8829) scale(9.6968 9.69679)"), t(Ne, "offset", "0.206"), t(Ne, "stop-color", "#9059FF"), t(Ne, "stop-opacity", "0"), t(Ze, "offset", "0.278"), t(Ze, "stop-color", "#8C4FF3"), t(Ze, "stop-opacity", "0.064"), t(Ae, "offset", "0.747"), t(Ae, "stop-color", "#7716A8"), t(Ae, "stop-opacity", "0.45"), t(ze, "offset", "0.975"), t(ze, "stop-color", "#6E008B"), t(ze, "stop-opacity", "0.6"), t(de, "id", "paint5_radial_6297_45086"), t(de, "cx", "0"), t(de, "cy", "0"), t(de, "r", "1"), t(de, "gradientUnits", "userSpaceOnUse"), t(de, "gradientTransform", "translate(10.6576 8.89437) rotate(-13.5916) scale(5.14118 6.01904)"), t(Ft, "stop-color", "#FFE226"), t(qe, "offset", "0.121"), t(qe, "stop-color", "#FFDB27"), t(Xe, "offset", "0.295"), t(Xe, "stop-color", "#FFC82A"), t(Qe, "offset", "0.502"), t(Qe, "stop-color", "#FFA930"), t(Ye, "offset", "0.732"), t(Ye, "stop-color", "#FF7E37"), t(Je, "offset", "0.792"), t(Je, "stop-color", "#FF7139"), t(ne, "id", "paint6_radial_6297_45086"), t(ne, "cx", "0"), t(ne, "cy", "0"), t(ne, "r", "1"), t(ne, "gradientUnits", "userSpaceOnUse"), t(ne, "gradientTransform", "translate(9.28808 1.40453) scale(6.97586)"), t(et, "offset", "0.113"), t(et, "stop-color", "#FFF44F"), t(tt, "offset", "0.456"), t(tt, "stop-color", "#FF980E"), t(rt, "offset", "0.622"), t(rt, "stop-color", "#FF5634"), t(ot, "offset", "0.716"), t(ot, "stop-color", "#FF3647"), t(lt, "offset", "0.904"), t(lt, "stop-color", "#E31587"), t(ie, "id", "paint7_radial_6297_45086"), t(ie, "cx", "0"), t(ie, "cy", "0"), t(ie, "r", "1"), t(ie, "gradientUnits", "userSpaceOnUse"), t(ie, "gradientTransform", "translate(14.7705 -3.02052) scale(29.7629 29.7629)"), t(Lt, "stop-color", "#FFF44F"), t(at, "offset", "0.06"), t(at, "stop-color", "#FFE847"), t(nt, "offset", "0.168"), t(nt, "stop-color", "#FFC830"), t(it, "offset", "0.304"), t(it, "stop-color", "#FF980E"), t(st, "offset", "0.356"), t(st, "stop-color", "#FF8B16"), t(ct, "offset", "0.455"), t(ct, "stop-color", "#FF672A"), t(ft, "offset", "0.57"), t(ft, "stop-color", "#FF3647"), t(ut, "offset", "0.737"), t(ut, "stop-color", "#E31587"), t(re, "id", "paint8_radial_6297_45086"), t(re, "cx", "0"), t(re, "cy", "0"), t(re, "r", "1"), t(re, "gradientUnits", "userSpaceOnUse"), t(re, "gradientTransform", "translate(12.262 -1.36605) rotate(83.976) scale(21.8139 14.3163)"), t(dt, "offset", "0.137"), t(dt, "stop-color", "#FFF44F"), t(pt, "offset", "0.48"), t(pt, "stop-color", "#FF980E"), t(ht, "offset", "0.592"), t(ht, "stop-color", "#FF5634"), t(gt, "offset", "0.655"), t(gt, "stop-color", "#FF3647"), t(mt, "offset", "0.904"), t(mt, "stop-color", "#E31587"), t(se, "id", "paint9_radial_6297_45086"), t(se, "cx", "0"), t(se, "cy", "0"), t(se, "r", "1"), t(se, "gradientUnits", "userSpaceOnUse"), t(se, "gradientTransform", "translate(9.12354 3.98625) scale(18.5815 18.5815)"), t(bt, "offset", "0.094"), t(bt, "stop-color", "#FFF44F"), t(wt, "offset", "0.231"), t(wt, "stop-color", "#FFE141"), t(Ct, "offset", "0.509"), t(Ct, "stop-color", "#FFAF1E"), t(vt, "offset", "0.626"), t(vt, "stop-color", "#FF980E"), t(pe, "id", "paint10_radial_6297_45086"), t(pe, "cx", "0"), t(pe, "cy", "0"), t(pe, "r", "1"), t(pe, "gradientUnits", "userSpaceOnUse"), t(pe, "gradientTransform", "translate(14.0286 5.07494) scale(20.3373 20.3373)"), t(Oe, "offset", "0.167"), t(Oe, "stop-color", "#FFF44F"), t(Oe, "stop-opacity", "0.8"), t(Ue, "offset", "0.266"), t(Ue, "stop-color", "#FFF44F"), t(Ue, "stop-opacity", "0.634"), t(Ge, "offset", "0.489"), t(Ge, "stop-color", "#FFF44F"), t(Ge, "stop-opacity", "0.217"), t(Ie, "offset", "0.6"), t(Ie, "stop-color", "#FFF44F"), t(Ie, "stop-opacity", "0"), t(he, "id", "paint11_linear_6297_45086"), t(he, "x1", "17.5413"), t(he, "y1", "3.039"), t(he, "x2", "3.74231"), t(he, "y2", "16.8377"), t(he, "gradientUnits", "userSpaceOnUse"), t(e, "xmlns", "http://www.w3.org/2000/svg"), t(
        e,
        "width",
        /*size*/
        l[0]
      ), t(
        e,
        "height",
        /*size*/
        l[0]
      ), t(e, "viewBox", "0 0 20 21"), t(e, "fill", "none");
    },
    m(Le, kt) {
      $(Le, e, kt), i(e, o), i(e, r), i(e, a), i(e, n), i(e, s), i(e, f), i(e, d), i(e, p), i(e, h), i(e, u), i(e, g), i(e, m), i(e, b), i(b, v), i(v, F), i(v, O), i(v, z), i(v, L), i(v, k), i(v, y), i(v, S), i(v, G), i(b, _), i(_, E), i(_, R), i(_, D), i(_, T), i(_, q), i(_, I), i(_, P), i(_, V), i(_, Y), i(b, H), i(H, J), i(H, oe), i(H, ee), i(H, le), i(H, ae), i(b, te), i(te, Fe), i(te, De), i(te, Ve), i(te, Re), i(b, ue), i(ue, Te), i(ue, He), i(ue, Ke), i(ue, Pe), i(b, de), i(de, Ne), i(de, Ze), i(de, Ae), i(de, ze), i(b, ne), i(ne, Ft), i(ne, qe), i(ne, Xe), i(ne, Qe), i(ne, Ye), i(ne, Je), i(b, ie), i(ie, et), i(ie, tt), i(ie, rt), i(ie, ot), i(ie, lt), i(b, re), i(re, Lt), i(re, at), i(re, nt), i(re, it), i(re, st), i(re, ct), i(re, ft), i(re, ut), i(b, se), i(se, dt), i(se, pt), i(se, ht), i(se, gt), i(se, mt), i(b, pe), i(pe, bt), i(pe, wt), i(pe, Ct), i(pe, vt), i(b, he), i(he, Oe), i(he, Ue), i(he, Ge), i(he, Ie);
    },
    p(Le, [kt]) {
      kt & /*size*/
      1 && t(
        e,
        "width",
        /*size*/
        Le[0]
      ), kt & /*size*/
      1 && t(
        e,
        "height",
        /*size*/
        Le[0]
      );
    },
    i: B,
    o: B,
    d(Le) {
      Le && M(e);
    }
  };
}
function Lo(l, e, o) {
  let { size: r = "21" } = e;
  return l.$$set = (a) => {
    "size" in a && o(0, r = a.size);
  }, [r];
}
class Bo extends j {
  constructor(e) {
    super(), W(this, e, Lo, Fo, U, { size: 0 });
  }
}
function Mo(l) {
  let e, o, r, a, n, s, f, d, p, h, u, g, m, b, v, F, O, z, L, k, y;
  return {
    c() {
      e = c("svg"), o = c("g"), r = c("mask"), a = c("circle"), n = c("g"), s = c("path"), f = c("path"), d = c("path"), p = c("path"), h = c("path"), u = c("path"), g = c("path"), m = c("path"), b = c("path"), v = c("path"), F = c("path"), O = c("path"), z = c("path"), L = c("defs"), k = c("clipPath"), y = c("rect"), t(a, "cx", "59.2004"), t(a, "cy", "53.9996"), t(a, "r", "53.9"), t(a, "fill", "#F2F2F2"), t(r, "id", "mask0_6297_45055"), xe(r, "mask-type", "alpha"), t(r, "maskUnits", "userSpaceOnUse"), t(r, "x", "5"), t(r, "y", "0"), t(r, "width", "109"), t(r, "height", "108"), t(s, "d", "M59.4166 107.844C89.1694 107.844 113.289 83.7244 113.289 53.9717C113.289 24.219 89.1694 0.0996094 59.4166 0.0996094C29.6639 0.0996094 5.54456 24.219 5.54456 53.9717C5.54456 83.7244 29.6639 107.844 59.4166 107.844Z"), t(s, "fill", "#BFBFBF"), t(s, "fill-opacity", "0.2"), t(n, "mask", "url(#mask0_6297_45055)"), t(f, "d", "M54.8457 95.6447L39.8764 33.1839C39.1425 30.1277 40.4294 26.9308 43.0935 25.2418L63.6825 12.1625C64.8688 11.4086 66.2964 11.4789 67.3721 12.1424C65.1503 9.74977 62.1946 9.70956 60.8978 10.745L39.3537 24.4275C36.2774 26.3778 34.7895 30.0674 35.634 33.6061L49.9397 93.3224C50.6636 96.3585 56.8463 99.3041 59.4501 97.5749V97.5548C57.6707 98.6305 55.3383 97.6754 54.8457 95.6246V95.6447Z"), t(f, "fill", "#8C8C8C"), t(f, "stroke", "#0F0F0F"), t(f, "stroke-width", "0.502662"), t(f, "stroke-miterlimit", "10"), t(d, "d", "M58.4029 90.9901L60.7654 96.4993L59.4484 97.5851C56.8446 99.3142 50.6719 96.3686 49.938 93.3325L48.5406 87.5117L58.4029 90.9901Z"), t(d, "fill", "#0F0F0F"), t(d, "stroke", "#0F0F0F"), t(d, "stroke-width", "0.603195"), t(d, "stroke-miterlimit", "10"), t(p, "d", "M83.7591 75.3371L68.9305 14.2838C68.3676 11.9715 65.6934 10.8958 63.6827 12.1625L43.0937 25.2418C40.4396 26.9307 39.1428 30.1176 39.8767 33.1839L54.8359 95.6447C55.3385 97.7357 57.7614 98.6908 59.5609 97.5045L80.532 83.5808C83.2363 81.7812 84.5231 78.4938 83.7591 75.3371Z"), t(p, "fill", "#8C8C8C"), t(p, "stroke", "#0F0F0F"), t(p, "stroke-width", "0.502662"), t(p, "stroke-miterlimit", "10"), t(h, "d", "M82.3298 74.4824L68.5367 16.0429C68.014 13.8312 65.5509 12.7857 63.7112 14.0021L61.1476 15.6911L60.2127 19.0288C60.0418 19.642 59.6597 20.1648 59.1269 20.5166L50.6922 26.1666C49.9382 26.6692 48.9832 26.7597 48.1588 26.3978L46.1884 25.5533L44.8111 26.4682C42.3681 28.0767 41.1919 31.1228 41.8755 34.0584L55.7993 93.855C56.2718 95.8556 58.5036 96.7805 60.1523 95.6445L79.4043 82.3843C81.8875 80.6752 83.0637 77.5185 82.3499 74.5026L82.3298 74.4824Z"), t(h, "fill", "#FF875B"), t(h, "stroke", "#0F0F0F"), t(h, "stroke-width", "0.603195"), t(h, "stroke-miterlimit", "10"), t(u, "d", "M51.5376 22.9399C51.5979 23.1208 51.809 23.161 52.0101 23.0203L58.1224 18.9085C58.3235 18.7778 58.4341 18.5164 58.3738 18.3355C58.3134 18.1545 58.1023 18.1143 57.9013 18.255L51.7889 22.3668C51.5878 22.4975 51.4772 22.7589 51.5376 22.9399Z"), t(u, "fill", "#0F0F0F"), t(g, "d", "M50.1602 24.5681C50.4718 24.548 50.703 24.2263 50.6729 23.8643C50.6427 23.4924 50.3713 23.2109 50.0697 23.231C49.758 23.2511 49.5268 23.5728 49.557 23.9347C49.5871 24.3067 49.8586 24.5882 50.1602 24.5681Z"), t(g, "fill", "#0F0F0F"), t(m, "d", "M50.9649 73.1354L54.2624 23.7739L50.6834 26.1666C50.0199 26.6089 49.1955 26.7296 48.4415 26.5084L46.2399 52.8982L50.9548 73.1454L50.9649 73.1354Z"), t(m, "fill", "#FFCBB8"), t(b, "d", "M44.7614 46.4642L45.2841 48.696L47.2847 26.0159L46.6413 25.7344L44.7614 46.4541V46.4642Z"), t(b, "fill", "#FFCBB8"), t(v, "d", "M82.3298 74.4824L68.5367 16.0429C68.014 13.8312 65.5509 12.7857 63.7112 14.0021L61.1476 15.6911L60.2127 19.0288C60.0418 19.642 59.6597 20.1648 59.1269 20.5166L50.6922 26.1666C49.9382 26.6692 48.9832 26.7597 48.1588 26.3978L46.1884 25.5533L44.8111 26.4682C42.3681 28.0767 41.1919 31.1228 41.8755 34.0584L55.7993 93.8551C56.2718 95.8556 58.5036 96.7805 60.1523 95.6445L79.4043 82.3843C81.8875 80.6752 83.0637 77.5185 82.3499 74.5026L82.3298 74.4824Z"), t(v, "stroke", "#0F0F0F"), t(v, "stroke-width", "0.502662"), t(v, "stroke-miterlimit", "10"), t(F, "d", "M47.7989 83.8722L52.4535 85.1389L41.4151 39.5675L36.861 38.7432L47.7989 83.8822V83.8722Z"), t(F, "fill", "#0F0F0F"), t(F, "stroke", "#0F0F0F"), t(F, "stroke-width", "0.603195"), t(F, "stroke-miterlimit", "10"), t(O, "d", "M39.5939 33.3041C38.9806 30.7104 39.6341 27.9759 41.7654 25.8949L38.679 25.3018C36.3969 27.1918 35.4419 30.0268 35.7837 32.7713L39.5939 33.3041Z"), t(O, "fill", "#F8F8F8"), t(z, "d", "M62.0044 48.0829L59.8027 49.2792C59.7324 49.3194 59.6921 49.43 59.7223 49.5205C60.5366 52.2147 60.2451 55.3413 58.8678 58.1562C58.8276 58.2467 58.8477 58.3573 58.9181 58.4176L61.0694 60.0161C61.1398 60.0663 61.2404 60.0462 61.2906 59.9557C62.1451 58.2065 62.6679 56.3466 62.8489 54.4868C64.0653 55.6127 65.523 56.4472 67.1416 56.9297C67.222 56.9498 67.3025 56.8895 67.3226 56.789L67.775 53.783C67.785 53.6825 67.7347 53.582 67.6543 53.5619C65.1209 52.7777 63.1404 50.8374 62.2055 48.1834C62.1753 48.0929 62.0848 48.0426 62.0145 48.0829H62.0044Z"), t(z, "fill", "white"), t(o, "clip-path", "url(#clip0_6297_45055)"), t(y, "width", "117.6"), t(y, "height", "107.8"), t(y, "fill", "white"), t(y, "transform", "translate(0.400024 0.100098)"), t(k, "id", "clip0_6297_45055"), t(e, "xmlns", "http://www.w3.org/2000/svg"), t(e, "width", "118"), t(e, "height", "108"), t(e, "viewBox", "0 0 118 108"), t(e, "fill", "none");
    },
    m(S, G) {
      $(S, e, G), i(e, o), i(o, r), i(r, a), i(o, n), i(n, s), i(o, f), i(o, d), i(o, p), i(o, h), i(o, u), i(o, g), i(o, m), i(o, b), i(o, v), i(o, F), i(o, O), i(o, z), i(e, L), i(L, k), i(k, y);
    },
    p: B,
    i: B,
    o: B,
    d(S) {
      S && M(e);
    }
  };
}
class So extends j {
  constructor(e) {
    super(), W(this, e, null, Mo, U, {});
  }
}
function $o(l) {
  let e, o, r, a, n, s, f, d, p, h, u, g, m, b, v, F, O, z, L, k, y, S, G, _, E, R, D, T, q, I, P, V, Y, H, J, oe, ee, le, ae;
  return {
    c() {
      e = c("svg"), o = c("mask"), r = c("ellipse"), a = c("g"), n = c("path"), s = c("rect"), f = c("ellipse"), d = c("rect"), p = c("rect"), h = c("rect"), u = c("rect"), g = c("rect"), m = c("rect"), b = c("path"), v = c("g"), F = c("path"), O = c("path"), z = c("rect"), L = c("rect"), k = c("path"), y = c("rect"), S = c("rect"), G = c("rect"), _ = c("path"), E = c("rect"), R = c("rect"), D = c("rect"), T = c("rect"), q = c("path"), I = c("path"), P = c("defs"), V = c("filter"), Y = c("feFlood"), H = c("feColorMatrix"), J = c("feOffset"), oe = c("feGaussianBlur"), ee = c("feColorMatrix"), le = c("feBlend"), ae = c("feBlend"), t(r, "cx", "59.2004"), t(r, "cy", "55.0779"), t(r, "rx", "53.9"), t(r, "ry", "54.2667"), t(r, "fill", "#F2F2F2"), t(o, "id", "mask0_6297_45112"), xe(o, "mask-type", "alpha"), t(o, "maskUnits", "userSpaceOnUse"), t(o, "x", "5"), t(o, "y", "0"), t(o, "width", "109"), t(o, "height", "110"), t(n, "d", "M59.4166 109.288C89.1694 109.288 113.289 85.005 113.289 55.0498C113.289 25.0947 89.1694 0.811279 59.4166 0.811279C29.6639 0.811279 5.54456 25.0947 5.54456 55.0498C5.54456 85.005 29.6639 109.288 59.4166 109.288Z"), t(n, "fill", "#BFBFBF"), t(n, "fill-opacity", "0.2"), t(s, "x", "21.4704"), t(s, "y", "22.5181"), t(s, "width", "75.46"), t(s, "height", "105.573"), t(s, "rx", "4.9"), t(s, "fill", "black"), t(f, "opacity", "0.4"), t(f, "cx", "59.2004"), t(f, "cy", "45.2112"), t(f, "rx", "9.31"), t(f, "ry", "9.37333"), t(f, "fill", "#FF875B"), t(d, "x", "54.7904"), t(d, "y", "44.2246"), t(d, "width", "5.39"), t(d, "height", "1.97333"), t(d, "rx", "0.986666"), t(d, "fill", "#8C8C8C"), t(p, "x", "61.65"), t(p, "y", "44.2244"), t(p, "width", "1.96"), t(p, "height", "1.97333"), t(p, "rx", "0.98"), t(p, "fill", "#8C8C8C"), t(h, "x", "54.3004"), t(h, "y", "28.438"), t(h, "width", "10.29"), t(h, "height", "1.48"), t(h, "rx", "0.74"), t(h, "fill", "#595959"), t(u, "x", "26.3708"), t(u, "y", "70.3047"), t(u, "width", "65.66"), t(u, "height", "11.76"), t(u, "rx", "1.96"), t(u, "fill", "#595959"), t(g, "x", "36.1716"), t(g, "y", "74.2246"), t(g, "width", "18.62"), t(g, "height", "1.47"), t(g, "rx", "0.735"), t(g, "fill", "#F9F9F9"), t(m, "x", "36.1716"), t(m, "y", "76.6748"), t(m, "width", "34.3"), t(m, "height", "1.47"), t(m, "rx", "0.735"), t(m, "fill", "#8C8C8C"), t(b, "fill-rule", "evenodd"), t(b, "clip-rule", "evenodd"), t(b, "d", "M88.5198 76.1844C88.5198 74.8313 87.4229 73.7344 86.0698 73.7344H83.2114C81.8583 73.7344 80.7614 74.8313 80.7614 76.1844C80.7614 77.5375 81.8583 78.6344 83.2114 78.6344H86.0698C87.4229 78.6344 88.5198 77.5375 88.5198 76.1844Z"), t(b, "fill", "#FF875B"), t(F, "fill-rule", "evenodd"), t(F, "clip-rule", "evenodd"), t(F, "d", "M86.0698 78.2262C84.9422 78.2262 84.0281 77.3121 84.0281 76.1845C84.0281 75.0569 84.9422 74.1428 86.0698 74.1428C87.1974 74.1428 88.1115 75.0569 88.1115 76.1845C88.1115 77.3121 87.1974 78.2262 86.0698 78.2262Z"), t(F, "fill", "black"), t(v, "filter", "url(#filter0_d_6297_45112)"), t(O, "d", "M29.8866 74.895C29.9417 74.8416 30.0165 74.8115 30.0945 74.8115H32.4465C32.5244 74.8115 32.5992 74.8416 32.6544 74.895C32.7095 74.9485 32.7405 75.021 32.7405 75.0966V75.9334C32.7405 77.2483 31.5889 77.6834 31.365 77.7557C31.3038 77.7768 31.2371 77.7768 31.176 77.7557C30.952 77.6834 29.8005 77.2483 29.8005 75.9334V75.0966C29.8005 75.021 29.8315 74.9485 29.8866 74.895Z"), t(O, "fill", "white"), t(z, "x", "26.3708"), t(z, "y", "87.9446"), t(z, "width", "65.66"), t(z, "height", "6.86"), t(z, "rx", "1.96"), t(z, "fill", "#262626"), t(L, "x", "35.6814"), t(L, "y", "90.8845"), t(L, "width", "34.79"), t(L, "height", "1.47"), t(L, "rx", "0.735"), t(L, "fill", "#595959"), t(k, "d", "M86.1501 90.3948L87.1301 91.3748L86.1501 92.3548"), t(k, "stroke", "#595959"), t(k, "stroke-width", "0.98"), t(k, "stroke-linecap", "round"), t(k, "stroke-linejoin", "round"), t(y, "x", "30.7809"), t(y, "y", "90.3945"), t(y, "width", "2.45"), t(y, "height", "2.45"), t(y, "rx", "0.49"), t(y, "fill", "#595959"), t(S, "x", "26.3708"), t(S, "y", "97.7446"), t(S, "width", "65.66"), t(S, "height", "6.86"), t(S, "rx", "1.96"), t(S, "fill", "#262626"), t(G, "x", "35.6814"), t(G, "y", "100.685"), t(G, "width", "34.79"), t(G, "height", "1.47"), t(G, "rx", "0.735"), t(G, "fill", "#595959"), t(_, "d", "M86.1501 100.195L87.1301 101.175L86.1501 102.155"), t(_, "stroke", "#595959"), t(_, "stroke-width", "0.98"), t(_, "stroke-linecap", "round"), t(_, "stroke-linejoin", "round"), t(E, "x", "30.7809"), t(E, "y", "100.195"), t(E, "width", "2.45"), t(E, "height", "2.45"), t(E, "rx", "0.49"), t(E, "fill", "#595959"), t(R, "x", "26.3707"), t(R, "y", "60.5046"), t(R, "width", "65.66"), t(R, "height", "6.86"), t(R, "rx", "1.96"), t(R, "fill", "#0F0F0F"), t(D, "x", "44.0114"), t(D, "y", "63.4446"), t(D, "width", "30.87"), t(D, "height", "1.47"), t(D, "rx", "0.735"), t(D, "fill", "#595959"), t(T, "x", "85.1715"), t(T, "y", "62.9548"), t(T, "width", "2.45"), t(T, "height", "2.45"), t(T, "rx", "0.49"), t(T, "fill", "#595959"), t(a, "mask", "url(#mask0_6297_45112)"), t(q, "d", "M27.8414 29.7214L26.8614 28.7414L27.8414 27.7614"), t(q, "stroke", "#595959"), t(q, "stroke-width", "0.98"), t(q, "stroke-linecap", "round"), t(q, "stroke-linejoin", "round"), t(I, "d", "M27.3512 28.7415L29.5562 28.7415"), t(I, "stroke", "#595959"), t(I, "stroke-width", "0.98"), t(I, "stroke-linecap", "round"), t(Y, "flood-opacity", "0"), t(Y, "result", "BackgroundImageFix"), t(H, "in", "SourceAlpha"), t(H, "type", "matrix"), t(H, "values", "0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"), t(H, "result", "hardAlpha"), t(J, "dy", "0.326667"), t(oe, "stdDeviation", "0.163333"), t(ee, "type", "matrix"), t(ee, "values", "0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.1 0"), t(le, "mode", "normal"), t(le, "in2", "BackgroundImageFix"), t(le, "result", "effect1_dropShadow_6297_45112"), t(ae, "mode", "normal"), t(ae, "in", "SourceGraphic"), t(ae, "in2", "effect1_dropShadow_6297_45112"), t(ae, "result", "shape"), t(V, "id", "filter0_d_6297_45112"), t(V, "x", "83.7014"), t(V, "y", "74.1428"), t(V, "width", "4.73671"), t(V, "height", "4.73659"), t(V, "filterUnits", "userSpaceOnUse"), t(V, "color-interpolation-filters", "sRGB"), t(e, "width", "118"), t(e, "height", "110"), t(e, "viewBox", "0 0 118 110"), t(e, "fill", "none"), t(e, "xmlns", "http://www.w3.org/2000/svg");
    },
    m(te, Fe) {
      $(te, e, Fe), i(e, o), i(o, r), i(e, a), i(a, n), i(a, s), i(a, f), i(a, d), i(a, p), i(a, h), i(a, u), i(a, g), i(a, m), i(a, b), i(a, v), i(v, F), i(a, O), i(a, z), i(a, L), i(a, k), i(a, y), i(a, S), i(a, G), i(a, _), i(a, E), i(a, R), i(a, D), i(a, T), i(e, q), i(e, I), i(e, P), i(P, V), i(V, Y), i(V, H), i(V, J), i(V, oe), i(V, ee), i(V, le), i(V, ae);
    },
    p: B,
    i: B,
    o: B,
    d(te) {
      te && M(e);
    }
  };
}
class Eo extends j {
  constructor(e) {
    super(), W(this, e, null, $o, U, {});
  }
}
function No(l) {
  let e, o, r, a, n, s, f, d, p, h, u, g, m, b, v, F, O, z, L, k, y, S, G, _, E, R, D, T, q, I, P, V, Y, H, J, oe, ee;
  return {
    c() {
      e = c("svg"), o = c("mask"), r = c("ellipse"), a = c("g"), n = c("path"), s = c("rect"), f = c("ellipse"), d = c("rect"), p = c("rect"), h = c("rect"), u = c("rect"), g = c("rect"), m = c("rect"), b = c("path"), v = c("g"), F = c("path"), O = c("path"), z = c("rect"), L = c("rect"), k = c("path"), y = c("rect"), S = c("rect"), G = c("rect"), _ = c("path"), E = c("rect"), R = c("rect"), D = c("rect"), T = c("rect"), q = c("defs"), I = c("filter"), P = c("feFlood"), V = c("feColorMatrix"), Y = c("feOffset"), H = c("feGaussianBlur"), J = c("feColorMatrix"), oe = c("feBlend"), ee = c("feBlend"), t(r, "cx", "54.201"), t(r, "cy", "54.2667"), t(r, "rx", "53.9"), t(r, "ry", "54.2667"), t(r, "fill", "#F2F2F2"), t(o, "id", "mask0_697_1274"), xe(o, "mask-type", "alpha"), t(o, "maskUnits", "userSpaceOnUse"), t(o, "x", "0"), t(o, "y", "0"), t(o, "width", "109"), t(o, "height", "109"), t(n, "d", "M54.4182 108.477C84.171 108.477 108.29 84.1937 108.29 54.2386C108.29 24.2834 84.171 0 54.4182 0C24.6655 0 0.546143 24.2834 0.546143 54.2386C0.546143 84.1937 24.6655 108.477 54.4182 108.477Z"), t(n, "fill", "#BFBFBF"), t(n, "fill-opacity", "0.2"), t(s, "x", "16.4709"), t(s, "y", "21.7067"), t(s, "width", "75.46"), t(s, "height", "105.573"), t(s, "rx", "4.9"), t(s, "fill", "black"), t(f, "opacity", "0.4"), t(f, "cx", "54.2011"), t(f, "cy", "44.3999"), t(f, "rx", "9.31"), t(f, "ry", "9.37333"), t(f, "fill", "#6D6D6D"), t(d, "x", "49.791"), t(d, "y", "43.4133"), t(d, "width", "5.39"), t(d, "height", "1.97333"), t(d, "rx", "0.986666"), t(d, "fill", "#8C8C8C"), t(p, "x", "56.6511"), t(p, "y", "43.4133"), t(p, "width", "1.96"), t(p, "height", "1.97333"), t(p, "rx", "0.98"), t(p, "fill", "#8C8C8C"), t(h, "x", "49.301"), t(h, "y", "27.6266"), t(h, "width", "10.29"), t(h, "height", "1.48"), t(h, "rx", "0.74"), t(h, "fill", "#595959"), t(u, "x", "21.3711"), t(u, "y", "69.4933"), t(u, "width", "65.66"), t(u, "height", "11.76"), t(u, "rx", "1.96"), t(u, "fill", "#3A3A3A"), t(g, "x", "31.1711"), t(g, "y", "73.4133"), t(g, "width", "18.62"), t(g, "height", "1.47"), t(g, "rx", "0.735"), t(g, "fill", "#CDCDCD"), t(m, "x", "31.1711"), t(m, "y", "75.8633"), t(m, "width", "34.3"), t(m, "height", "1.47"), t(m, "rx", "0.735"), t(m, "fill", "#8C8C8C"), t(b, "fill-rule", "evenodd"), t(b, "clip-rule", "evenodd"), t(b, "d", "M83.52 75.373C83.52 74.0199 82.4231 72.923 81.07 72.923H78.2117C76.8586 72.923 75.7617 74.0199 75.7617 75.373C75.7617 76.7261 76.8586 77.823 78.2117 77.823H81.07C82.4231 77.823 83.52 76.7261 83.52 75.373Z"), t(b, "fill", "#919191"), t(F, "fill-rule", "evenodd"), t(F, "clip-rule", "evenodd"), t(F, "d", "M81.0701 77.4147C79.9426 77.4147 79.0285 76.5006 79.0285 75.373C79.0285 74.2454 79.9426 73.3314 81.0701 73.3314C82.1977 73.3314 83.1118 74.2454 83.1118 75.373C83.1118 76.5006 82.1977 77.4147 81.0701 77.4147Z"), t(F, "fill", "#CDCDCD"), t(v, "filter", "url(#filter0_d_697_1274)"), t(O, "d", "M24.8874 74.0835C24.9425 74.03 25.0173 74 25.0953 74H27.4473C27.5252 74 27.6 74.03 27.6552 74.0835C27.7103 74.137 27.7413 74.2095 27.7413 74.2851V75.1219C27.7413 76.4368 26.5897 76.8719 26.3658 76.9442C26.3046 76.9653 26.2379 76.9653 26.1768 76.9442C25.9528 76.8719 24.8013 76.4368 24.8013 75.1219V74.2851C24.8013 74.2095 24.8322 74.137 24.8874 74.0835Z"), t(O, "fill", "#CDCDCD"), t(z, "x", "21.3711"), t(z, "y", "87.1334"), t(z, "width", "65.66"), t(z, "height", "6.86"), t(z, "rx", "1.96"), t(z, "fill", "#262626"), t(L, "x", "30.6812"), t(L, "y", "90.0734"), t(L, "width", "34.79"), t(L, "height", "1.47"), t(L, "rx", "0.735"), t(L, "fill", "#595959"), t(k, "d", "M81.1511 89.5834L82.1311 90.5634L81.1511 91.5434"), t(k, "stroke", "#595959"), t(k, "stroke-width", "0.98"), t(k, "stroke-linecap", "round"), t(k, "stroke-linejoin", "round"), t(y, "x", "25.7812"), t(y, "y", "89.5834"), t(y, "width", "2.45"), t(y, "height", "2.45"), t(y, "rx", "0.49"), t(y, "fill", "#595959"), t(S, "x", "21.3711"), t(S, "y", "96.9333"), t(S, "width", "65.66"), t(S, "height", "6.86"), t(S, "rx", "1.96"), t(S, "fill", "#262626"), t(G, "x", "30.6812"), t(G, "y", "99.8734"), t(G, "width", "34.79"), t(G, "height", "1.47"), t(G, "rx", "0.735"), t(G, "fill", "#595959"), t(_, "d", "M81.1511 99.3834L82.1311 100.363L81.1511 101.343"), t(_, "stroke", "#595959"), t(_, "stroke-width", "0.98"), t(_, "stroke-linecap", "round"), t(_, "stroke-linejoin", "round"), t(E, "x", "25.7812"), t(E, "y", "99.3834"), t(E, "width", "2.45"), t(E, "height", "2.45"), t(E, "rx", "0.49"), t(E, "fill", "#595959"), t(R, "x", "21.3711"), t(R, "y", "59.6934"), t(R, "width", "65.66"), t(R, "height", "6.86"), t(R, "rx", "1.96"), t(R, "fill", "#0F0F0F"), t(D, "x", "39.0112"), t(D, "y", "62.6334"), t(D, "width", "30.87"), t(D, "height", "1.47"), t(D, "rx", "0.735"), t(D, "fill", "#595959"), t(T, "x", "80.1711"), t(T, "y", "62.1434"), t(T, "width", "2.45"), t(T, "height", "2.45"), t(T, "rx", "0.49"), t(T, "fill", "#595959"), t(a, "mask", "url(#mask0_697_1274)"), t(P, "flood-opacity", "0"), t(P, "result", "BackgroundImageFix"), t(V, "in", "SourceAlpha"), t(V, "type", "matrix"), t(V, "values", "0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"), t(V, "result", "hardAlpha"), t(Y, "dy", "0.326667"), t(H, "stdDeviation", "0.163333"), t(J, "type", "matrix"), t(J, "values", "0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.1 0"), t(oe, "mode", "normal"), t(oe, "in2", "BackgroundImageFix"), t(oe, "result", "effect1_dropShadow_697_1274"), t(ee, "mode", "normal"), t(ee, "in", "SourceGraphic"), t(ee, "in2", "effect1_dropShadow_697_1274"), t(ee, "result", "shape"), t(I, "id", "filter0_d_697_1274"), t(I, "x", "78.7019"), t(I, "y", "73.3314"), t(I, "width", "4.73659"), t(I, "height", "4.73665"), t(I, "filterUnits", "userSpaceOnUse"), t(I, "color-interpolation-filters", "sRGB"), t(e, "width", "109"), t(e, "height", "109"), t(e, "viewBox", "0 0 109 109"), t(e, "fill", "none"), t(e, "xmlns", "http://www.w3.org/2000/svg");
    },
    m(le, ae) {
      $(le, e, ae), i(e, o), i(o, r), i(e, a), i(a, n), i(a, s), i(a, f), i(a, d), i(a, p), i(a, h), i(a, u), i(a, g), i(a, m), i(a, b), i(a, v), i(v, F), i(a, O), i(a, z), i(a, L), i(a, k), i(a, y), i(a, S), i(a, G), i(a, _), i(a, E), i(a, R), i(a, D), i(a, T), i(e, q), i(q, I), i(I, P), i(I, V), i(I, Y), i(I, H), i(I, J), i(I, oe), i(I, ee);
    },
    p: B,
    i: B,
    o: B,
    d(le) {
      le && M(e);
    }
  };
}
class Zo extends j {
  constructor(e) {
    super(), W(this, e, null, No, U, {});
  }
}
const Ao = (l) => ({}), tr = (l) => ({}), zo = (l) => ({}), rr = (l) => ({}), Oo = (l) => ({}), or = (l) => ({});
function Uo(l) {
  let e;
  const o = (
    /*#slots*/
    l[4].button
  ), r = be(
    o,
    l,
    /*$$scope*/
    l[6],
    rr
  );
  return {
    c() {
      r && r.c();
    },
    m(a, n) {
      r && r.m(a, n), e = !0;
    },
    p(a, n) {
      r && r.p && (!e || n & /*$$scope*/
      64) && Ce(
        r,
        o,
        a,
        /*$$scope*/
        a[6],
        e ? we(
          o,
          /*$$scope*/
          a[6],
          n,
          zo
        ) : ve(
          /*$$scope*/
          a[6]
        ),
        rr
      );
    },
    i(a) {
      e || (w(r, a), e = !0);
    },
    o(a) {
      C(r, a), e = !1;
    },
    d(a) {
      r && r.d(a);
    }
  };
}
function Go(l) {
  let e, o, r, a, n, s, f, d, p, h, u, g, m, b, v;
  const F = (
    /*#slots*/
    l[4].icons
  ), O = be(
    F,
    l,
    /*$$scope*/
    l[6],
    or
  );
  g = new vr({
    props: {
      as: (
        /*handleClick*/
        l[3] ? "button" : "a"
      ),
      href: (
        /*link*/
        l[2]
      ),
      handleClick: (
        /*handleClick*/
        l[3]
      ),
      handleKeyup: (
        /*func*/
        l[5]
      ),
      className: "bg-button-secondary text-primary hover:bg-button-secondary-hover flex w-full h-[32px] text-b3 rounded-3xl justify-center items-center",
      $$slots: { default: [Uo] },
      $$scope: { ctx: l }
    }
  });
  const z = (
    /*#slots*/
    l[4].graphic
  ), L = be(
    z,
    l,
    /*$$scope*/
    l[6],
    tr
  );
  return {
    c() {
      e = x("div"), o = x("div"), r = x("div"), a = x("h4"), n = X(
        /*title*/
        l[0]
      ), s = K(), f = x("p"), d = X(
        /*subtitle*/
        l[1]
      ), p = K(), h = x("div"), O && O.c(), u = K(), A(g.$$.fragment), m = K(), b = x("div"), L && L.c(), t(a, "class", "text-heading5 text-primary capitalize"), t(f, "class", "text-l1 text-secondary-web"), t(r, "class", "flex flex-col gap-1 text-left"), t(h, "class", "flex gap-3 text-primary"), t(o, "class", "flex flex-col gap-3"), t(b, "class", "flex items-center"), t(e, "class", "flex flex-row gap-3 px-4 py-3 rounded-xl bg-surface-elevated");
    },
    m(k, y) {
      $(k, e, y), i(e, o), i(o, r), i(r, a), i(a, n), i(r, s), i(r, f), i(f, d), i(o, p), i(o, h), O && O.m(h, null), i(o, u), N(g, o, null), i(e, m), i(e, b), L && L.m(b, null), v = !0;
    },
    p(k, [y]) {
      (!v || y & /*title*/
      1) && ge(
        n,
        /*title*/
        k[0]
      ), (!v || y & /*subtitle*/
      2) && ge(
        d,
        /*subtitle*/
        k[1]
      ), O && O.p && (!v || y & /*$$scope*/
      64) && Ce(
        O,
        F,
        k,
        /*$$scope*/
        k[6],
        v ? we(
          F,
          /*$$scope*/
          k[6],
          y,
          Oo
        ) : ve(
          /*$$scope*/
          k[6]
        ),
        or
      );
      const S = {};
      y & /*handleClick*/
      8 && (S.as = /*handleClick*/
      k[3] ? "button" : "a"), y & /*link*/
      4 && (S.href = /*link*/
      k[2]), y & /*handleClick*/
      8 && (S.handleClick = /*handleClick*/
      k[3]), y & /*handleClick*/
      8 && (S.handleKeyup = /*func*/
      k[5]), y & /*$$scope*/
      64 && (S.$$scope = { dirty: y, ctx: k }), g.$set(S), L && L.p && (!v || y & /*$$scope*/
      64) && Ce(
        L,
        z,
        k,
        /*$$scope*/
        k[6],
        v ? we(
          z,
          /*$$scope*/
          k[6],
          y,
          Ao
        ) : ve(
          /*$$scope*/
          k[6]
        ),
        tr
      );
    },
    i(k) {
      v || (w(O, k), w(g.$$.fragment, k), w(L, k), v = !0);
    },
    o(k) {
      C(O, k), C(g.$$.fragment, k), C(L, k), v = !1;
    },
    d(k) {
      k && M(e), O && O.d(k), Z(g), L && L.d(k);
    }
  };
}
function Io(l, e, o) {
  let { $$slots: r = {}, $$scope: a } = e, { title: n } = e, { subtitle: s } = e, { link: f = void 0 } = e, { handleClick: d = void 0 } = e;
  const p = (h) => {
    d && h.key === "Enter" && d();
  };
  return l.$$set = (h) => {
    "title" in h && o(0, n = h.title), "subtitle" in h && o(1, s = h.subtitle), "link" in h && o(2, f = h.link), "handleClick" in h && o(3, d = h.handleClick), "$$scope" in h && o(6, a = h.$$scope);
  }, [n, s, f, d, r, p, a];
}
class kr extends j {
  constructor(e) {
    super(), W(this, e, Io, Go, U, {
      title: 0,
      subtitle: 1,
      link: 2,
      handleClick: 3
    });
  }
}
function lr(l) {
  let e, o;
  return e = new kr({
    props: {
      title: "Ready mobile",
      subtitle: "Download Ready wallet on your mobile.",
      link: "https://www.argent.xyz/app",
      $$slots: {
        graphic: [Do],
        button: [jo],
        icons: [Wo]
      },
      $$scope: { ctx: l }
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function Wo(l) {
  let e, o, r, a;
  return e = new br({}), r = new gr({}), {
    c() {
      A(e.$$.fragment), o = K(), A(r.$$.fragment);
    },
    m(n, s) {
      N(e, n, s), $(n, o, s), N(r, n, s), a = !0;
    },
    i(n) {
      a || (w(e.$$.fragment, n), w(r.$$.fragment, n), a = !0);
    },
    o(n) {
      C(e.$$.fragment, n), C(r.$$.fragment, n), a = !1;
    },
    d(n) {
      n && M(o), Z(e, n), Z(r, n);
    }
  };
}
function jo(l) {
  let e;
  return {
    c() {
      e = X("Download");
    },
    m(o, r) {
      $(o, e, r);
    },
    d(o) {
      o && M(e);
    }
  };
}
function Do(l) {
  let e, o;
  return e = new So({}), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function ar(l) {
  let e, o;
  return e = new kr({
    props: {
      title: (
        /*extensionName*/
        l[1]
      ),
      subtitle: `Install ${/*extensionName*/
      l[1]} extension.`,
      link: (
        /*storeLink*/
        l[3]
      ),
      $$slots: {
        graphic: [Ho],
        button: [Vo]
      },
      $$scope: { ctx: l }
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, a) {
      const n = {};
      a & /*extensionName*/
      2 && (n.title = /*extensionName*/
      r[1]), a & /*extensionName*/
      2 && (n.subtitle = `Install ${/*extensionName*/
      r[1]} extension.`), a & /*storeLink*/
      8 && (n.link = /*storeLink*/
      r[3]), a & /*$$scope, isArgent, store*/
      37 && (n.$$scope = { dirty: a, ctx: r }), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function Vo(l) {
  let e, o, r, a = (
    /*storeData*/
    l[4][
      /*store*/
      l[2]
    ].name + ""
  ), n, s;
  var f = (
    /*storeData*/
    l[4][
      /*store*/
      l[2]
    ].icon
  );
  function d(p, h) {
    return {};
  }
  return f && (o = Wt(f, d())), {
    c() {
      e = x("div"), o && A(o.$$.fragment), r = X(" Install for "), n = X(a), t(e, "class", "flex gap-1 items-center whitespace-nowrap");
    },
    m(p, h) {
      $(p, e, h), o && N(o, e, null), i(e, r), i(e, n), s = !0;
    },
    p(p, h) {
      if (h & /*store*/
      4 && f !== (f = /*storeData*/
      p[4][
        /*store*/
        p[2]
      ].icon)) {
        if (o) {
          ce();
          const u = o;
          C(u.$$.fragment, 1, 0, () => {
            Z(u, 1);
          }), fe();
        }
        f ? (o = Wt(f, d()), A(o.$$.fragment), w(o.$$.fragment, 1), N(o, e, r)) : o = null;
      }
      (!s || h & /*store*/
      4) && a !== (a = /*storeData*/
      p[4][
        /*store*/
        p[2]
      ].name + "") && ge(n, a);
    },
    i(p) {
      s || (o && w(o.$$.fragment, p), s = !0);
    },
    o(p) {
      o && C(o.$$.fragment, p), s = !1;
    },
    d(p) {
      p && M(e), o && Z(o);
    }
  };
}
function Ro(l) {
  let e, o;
  return e = new Zo({}), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function To(l) {
  let e, o;
  return e = new Eo({}), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function Ho(l) {
  let e, o, r, a;
  const n = [To, Ro], s = [];
  function f(d, p) {
    return (
      /*isArgent*/
      d[0] ? 0 : 1
    );
  }
  return e = f(l), o = s[e] = n[e](l), {
    c() {
      o.c(), r = ye();
    },
    m(d, p) {
      s[e].m(d, p), $(d, r, p), a = !0;
    },
    p(d, p) {
      let h = e;
      e = f(d), e !== h && (ce(), C(s[h], 1, 1, () => {
        s[h] = null;
      }), fe(), o = s[e], o || (o = s[e] = n[e](d), o.c()), w(o, 1), o.m(r.parentNode, r));
    },
    i(d) {
      a || (w(o), a = !0);
    },
    o(d) {
      C(o), a = !1;
    },
    d(d) {
      d && M(r), s[e].d(d);
    }
  };
}
function nr(l) {
  let e, o, r, a, n, s;
  return e = new Ut({}), n = new Cr({
    props: {
      as: "a",
      className: "text-brand",
      href: "https://www.ready.co/",
      $$slots: { default: [Ko] },
      $$scope: { ctx: l }
    }
  }), {
    c() {
      A(e.$$.fragment), o = K(), r = x("p"), a = X(`If you want to learn more about Ready visit our site:
        `), A(n.$$.fragment), t(r, "class", "text-[13px] text-secondary");
    },
    m(f, d) {
      N(e, f, d), $(f, o, d), $(f, r, d), i(r, a), N(n, r, null), s = !0;
    },
    i(f) {
      s || (w(e.$$.fragment, f), w(n.$$.fragment, f), s = !0);
    },
    o(f) {
      C(e.$$.fragment, f), C(n.$$.fragment, f), s = !1;
    },
    d(f) {
      f && (M(o), M(r)), Z(e, f), Z(n);
    }
  };
}
function Ko(l) {
  let e;
  return {
    c() {
      e = X("www.ready.co");
    },
    m(o, r) {
      $(o, e, r);
    },
    d(o) {
      o && M(e);
    }
  };
}
function Po(l) {
  let e, o, r, a, n, s, f = (
    /*isArgent*/
    l[0] && lr(l)
  ), d = (
    /*store*/
    l[2] && /*storeLink*/
    l[3] && ar(l)
  ), p = (
    /*isArgent*/
    l[0] && nr(l)
  );
  return {
    c() {
      e = x("section"), o = x("div"), f && f.c(), r = K(), d && d.c(), a = K(), n = x("footer"), p && p.c(), t(o, "class", "flex flex-col gap-2"), t(n, "class", "flex flex-col gap-4"), t(e, "class", "flex flex-col flex-grow justify-between");
    },
    m(h, u) {
      $(h, e, u), i(e, o), f && f.m(o, null), i(o, r), d && d.m(o, null), i(e, a), i(e, n), p && p.m(n, null), s = !0;
    },
    p(h, [u]) {
      /*isArgent*/
      h[0] ? f ? u & /*isArgent*/
      1 && w(f, 1) : (f = lr(h), f.c(), w(f, 1), f.m(o, r)) : f && (ce(), C(f, 1, 1, () => {
        f = null;
      }), fe()), /*store*/
      h[2] && /*storeLink*/
      h[3] ? d ? (d.p(h, u), u & /*store, storeLink*/
      12 && w(d, 1)) : (d = ar(h), d.c(), w(d, 1), d.m(o, null)) : d && (ce(), C(d, 1, 1, () => {
        d = null;
      }), fe()), /*isArgent*/
      h[0] ? p ? u & /*isArgent*/
      1 && w(p, 1) : (p = nr(h), p.c(), w(p, 1), p.m(n, null)) : p && (ce(), C(p, 1, 1, () => {
        p = null;
      }), fe());
    },
    i(h) {
      s || (w(f), w(d), w(p), s = !0);
    },
    o(h) {
      C(f), C(d), C(p), s = !1;
    },
    d(h) {
      h && M(e), f && f.d(), d && d.d(), p && p.d();
    }
  };
}
function qo(l, e, o) {
  let { isArgent: r = !1 } = e, { extensionName: a = "" } = e, { store: n } = e, { storeLink: s } = e;
  const f = {
    // @dev - Be mindful of name property length, it might break the UI
    chrome: { name: "Chrome", icon: mr },
    edge: { name: "Edge", icon: xo },
    firefox: { name: "Firefox", icon: Bo }
  };
  return l.$$set = (d) => {
    "isArgent" in d && o(0, r = d.isArgent), "extensionName" in d && o(1, a = d.extensionName), "store" in d && o(2, n = d.store), "storeLink" in d && o(3, s = d.storeLink);
  }, [r, a, n, s, f];
}
class Xo extends j {
  constructor(e) {
    super(), W(this, e, qo, Po, U, {
      isArgent: 0,
      extensionName: 1,
      store: 2,
      storeLink: 3
    });
  }
}
function Qo(l) {
  let e, o, r, a, n, s, f;
  return a = new wr({}), {
    c() {
      e = x("section"), o = x("div"), r = x("div"), A(a.$$.fragment), n = K(), s = x("div"), s.innerHTML = '<h3 class="text-primary text-h4 font-bold">Couldn&#39;t perform the action</h3> <p class="text-primary text-p3 font-[400]">Please try again.</p>', t(r, "class", "bg-button-secondary rounded-full p-5"), t(o, "class", "flex flex-col h-full justify-center items-center gap-4 w-full flex-grow"), t(e, "class", "flex flex-col justify-center items-center flex-grow");
    },
    m(d, p) {
      $(d, e, p), i(e, o), i(o, r), N(a, r, null), i(o, n), i(o, s), f = !0;
    },
    p: B,
    i(d) {
      f || (w(a.$$.fragment, d), f = !0);
    },
    o(d) {
      C(a.$$.fragment, d), f = !1;
    },
    d(d) {
      d && M(e), Z(a);
    }
  };
}
class Yo extends j {
  constructor(e) {
    super(), W(this, e, null, Qo, U, {});
  }
}
function ir(l) {
  let e, o, r, a, n, s, f, d, p;
  r = new n1({
    props: {
      handleBack: (
        /*func*/
        l[14]
      ),
      handleClose: (
        /*func_1*/
        l[15]
      ),
      title: (
        /*dappName*/
        l[2]
      ),
      showBackButton: (
        /*showBackButton*/
        l[5] && ![Q.walletList, Q.success].includes(
          /*layout*/
          l[0]
        )
      )
    }
  });
  const h = [
    nl,
    al,
    ll,
    ol,
    rl,
    tl,
    el,
    Jo
  ], u = [];
  function g(m, b) {
    return (
      /*layout*/
      m[0] === Q.walletList ? 0 : (
        /*layout*/
        m[0] === Q.connecting ? 1 : (
          /*layout*/
          m[0] === Q.success ? 2 : (
            /*layout*/
            m[0] === Q.loginFailure ? 3 : (
              /*layout*/
              m[0] === Q.requestFailure ? 4 : (
                /*layout*/
                m[0] === Q.qrCode ? 5 : (
                  /*layout*/
                  m[0] === Q.approval ? 6 : (
                    /*layout*/
                    m[0] === Q.download ? 7 : -1
                  )
                )
              )
            )
          )
        )
      )
    );
  }
  return ~(n = g(l)) && (s = u[n] = h[n](l)), {
    c() {
      e = x("div"), o = x("main"), A(r.$$.fragment), a = K(), s && s.c(), t(o, "role", "dialog"), t(o, "class", f = `
        rounded-3xl bg-surface-default shadow-modal dark:shadow-none flex flex-col
        z-[9999] w-full max-w-[380px] mx-6 p-6 text-center gap-8
        ${/*layout*/
      l[0] !== Q.walletList ? "min-h-[570px]" : ""}
      `), t(e, "part", "starknetkit-modal"), t(e, "class", d = `${/*darkModeControlClass*/
      l[1]} modal-font fixed inset-0 z-[9998] flex items-center justify-center backdrop-blur-sm bg-black/25`);
    },
    m(m, b) {
      $(m, e, b), i(e, o), N(r, o, null), i(o, a), ~n && u[n].m(o, null), l[20](e), p = !0;
    },
    p(m, b) {
      const v = {};
      b & /*nodeRef*/
      128 && (v.handleClose = /*func_1*/
      m[15]), b & /*dappName*/
      4 && (v.title = /*dappName*/
      m[2]), b & /*showBackButton, layout*/
      33 && (v.showBackButton = /*showBackButton*/
      m[5] && ![Q.walletList, Q.success].includes(
        /*layout*/
        m[0]
      )), r.$set(v);
      let F = n;
      n = g(m), n === F ? ~n && u[n].p(m, b) : (s && (ce(), C(u[F], 1, 1, () => {
        u[F] = null;
      }), fe()), ~n ? (s = u[n], s ? s.p(m, b) : (s = u[n] = h[n](m), s.c()), w(s, 1), s.m(o, null)) : s = null), (!p || b & /*layout*/
      1 && f !== (f = `
        rounded-3xl bg-surface-default shadow-modal dark:shadow-none flex flex-col
        z-[9999] w-full max-w-[380px] mx-6 p-6 text-center gap-8
        ${/*layout*/
      m[0] !== Q.walletList ? "min-h-[570px]" : ""}
      `)) && t(o, "class", f), (!p || b & /*darkModeControlClass*/
      2 && d !== (d = `${/*darkModeControlClass*/
      m[1]} modal-font fixed inset-0 z-[9998] flex items-center justify-center backdrop-blur-sm bg-black/25`)) && t(e, "class", d);
    },
    i(m) {
      p || (w(r.$$.fragment, m), w(s), p = !0);
    },
    o(m) {
      C(r.$$.fragment, m), C(s), p = !1;
    },
    d(m) {
      m && M(e), Z(r), ~n && u[n].d(), l[20](null);
    }
  };
}
function Jo(l) {
  let e, o;
  return e = new Xo({
    props: {
      store: Br(),
      isArgent: !!/*selectedConnector*/
      (l[9] && /*selectedConnector*/
      (l[9]?.id === "argentMobile" || /*selectedConnector*/
      l[9]?.id === "argentX")),
      storeLink: (
        /*selectedWallet*/
        l[4]?.download
      ),
      extensionName: (
        /*selectedWallet*/
        l[4]?.name.includes("Ready") ? "Ready Wallet (formerly Argent)" : (
          /*selectedConnector*/
          l[9]?.name
        )
      )
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, a) {
      const n = {};
      a & /*selectedConnector*/
      512 && (n.isArgent = !!/*selectedConnector*/
      (r[9] && /*selectedConnector*/
      (r[9]?.id === "argentMobile" || /*selectedConnector*/
      r[9]?.id === "argentX"))), a & /*selectedWallet*/
      16 && (n.storeLink = /*selectedWallet*/
      r[4]?.download), a & /*selectedWallet, selectedConnector*/
      528 && (n.extensionName = /*selectedWallet*/
      r[4]?.name.includes("Ready") ? "Ready Wallet (formerly Argent)" : (
        /*selectedConnector*/
        r[9]?.name
      )), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function el(l) {
  let e, o;
  return e = new $1({}), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p: B,
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function tl(l) {
  let e, o;
  return e = new lo({
    props: { handleInstallClick: (
      /*func_5*/
      l[19]
    ) }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p: B,
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function rl(l) {
  let e, o;
  return e = new Yo({}), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p: B,
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function ol(l) {
  let e, o;
  return e = new bo({
    props: {
      walletName: (
        /*selectedConnector*/
        l[9]?.name
      ),
      handleCallback: (
        /*func_3*/
        l[17]
      ),
      showFallback: (
        /*showFallback*/
        l[8]
      ),
      handleFallback: (
        /*func_4*/
        l[18]
      )
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, a) {
      const n = {};
      a & /*selectedConnector*/
      512 && (n.walletName = /*selectedConnector*/
      r[9]?.name), a & /*callback, selectedWallet*/
      80 && (n.handleCallback = /*func_3*/
      r[17]), a & /*showFallback*/
      256 && (n.showFallback = /*showFallback*/
      r[8]), a & /*callback, selectedWallet*/
      80 && (n.handleFallback = /*func_4*/
      r[18]), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function ll(l) {
  let e, o;
  return e = new ko({}), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p: B,
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function al(l) {
  let e, o;
  return e = new B1({
    props: {
      walletName: (
        /*selectedConnector*/
        l[9]?.name
      ),
      showFallback: (
        /*showFallback*/
        l[8]
      ),
      handleFallback: (
        /*func_2*/
        l[16]
      ),
      $$slots: { default: [il] },
      $$scope: { ctx: l }
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, a) {
      const n = {};
      a & /*selectedConnector*/
      512 && (n.walletName = /*selectedConnector*/
      r[9]?.name), a & /*showFallback*/
      256 && (n.showFallback = /*showFallback*/
      r[8]), a & /*callback, selectedWallet*/
      80 && (n.handleFallback = /*func_2*/
      r[16]), a & /*$$scope, selectedConnector, darkModeControlClass*/
      2097666 && (n.$$scope = { dirty: a, ctx: r }), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function nl(l) {
  let e, o;
  return e = new C1({
    props: {
      walletList: (
        /*modalWallets*/
        l[3]
      ),
      theme: (
        /*darkModeControlClass*/
        l[1]
      ),
      callback: (
        /*callback*/
        l[6]
      )
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, a) {
      const n = {};
      a & /*modalWallets*/
      8 && (n.walletList = /*modalWallets*/
      r[3]), a & /*darkModeControlClass*/
      2 && (n.theme = /*darkModeControlClass*/
      r[1]), a & /*callback*/
      64 && (n.callback = /*callback*/
      r[6]), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function sr(l) {
  let e, o;
  return e = new pr({
    props: {
      icon: (
        /*selectedConnector*/
        l[9].icon
      ),
      theme: (
        /*darkModeControlClass*/
        l[1]
      )
    }
  }), {
    c() {
      A(e.$$.fragment);
    },
    m(r, a) {
      N(e, r, a), o = !0;
    },
    p(r, a) {
      const n = {};
      a & /*selectedConnector*/
      512 && (n.icon = /*selectedConnector*/
      r[9].icon), a & /*darkModeControlClass*/
      2 && (n.theme = /*darkModeControlClass*/
      r[1]), e.$set(n);
    },
    i(r) {
      o || (w(e.$$.fragment, r), o = !0);
    },
    o(r) {
      C(e.$$.fragment, r), o = !1;
    },
    d(r) {
      Z(e, r);
    }
  };
}
function il(l) {
  let e, o, r = (
    /*selectedConnector*/
    l[9]?.icon && sr(l)
  );
  return {
    c() {
      r && r.c(), e = ye();
    },
    m(a, n) {
      r && r.m(a, n), $(a, e, n), o = !0;
    },
    p(a, n) {
      /*selectedConnector*/
      a[9]?.icon ? r ? (r.p(a, n), n & /*selectedConnector*/
      512 && w(r, 1)) : (r = sr(a), r.c(), w(r, 1), r.m(e.parentNode, e)) : r && (ce(), C(r, 1, 1, () => {
        r = null;
      }), fe());
    },
    i(a) {
      o || (w(r), o = !0);
    },
    o(a) {
      C(r), o = !1;
    },
    d(a) {
      a && M(e), r && r.d(a);
    }
  };
}
function sl(l) {
  let e, o, r = !/*isInAppBrowser*/
  l[11] && /*layout*/
  (l[0] === Q.walletList ? (
    /*modalWallets*/
    l[3].length > 1
  ) : !0) && ir(l);
  return {
    c() {
      r && r.c(), e = ye();
    },
    m(a, n) {
      r && r.m(a, n), $(a, e, n), o = !0;
    },
    p(a, [n]) {
      !/*isInAppBrowser*/
      a[11] && /*layout*/
      (a[0] !== Q.walletList || /*modalWallets*/
      a[3].length > 1) ? r ? (r.p(a, n), n & /*layout, modalWallets*/
      9 && w(r, 1)) : (r = ir(a), r.c(), w(r, 1), r.m(e.parentNode, e)) : r && (ce(), C(r, 1, 1, () => {
        r = null;
      }), fe());
    },
    i(a) {
      o || (w(r), o = !0);
    },
    o(a) {
      C(r), o = !1;
    },
    d(a) {
      a && M(e), r && r.d(a);
    }
  };
}
function cl(l, e, o) {
  let r, a, n, { dappName: s = window?.document.title ?? "" } = e, { layout: f = Q.walletList } = e;
  function d(_) {
    o(0, f = _);
  }
  function p() {
    return f;
  }
  let { modalWallets: h = [] } = e, { selectedWallet: u = null } = e, { showBackButton: g = !0 } = e, { callback: m = async () => {
  } } = e, b = xr(), { theme: v = "dark" } = e, { darkModeControlClass: F = v === "dark" ? "dark" : "" } = e;
  Vr(async () => {
    if (v === "dark" || v == null && window.matchMedia("(prefers-color-scheme: dark)").matches ? o(1, F = "dark") : o(1, F = "light"), b) {
      try {
        setTimeout(() => {
          m(Gt(new Mr()));
        });
      } catch (E) {
        console.error(E);
      }
      return;
    }
    if (navigator?.userAgent?.toLowerCase()?.includes("braavos") && window?.starknet_braavos) {
      try {
        setTimeout(() => {
          m(Gt(new Sr()));
        });
      } catch (E) {
        console.error(E);
      }
      return;
    }
    if (h.length === 1)
      try {
        await m(h[0]);
      } catch (E) {
        console.error(E);
      }
  });
  const O = () => d(Q.walletList), z = () => n?.parentNode?.removeChild(n), L = () => m(u, !0), k = () => m(u), y = () => m(u, !0), S = () => d(Q.download);
  function G(_) {
    Nt[_ ? "unshift" : "push"](() => {
      n = _, o(7, n);
    });
  }
  return l.$$set = (_) => {
    "dappName" in _ && o(2, s = _.dappName), "layout" in _ && o(0, f = _.layout), "modalWallets" in _ && o(3, h = _.modalWallets), "selectedWallet" in _ && o(4, u = _.selectedWallet), "showBackButton" in _ && o(5, g = _.showBackButton), "callback" in _ && o(6, m = _.callback), "theme" in _ && o(13, v = _.theme), "darkModeControlClass" in _ && o(1, F = _.darkModeControlClass);
  }, l.$$.update = () => {
    l.$$.dirty & /*selectedWallet*/
    16 && o(9, r = u?.connector && Fr(u.connector)), l.$$.dirty & /*selectedWallet*/
    16 && o(8, a = !!(u && Lr(u?.connector) && u?.connector?.fallbackConnector));
  }, [
    f,
    F,
    s,
    h,
    u,
    g,
    m,
    n,
    a,
    r,
    d,
    b,
    p,
    v,
    O,
    z,
    L,
    k,
    y,
    S,
    G
  ];
}
class wl extends j {
  constructor(e) {
    super(), W(this, e, cl, sl, U, {
      dappName: 2,
      layout: 0,
      getLayout: 12,
      modalWallets: 3,
      selectedWallet: 4,
      showBackButton: 5,
      callback: 6,
      theme: 13,
      darkModeControlClass: 1
    });
  }
  get getLayout() {
    return this.$$.ctx[12];
  }
}
export {
  Q as L,
  wl as M,
  R1 as a,
  I1 as d,
  bl as g
};
